{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e8fb75c1",
   "metadata": {},
   "source": [
    "# Project 4. \"Tardigrades: from genestealers to space marines\". Eukaryotic genome analysis.\n",
    "\n",
    "Lab notebook by Nikita Vyatkin and Guseva Polina"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c06e3e75",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "984c1d9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def show_file_content(path, lines=10, headers_only=False):\n",
    "    if headers_only:\n",
    "        ! mkdir -p temp\n",
    "        ! touch temp/show_file_content.tmp\n",
    "        ! grep '>' {path} > temp/show_file_content.tmp\n",
    "        path = \"temp/show_file_content.tmp\"\n",
    "    \n",
    "    ! head -{lines} {path}\n",
    "    print(\"...\")\n",
    "    ! tail -{lines} {path}\n",
    "    \n",
    "    if headers_only:\n",
    "        ! rm {path}\n",
    "\n",
    "\n",
    "def count_seqs(path):\n",
    "    res = ! grep '>' {path} | wc -l\n",
    "    print(f\"Total: {res[0]} sequencies\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "668b748b",
   "metadata": {},
   "source": [
    "## Data source\n",
    "\n",
    "For this project we will be using a sequence of the _Ramazzottius varieornatus_, the YOKOZUNA-1 strain (sequenced in the University of Tokyo and named after the highest rank in professional sumo).\n",
    "\n",
    "One can download assembled genome [here](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b0b32a6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "! wget -P data/ https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
    "! gunzip data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
    "! mv data/GCA_001949185.1_Rvar_4.0_genomic.fna data/Ramazzottius_varieornatus.fna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a7a139c7",
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls -lh data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4190903f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BDGG01000001.1 Ramazzottius varieornatus DNA, contig: Scaffold001, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      "TGCTGCGACATACTCGAAACACCCAAGGTCTTACCGGGAGAATATCGTCGTCTGGAGGATTTGTATGGGCCATTGTGATG\n",
      "GCCGCTACCTGGCATTGTACAACTTGAATGGTTGGTCTGAGCGGGACATGATCGACAGGTGGGCTTTGATGGTGACACTT\n",
      "TTGGTGCCGCCGGCATTCCTTCTGCCCGCACGTGACATCGACAACCTCTTGACTGCCGGCCATGAAACCTCACCCCAAAC\n",
      "AGGTATTCAGTGGGCAGAAAAAGCTACAAAAATCCTATCGCACGGGAATAAGTCCGTGCTAAACTAGTTAGAGCGACTTT\n",
      "TGAAAGGAAGATGTAGAGTTTAGTTCATTGGAGTAATCAAGTCTTCGGTTAgtttcgtgacgtcacgattGATCCGTATG\n",
      "GTGAAGCGTCTGAGCACAGAGTTTTCTTATTTTTCAACAGGTGGGAGCGACGTCATGAGCGAGTATGTGACGCCCTTCTG\n",
      "TTACGTCTGTGATTCTCTCGTGGATGCCGACTGCCGAGAATCGCAAGACTGGATGCCTCGACGAAGAAACGGCttaagga\n",
      "agaaagtgatgTGGACAGGCCAAGGGGAGGACGATCATGGCAGACTTGCCTTCGACTGGAAAGAAGTCTACATtcggctg\n",
      "gaaaagttcaaGCAGCGATGTCCGTCCAACAATGGCTGCACCACTCACGTCTTCCTCAATAGCGGTACGTATCTTGCTGT\n",
      "...\n",
      "TCTTCTTTTAAGACAAGCCGGCTTGTCTTATTTAATATTAATCTTTATTTTTCTTTCTTCTTTTATAGCTATATTCATTA\n",
      "ATATAAAACAAAATTTATTTAAAAGAATGTTGACAGTTTCAAGATTGTCCAACACCAGATGACTCTTGACCGCAGTTGTA\n",
      "ACAACATATTCTGGATGAATATTTTATTTTGCAGTGTACTCATTTTCTTTAGTATTAAGTTCAACAATAAATAATAAGTA\n",
      "TTTAGACCAAAATTACCGTAATAGAGACACGCTAGAAAACCCCCTTTTATTTTTAAGCATAGGAGGACTCCCCCCCATAC\n",
      "CAGGATTTTTCCCTAAAATTTTAATTCTGATAAATTTAAGATATTCTCGGTTAAGATTGCTAAGAATTTTTCTCCTTCTT\n",
      "GCCGCGTGTGTAGATATTTTTATTTATTTACGATTTTCCTTCACAAGAGCATTGAAAAATAATGGTAATCTTATATATTG\n",
      "TAAAGCTGTAACAACAAAGAGAATTAAAATACTGTCTTTAAGAGCAGTCACTATAGTCACAATAATAGCAATTATATTAT\n",
      "AAAAAGATTTTAAGTTAAAAAAACTAATGTGTTTCAAACACTTCAATAATGAAAATTTAAATCTTAATTTTAAAGAAAAA\n",
      "CTTTTTAAATATGCTTGCAACATATCATATAAATAAAATTTCATAACAAAAAGAAAAAATCTATATGTAAATTTACAATT\n",
      "TACCGCCTATACTCAGCCATTTTGACTA\n",
      "Total: 200 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/Ramazzottius_varieornatus.fna\")\n",
    "count_seqs(\"data/Ramazzottius_varieornatus.fna\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8ab4773f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BDGG01000001.1 Ramazzottius varieornatus DNA, contig: Scaffold001, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000002.1 Ramazzottius varieornatus DNA, contig: Scaffold002, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000003.1 Ramazzottius varieornatus DNA, contig: Scaffold003, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      "...\n",
      ">BDGG01000198.1 Ramazzottius varieornatus DNA, contig: Scaffold277, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000199.1 Ramazzottius varieornatus DNA, contig: Scaffold279, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">AP017609.1 Ramazzottius varieornatus mitochondrial DNA, complete genome, strain: YOKOZUNA-1\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/Ramazzottius_varieornatus.fna\", lines=3, headers_only=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b577d86",
   "metadata": {},
   "source": [
    "## Structural annotation\n",
    "\n",
    "One can download AUGUSTUS results [here](https://drive.google.com/file/d/1wBxf6cDgu22NbjAOgTe-8b3Zx60hNKY0/view?usp=drive_web).\n",
    "\n",
    "[Script](http://augustus.gobics.de/binaries/scripts/getAnnoFasta.pl) for GFF to FASTA converting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8f182b8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls -lhR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a0a0ebbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "! chmod +x getAnnoFasta.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fda5f0da",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "97cda1bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "! ./getAnnoFasta.pl data/augustus.whole.gff\n",
    "! mv data/augustus.whole.aa proteins/all_proteins.faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "497d3541",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">g1.t1\n",
      "AATYSKHPRSYRENIVVWRICMGHCDGRYLALYNLNGWSERDMIDRWALMVTLLVPPAFLLPARDIDNLLTAGHETSPQTGGSDVMSEYVTPFCYVCDSL\n",
      "VDADCRESQDWMPRRRNGLRKKVMWTGQGEDDHGRLAFDWKEVYIRLEKFKQRCPSNNGCTTHVFLNSDRVIRRCFDPKEPPCSGHEWMVNRRNPKKRSK\n",
      "VKVVSRLTCADWPFCNVDSAYYWPNGTVEALLSPHDDPQLKVRSAKLSCYECDSILQPDCTGQNFVKAVRKPNDAVAHSVDWVGLRKGMEKFRTRCPVES\n",
      "GCFFQGDHMTNRVIRGCRNMTTDLPRHLLVKDRRKESSVFRTEEVCRESECNSRVLFFSHSGAFLDSDWHFNNSSAGRDNGQGEAKKEVYQGDAPATEET\n",
      "GFLPHVKQSGQGKRKEGRLQKRTIGDVEREERRADEARQRRGKPKAEGRGGAGTRVLSFGHRSRKTSAGGKLDSCWLLLLLPFFLL\n",
      ">g2.t1\n",
      "MTFAVTVATTSQSCAIAFDVFWLTTIKLFFIAKYFGSEFTPFDKVQSLTCPHCAQQGIEYKDLLPHLETNHADTSDFLVVSDNITIIWDVAHCI\n",
      ">g3.t1\n",
      "MLDKAKLVLLNKGGRPSIRSAEDGWKVFSDSLLCRASSDGRVFIGKGGGNGGAVAKAEVAAIGVVLVHHMLQQVVRHGEVVIVSANTNWTPLPY\n",
      "...\n",
      "FPHCLGYVDGSLIGLQEAPSMCGEDYYTRKGSYAINAMIVCDDLKRIRHVLAGFPGCSHDHRVFMYSKLALNPFKYFSGHEYLLADSAYPVSDIIVPTFK\n",
      "KPLANSH\n",
      ">g16433.t1\n",
      "MHLDADIDESSKEEKPEIVMEYNRTKAGVDTLDQLTGNYSCRRKTSRWPMALFYDILDISTLDAYIIWCEINPGWNSTLPTKRRMFLQDVSKKMMQRQLL\n",
      "RRSTTPSISFSMKKFVSTVVPRVQVSSDQAVDEMANTSKPKALQKKPTR\n",
      ">g16434.t1\n",
      "RIHLLADCHLLKTLNLLQRDREVFPKVDLKSGVIWKVAFFSQALAAMERSVSQKNFGSPVLV\n",
      ">g16435.t1\n",
      "MKRPGGQKYHPWSDETKLKSGDVGGANHPWADEAKLNQHPKCAVLSSKLSNVTSMMKFLATGPTGECMSESRYTAFLAGQKNSLGVATKVAGGITSPSQL\n",
      "CLCKPLLYEEFLTFSNSMVMCETGLDDTCLTRKFK\n",
      "Total: 16435 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"proteins/all_proteins.faa\")\n",
    "count_seqs(\"proteins/all_proteins.faa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3d4968d",
   "metadata": {},
   "source": [
    "## Physical localization\n",
    "\n",
    "Considering that DNA is a major target of radiation damage, we may hypothesize that tardigrades might have unique proteins associated with their DNA to protect and/or effectively repair it. To explore this possibility, we will combine genomic and proteomic data.\n",
    "\n",
    "[List](https://disk.yandex.ru/d/xJqQMGX77Xueqg) of peptides that were associated with the Tardigrade's DNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8a8964f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".:\r\n",
      "total 48K\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  5 18:41 \u001b[0m\u001b[01;34mannotation\u001b[0m/\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 16:25 \u001b[01;34mdata\u001b[0m/\r\n",
      "-rwxrwxr-x 1 nikitos nikitos 6,6K дек  5 18:27 \u001b[01;32mgetAnnoFasta.pl\u001b[0m*\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  21K дек  8 15:39 Lab_Notebook.ipynb\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 17:10 \u001b[01;34mtemp\u001b[0m/\r\n",
      "-rw-rw-r-- 1 nikitos nikitos   76 дек  7 17:23 temp.txt\r\n",
      "\r\n",
      "./annotation:\r\n",
      "total 7,0M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 7,0M дек  5 18:41 augustus.whole.aa\r\n",
      "\r\n",
      "./data:\r\n",
      "total 81M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 27M дек  5 18:21 augustus.whole.gff\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 737 дек  7 16:22 dna_associated_peptides.fa\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n",
      "\r\n",
      "./temp:\r\n",
      "total 0\r\n"
     ]
    }
   ],
   "source": [
    "ls -lhR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "bd887bff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">1\n",
      "VFSALDVLR\n",
      ">2\n",
      "SPLQTDEIR\n",
      ">3\n",
      "AATGAVLSSADSR\n",
      "...\n",
      ">41\n",
      "GPEDEAGGPPK\n",
      ">42\n",
      "SSGGNSPDVVVR\n",
      ">43\n",
      "STVGQTPPQNLQR\n",
      "Total: 43 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/dna_associated_peptides.fa\", lines=6)\n",
    "count_seqs(\"data/dna_associated_peptides.fa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "d0ae5e22",
   "metadata": {},
   "outputs": [],
   "source": [
    "peptides = []\n",
    "\n",
    "with open(\"data/dna_associated_peptides.fa\", 'r') as f_in:\n",
    "    for line in f_in:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            peptides.append([line[1:], []])\n",
    "        elif line:\n",
    "            peptides[-1][1].append(line)\n",
    "\n",
    "    for peptide in peptides:\n",
    "        peptide[0] = int(peptide[0])\n",
    "        peptide[1] = ''.join(peptide[1])\n",
    "\n",
    "peptides = dict(peptides)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ef5d9ae2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: 'VFSALDVLR',\n",
       " 2: 'SPLQTDEIR',\n",
       " 3: 'AATGAVLSSADSR',\n",
       " 4: 'LLDDENDYELK',\n",
       " 5: 'VTGSSQGAINQQQAK',\n",
       " 6: 'VTGSSAQQIDINQAR',\n",
       " 7: 'LTSSGTGAGSAPAAAK',\n",
       " 8: 'NIPVGGVNTEATGDNYIR',\n",
       " 9: 'QGGMGMSGGMGGADR',\n",
       " 10: 'GPEDEAGGPPK',\n",
       " 11: 'LTDDELNQAMK',\n",
       " 12: 'AATGAVQSSASK',\n",
       " 13: 'VLLDNQDDYELK',\n",
       " 14: 'VFSALDVLR',\n",
       " 15: 'EIYANAQPGK',\n",
       " 16: 'QIMDNEGLR',\n",
       " 17: 'SAPLSASPISAR',\n",
       " 18: 'SSSSSSQESQSSSSQVR',\n",
       " 19: 'STVGQTPPQNLQR',\n",
       " 20: 'APVNGIVTDANGNQIQVR',\n",
       " 21: 'TNAAGQLIGPGGIAINDAGLTR',\n",
       " 22: 'VFSALDVLR',\n",
       " 23: 'SPLQTDEIR',\n",
       " 24: 'QFQDYSNSR',\n",
       " 25: 'AATGAVLSSADSR',\n",
       " 26: 'VFSALDVLR',\n",
       " 27: 'EIYANAQPGK',\n",
       " 28: 'QFQDYSNSR',\n",
       " 29: 'NIPVGGVNTEATGDNYIR',\n",
       " 30: 'AATGAVQSSASK',\n",
       " 31: 'VLLDNQDDYELK',\n",
       " 32: 'SGEQAMVDAK',\n",
       " 33: 'SSTTTSTMNAAK',\n",
       " 34: 'TTTTTGTTGTSGTTMPPK',\n",
       " 35: 'VVEQTGASSSR',\n",
       " 36: 'SSSSSSQESQSSSSQVR',\n",
       " 37: 'NTSFNTTGDR',\n",
       " 38: 'GYSYTNTTTYR',\n",
       " 39: 'AVTGAVAGVK',\n",
       " 40: 'QFQDYSNSR',\n",
       " 41: 'GPEDEAGGPPK',\n",
       " 42: 'SSGGNSPDVVVR',\n",
       " 43: 'STVGQTPPQNLQR'}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2e1f1585",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(peptides.values()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3cdd9dbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_vals = set(peptides.values())\n",
    "new_peptides = dict()\n",
    "\n",
    "for key, val in peptides.items():\n",
    "    if val in unique_vals:\n",
    "        new_peptides[key] = val\n",
    "        unique_vals -= {val}\n",
    "\n",
    "peptides = new_peptides\n",
    "\n",
    "with open(\"data/dna_associated_peptides.fa\", 'w') as f_out:\n",
    "    for key, val in peptides.items():\n",
    "        f_out.write(f\">{key}\\n\")\n",
    "        f_out.write(val)\n",
    "        f_out.write('\\n')\n",
    "    \n",
    "len(peptides)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1effd975",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: 'VFSALDVLR',\n",
       " 2: 'SPLQTDEIR',\n",
       " 3: 'AATGAVLSSADSR',\n",
       " 4: 'LLDDENDYELK',\n",
       " 5: 'VTGSSQGAINQQQAK',\n",
       " 6: 'VTGSSAQQIDINQAR',\n",
       " 7: 'LTSSGTGAGSAPAAAK',\n",
       " 8: 'NIPVGGVNTEATGDNYIR',\n",
       " 9: 'QGGMGMSGGMGGADR',\n",
       " 10: 'GPEDEAGGPPK',\n",
       " 11: 'LTDDELNQAMK',\n",
       " 12: 'AATGAVQSSASK',\n",
       " 13: 'VLLDNQDDYELK',\n",
       " 15: 'EIYANAQPGK',\n",
       " 16: 'QIMDNEGLR',\n",
       " 17: 'SAPLSASPISAR',\n",
       " 18: 'SSSSSSQESQSSSSQVR',\n",
       " 19: 'STVGQTPPQNLQR',\n",
       " 20: 'APVNGIVTDANGNQIQVR',\n",
       " 21: 'TNAAGQLIGPGGIAINDAGLTR',\n",
       " 24: 'QFQDYSNSR',\n",
       " 32: 'SGEQAMVDAK',\n",
       " 33: 'SSTTTSTMNAAK',\n",
       " 34: 'TTTTTGTTGTSGTTMPPK',\n",
       " 35: 'VVEQTGASSSR',\n",
       " 37: 'NTSFNTTGDR',\n",
       " 38: 'GYSYTNTTTYR',\n",
       " 39: 'AVTGAVAGVK',\n",
       " 42: 'SSGGNSPDVVVR'}"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9a3434e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MQRLGYPQREQYPHNQRQSILEKLIRAVDQALKNPDRKQRVEIESWLKRNLPKDPKLQLKLASDARGEVSLEDLLMTKEDLEAASKALLNESSKSGLHLPGLDESVQNIWPNDARGVLSLTELLVEAPAEPLRIFPSTYVADLIVTLVASLLEKEEVTLQKVLEMFPQLRNTFKPVETYLRKXXXXXXXXXXXXXXXXXXXXDEQRKADEASGKVSPANARSPSKSPVKLELQHKPSARSAKEIKSPH'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins = []\n",
    "\n",
    "with open(\"proteins/all_proteins.faa\", 'r') as f_in:\n",
    "    for line in f_in:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            proteins.append([line[1:], []])\n",
    "        elif line:\n",
    "            proteins[-1][1].append(line)\n",
    "\n",
    "    for protein in proteins:\n",
    "        protein[1] = ''.join(protein[1])\n",
    "\n",
    "proteins = dict(proteins)\n",
    "proteins[\"g25.t1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "599747c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-12-08 16:12:47--  http://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
      "Resolving github.com (github.com)... 140.82.121.4\n",
      "Connecting to github.com (github.com)|140.82.121.4|:80... connected.\n",
      "HTTP request sent, awaiting response... 301 Moved Permanently\n",
      "Location: https://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz [following]\n",
      "--2022-12-08 16:12:47--  https://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
      "Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/31987083/4420191b-492f-4ec0-a80c-080436691f1e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221208T131247Z&X-Amz-Expires=300&X-Amz-Signature=b556836ddecfca6fd02dbf98564ee0598fd7f709f66560d9597b6863008f8adb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=31987083&response-content-disposition=attachment%3B%20filename%3Ddiamond-linux64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
      "--2022-12-08 16:12:47--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/31987083/4420191b-492f-4ec0-a80c-080436691f1e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221208T131247Z&X-Amz-Expires=300&X-Amz-Signature=b556836ddecfca6fd02dbf98564ee0598fd7f709f66560d9597b6863008f8adb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=31987083&response-content-disposition=attachment%3B%20filename%3Ddiamond-linux64.tar.gz&response-content-type=application%2Foctet-stream\n",
      "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.109.133, 185.199.108.133, ...\n",
      "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 13078237 (12M) [application/octet-stream]\n",
      "Saving to: ‘diamond-linux64.tar.gz’\n",
      "\n",
      "diamond-linux64.tar 100%[===================>]  12,47M  4,67MB/s    in 2,7s    \n",
      "\n",
      "2022-12-08 16:12:50 (4,67 MB/s) - ‘diamond-linux64.tar.gz’ saved [13078237/13078237]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# downloading the tool\n",
    "! wget http://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
    "! tar xzf diamond-linux64.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "31cf4083",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir search_diamond"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "64d11657",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\n",
      "\n",
      "#CPU threads: 12\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "Database input file: proteins/all_proteins.faa\n",
      "Opening the database file...  [0s]\n",
      "Loading sequences...  [0.021s]\n",
      "Masking sequences...  [0.037s]\n",
      "Writing sequences...  [0.003s]\n",
      "Hashing sequences...  [0.001s]\n",
      "Loading sequences...  [0s]\n",
      "Writing trailer...  [0s]\n",
      "Closing the input file...  [0s]\n",
      "Closing the database file...  [0.001s]\n",
      "\n",
      "Database sequences  16435\n",
      "  Database letters  7015168\n",
      "     Database hash  590081acba74e419860a4dfdcf0de70e\n",
      "        Total time  0.065000s\n"
     ]
    }
   ],
   "source": [
    "# creating a diamond-formatted database file\n",
    "! ./diamond makedb --in proteins/all_proteins.faa -d search_diamond/reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "ea9462ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\n",
      "\n",
      "#CPU threads: 12\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "Temporary directory: search_diamond\n",
      "#Target sequences to report alignments for: 25\n",
      "Opening the database...  [0.004s]\n",
      "Database: search_diamond/reference.dmnd (type: Diamond database, sequences: 16435, letters: 7015168)\n",
      "Block size = 2000000000\n",
      "Opening the input file...  [0s]\n",
      "Opening the output file...  [0s]\n",
      "Loading query sequences...  [0s]\n",
      "Masking queries...  [0.001s]\n",
      "Algorithm: Double-indexed\n",
      "Building query histograms...  [0s]\n",
      "Allocating buffers...  [0s]\n",
      "Loading reference sequences...  [0.006s]\n",
      "Masking reference...  [0.036s]\n",
      "Initializing temporary storage...  [0s]\n",
      "Building reference histograms...  [0.029s]\n",
      "Allocating buffers...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 1/4.\n",
      "Building reference seed array...  [0.011s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0.011s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 2/4.\n",
      "Building reference seed array...  [0.017s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 3/4.\n",
      "Building reference seed array...  [0.013s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 4/4.\n",
      "Building reference seed array...  [0.01s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 1/4.\n",
      "Building reference seed array...  [0.013s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 2/4.\n",
      "Building reference seed array...  [0.014s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 3/4.\n",
      "Building reference seed array...  [0.012s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 4/4.\n",
      "Building reference seed array...  [0.011s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Deallocating buffers...  [0s]\n",
      "Clearing query masking...  [0s]\n",
      "Computing alignments...  [0.003s]\n",
      "Deallocating reference...  [0s]\n",
      "Loading reference sequences...  [0s]\n",
      "Deallocating buffers...  [0s]\n",
      "Deallocating queries...  [0s]\n",
      "Loading query sequences...  [0s]\n",
      "Closing the input file...  [0s]\n",
      "Closing the output file...  [0s]\n",
      "Cleaning up...  [0s]\n",
      "Total time = 0.262s\n",
      "Reported 3 pairwise alignments, 3 HSPs.\n",
      "3 queries aligned.\n"
     ]
    }
   ],
   "source": [
    "# running a search in blastp mode\n",
    "! ./diamond blastp -d search_diamond/reference.dmnd -q data/dna_associated_peptides.fa -o search_diamond/matches.tsv -f 6 --iterate --header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "302a4c97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# DIAMOND v2.0.15. http://github.com/bbuchfink/diamond\r\n",
      "# Invocation: ./diamond blastp -d search_diamond/reference.dmnd -q data/dna_associated_peptides.fa -o search_diamond/matches.tsv -f 6 --iterate --header\r\n",
      "# Fields: Query ID, Subject ID, Percentage of identical matches, Alignment length, Number of mismatches, Number of gap openings, Start of alignment in query, End of alignment in query, Start of alignment in subject, End of alignment in subject, Expected value, Bit score\r\n",
      "8\tg4106.t1\t100\t18\t0\t0\t1\t18\t222\t239\t5.55e-07\t40.8\r\n",
      "20\tg12510.t1\t100\t18\t0\t0\t1\t18\t425\t442\t1.43e-06\t39.7\r\n",
      "21\tg12510.t1\t100\t22\t0\t0\t1\t22\t443\t464\t2.46e-09\t47.8\r\n"
     ]
    }
   ],
   "source": [
    "! cat search_diamond/matches.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "980d5f2b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MCAIWENSLVGSTDNIVMRNVVAAEETISRLLCHLTSIDMWVSQNPSFDLRQKKARLPSQSRCIVRWVCLQASLQVVKDAQLAFVTVCFVLQLMEKKCAWNVTSRSIIMVTTTTKTLNYSIIRGNHTNSVNATITSTPGFNINYFGKIPGILKVLELVTIVVALGLAASRRREAPGLELPAADSWRIEYFLTGEVFFLCVMSAGLMWLVAILLTYMMHVVSSVITPKTTIHEVVFNFLFMLMFWTAGIVELAQTVPNSYPYNGRPGMEQGDWGCRVAAGALSLFIGTFFLISFALAVKELQGPKRSAPLSASPISARDHPIVTVVALGLAASRRREGTTTVVGWPVEYFLAGEVFFLCICAAALTYLFGQLTTFAWSVVSSVLIPKTTYFEIVINFLLFLAWATAGIVQLAMTCREAWNGREIRAPVNGIVTDANGNQIQVRTNAAGQLIGPGGIAINDAGLTRAVYRSPDMLFQDWGCRIAAGSLALFLSIFHLISFALAVKEFQGPRNEAR',\n",
       " 'APVNGIVTDANGNQIQVR',\n",
       " 'TNAAGQLIGPGGIAINDAGLTR')"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g12510.t1\"], peptides[20], peptides[21]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "cfd54fd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('g702.t1', '18', 174),\n",
       " ('g702.t1', '35', 157),\n",
       " ('g1285.t1', '24', 137),\n",
       " ('g3428.t1', '10', 159),\n",
       " ('g3428.t1', '11', 128),\n",
       " ('g3679.t1', '42', 20),\n",
       " ('g4106.t1', '8', 221),\n",
       " ('g5237.t1', '19', 90),\n",
       " ('g5502.t1', '24', 135),\n",
       " ('g5503.t1', '24', 135),\n",
       " ('g5510.t1', '16', 168),\n",
       " ('g5616.t1', '12', 154),\n",
       " ('g5616.t1', '13', 142),\n",
       " ('g5641.t1', '1', 24),\n",
       " ('g5641.t1', '2', 35),\n",
       " ('g5641.t1', '3', 154),\n",
       " ('g5641.t1', '4', 143),\n",
       " ('g5641.t1', '24', 127),\n",
       " ('g10513.t1', '9', 304),\n",
       " ('g10514.t1', '37', 220),\n",
       " ('g10514.t1', '38', 41),\n",
       " ('g12510.t1', '17', 305),\n",
       " ('g12510.t1', '20', 424),\n",
       " ('g12510.t1', '21', 442),\n",
       " ('g12562.t1', '24', 127),\n",
       " ('g12562.t1', '39', 160),\n",
       " ('g13530.t1', '5', 223),\n",
       " ('g13530.t1', '6', 87),\n",
       " ('g14472.t1', '7', 367),\n",
       " ('g15153.t1', '1', 24),\n",
       " ('g15153.t1', '15', 44),\n",
       " ('g15153.t1', '24', 127),\n",
       " ('g15484.t1', '32', 830),\n",
       " ('g15484.t1', '33', 790),\n",
       " ('g15484.t1', '34', 754)]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def hamming_distance(s1: str, s2: str) -> int:\n",
    "    assert len(s1) == len(s2), \"Strings have different lengths\"\n",
    "    for i in range(len(s1)):\n",
    "        if s1[i] != s2[i]:\n",
    "            return False\n",
    "    return True\n",
    "\n",
    "\n",
    "def pattern_matching(text: str, pattern: str, mismatches: int = 0) -> list[int]:\n",
    "    res = []\n",
    "    pattern_len = len(pattern)\n",
    "    for i in range(len(text) - pattern_len + 1):\n",
    "        if hamming_distance(text[i:i + pattern_len], pattern):\n",
    "            res.append(i)\n",
    "    return res\n",
    "\n",
    "matches = []\n",
    "for protein in proteins.items():\n",
    "    for peptide in peptides.items():\n",
    "        res = pattern_matching(protein[1], peptide[1])\n",
    "        for ind in res:\n",
    "            matches.append((protein[0], peptide[0], ind))\n",
    "\n",
    "matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "98076cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "35"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matches)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0d9a02b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MKCLIVFSAFVASALAANGDQQPKVWDALNVLRGKDALFADQIQELLMNVVPGQDFPLLSEVPDTAVDCSRFKQPGFYADTEQGRCQVFHRCDINGHLTGYLCPNMTLFNQITLVCDWFYNVDCSQARQFYDYSNSRLYQEGVALLDDQNDFEVRARVVEQTGASSSRRLSGSRSSSSSSQESQSSSSQVRRRKSRKSKXXXXXXXXQSVSQEFA',\n",
       " 'SSSSSSQESQSSSSQVR',\n",
       " 'VVEQTGASSSR')"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g702.t1\"], peptides[\"18\"], peptides[\"35\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d2d58cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['g702.t1',\n",
       " 'g1285.t1',\n",
       " 'g3428.t1',\n",
       " 'g3679.t1',\n",
       " 'g4106.t1',\n",
       " 'g5237.t1',\n",
       " 'g5502.t1',\n",
       " 'g5503.t1',\n",
       " 'g5510.t1',\n",
       " 'g5616.t1',\n",
       " 'g5641.t1',\n",
       " 'g10513.t1',\n",
       " 'g10514.t1',\n",
       " 'g12510.t1',\n",
       " 'g12562.t1',\n",
       " 'g13530.t1',\n",
       " 'g14472.t1',\n",
       " 'g15153.t1',\n",
       " 'g15484.t1']"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matched_proteins = [matches[0][0]]\n",
    "for prot, *_ in matches:\n",
    "    if matched_proteins[-1] != prot:\n",
    "        matched_proteins.append(prot)\n",
    "matched_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "0bcb96e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"matched_proteins.fa\", 'w') as f_out:\n",
    "    for protein_id in matched_proteins:\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "33ea64b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir filter_blast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b792e333",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 12/15/2022 14:40:37\n",
      "New DB name:   /home/nikitos/Bioinformatics/BI/Workshop_on_bioinformatics/Project4/filter_blast/reference\n",
      "New DB title:  proteins/all_proteins.faa\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 16435 sequences in 0.177495 seconds.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! makeblastdb -in proteins/all_proteins.faa -dbtype prot -out filter_blast/reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1e18e797",
   "metadata": {},
   "outputs": [],
   "source": [
    "! blastp -db filter_blast/reference -query data/dna_associated_peptides.fa -outfmt 6 -out filter_blast/matches.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "00469688",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>33</td>\n",
       "      <td>2.000</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>33</td>\n",
       "      <td>2.100</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>1.500</td>\n",
       "      <td>21.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>3.300</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>6.700</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>37</td>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>221</td>\n",
       "      <td>230</td>\n",
       "      <td>0.970</td>\n",
       "      <td>22.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>37</td>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>77.778</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>359</td>\n",
       "      <td>367</td>\n",
       "      <td>8.400</td>\n",
       "      <td>19.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>38</td>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>42</td>\n",
       "      <td>52</td>\n",
       "      <td>0.600</td>\n",
       "      <td>23.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>39</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>161</td>\n",
       "      <td>170</td>\n",
       "      <td>0.820</td>\n",
       "      <td>22.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>42</td>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>21</td>\n",
       "      <td>32</td>\n",
       "      <td>0.046</td>\n",
       "      <td>26.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    0          1        2   3   4   5   6   7    8    9      10    11\n",
       "0    1   g5641.t1  100.000   9   0   0   1   9   25   33  2.000  21.6\n",
       "1    1  g15153.t1  100.000   9   0   0   1   9   25   33  2.100  21.6\n",
       "2    2   g5641.t1  100.000   9   0   0   1   9   36   44  1.500  21.9\n",
       "3    2  g12562.t1   88.889   9   1   0   1   9   36   44  3.300  20.8\n",
       "4    2   g5616.t1   88.889   9   1   0   1   9   36   44  6.700  20.0\n",
       "..  ..        ...      ...  ..  ..  ..  ..  ..  ...  ...    ...   ...\n",
       "63  37  g10514.t1  100.000  10   0   0   1  10  221  230  0.970  22.7\n",
       "64  37   g2203.t1   77.778   9   2   0   1   9  359  367  8.400  19.6\n",
       "65  38  g10514.t1  100.000  11   0   0   1  11   42   52  0.600  23.1\n",
       "66  39  g12562.t1  100.000  10   0   0   1  10  161  170  0.820  22.7\n",
       "67  42   g3679.t1  100.000  12   0   0   1  12   21   32  0.046  26.6\n",
       "\n",
       "[68 rows x 12 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\"filter_blast/matches.tsv\", sep='\\t', header=None)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a9121596",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>3.300</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>6.700</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>77.778</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>7.500</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>72.727</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>1.600</td>\n",
       "      <td>21.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>72.727</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>3.100</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4</td>\n",
       "      <td>g702.t1</td>\n",
       "      <td>63.636</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>145</td>\n",
       "      <td>155</td>\n",
       "      <td>3.700</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>63.636</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>5.400</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>8</td>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>62.500</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>64</td>\n",
       "      <td>79</td>\n",
       "      <td>0.650</td>\n",
       "      <td>23.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>8</td>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>57.143</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>134</td>\n",
       "      <td>147</td>\n",
       "      <td>8.800</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9</td>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>152</td>\n",
       "      <td>166</td>\n",
       "      <td>3.900</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9</td>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>224</td>\n",
       "      <td>238</td>\n",
       "      <td>4.900</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>204</td>\n",
       "      <td>218</td>\n",
       "      <td>5.200</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10</td>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>87.500</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>872</td>\n",
       "      <td>879</td>\n",
       "      <td>8.300</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>13</td>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>91.667</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>56</td>\n",
       "      <td>67</td>\n",
       "      <td>0.094</td>\n",
       "      <td>25.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>13</td>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>91.667</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>151</td>\n",
       "      <td>162</td>\n",
       "      <td>0.140</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>13</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>90.909</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>0.210</td>\n",
       "      <td>24.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>13</td>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>91.667</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>151</td>\n",
       "      <td>162</td>\n",
       "      <td>0.280</td>\n",
       "      <td>24.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>13</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>72.727</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>3.000</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>15</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>54</td>\n",
       "      <td>0.950</td>\n",
       "      <td>22.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>15</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>54</td>\n",
       "      <td>1.100</td>\n",
       "      <td>22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>15</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>54</td>\n",
       "      <td>1.400</td>\n",
       "      <td>21.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>15</td>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>53</td>\n",
       "      <td>62</td>\n",
       "      <td>5.400</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>15</td>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>53</td>\n",
       "      <td>62</td>\n",
       "      <td>6.600</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>20</td>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>166</td>\n",
       "      <td>177</td>\n",
       "      <td>4.100</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>20</td>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>61.538</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>148</td>\n",
       "      <td>160</td>\n",
       "      <td>5.600</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>21</td>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>56.250</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>22</td>\n",
       "      <td>598</td>\n",
       "      <td>613</td>\n",
       "      <td>2.400</td>\n",
       "      <td>22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>21</td>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>56.250</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>927</td>\n",
       "      <td>942</td>\n",
       "      <td>9.900</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>24</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>128</td>\n",
       "      <td>136</td>\n",
       "      <td>2.300</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>24</td>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>136</td>\n",
       "      <td>144</td>\n",
       "      <td>4.900</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>24</td>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>113</td>\n",
       "      <td>121</td>\n",
       "      <td>5.300</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>34</td>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>53.846</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>556</td>\n",
       "      <td>568</td>\n",
       "      <td>3.500</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>37</td>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>77.778</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>359</td>\n",
       "      <td>367</td>\n",
       "      <td>8.400</td>\n",
       "      <td>19.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    0          1       2   3   4   5   6   7    8    9      10    11\n",
       "3    2  g12562.t1  88.889   9   1   0   1   9   36   44  3.300  20.8\n",
       "4    2   g5616.t1  88.889   9   1   0   1   9   36   44  6.700  20.0\n",
       "5    2  g15153.t1  77.778   9   2   0   1   9   36   44  7.500  20.0\n",
       "8    4  g12562.t1  72.727  11   3   0   1  11  144  154  1.600  21.9\n",
       "9    4   g5616.t1  72.727  11   3   0   1  11  144  154  3.100  21.2\n",
       "10   4    g702.t1  63.636  11   4   0   1  11  145  155  3.700  21.2\n",
       "11   4  g15153.t1  63.636  11   4   0   1  11  144  154  5.400  20.4\n",
       "16   8   g7861.t1  62.500  16   6   0   1  16   64   79  0.650  23.5\n",
       "17   8   g4970.t1  57.143  14   6   0   5  18  134  147  8.800  20.4\n",
       "19   9  g11806.t1  66.667  15   5   0   1  15  152  166  3.900  21.2\n",
       "20   9  g16318.t1  66.667  15   5   0   1  15  224  238  4.900  21.2\n",
       "21   9  g16368.t1  66.667  15   5   0   1  15  204  218  5.200  20.8\n",
       "23  10  g11513.t1  87.500   8   1   0   2   9  872  879  8.300  20.0\n",
       "28  13   g5443.t1  91.667  12   1   0   1  12   56   67  0.094  25.8\n",
       "29  13   g5502.t1  91.667  12   1   0   1  12  151  162  0.140  25.0\n",
       "30  13  g15153.t1  90.909  11   1   0   2  12  144  154  0.210  24.6\n",
       "31  13   g5503.t1  91.667  12   1   0   1  12  151  162  0.280  24.3\n",
       "33  13   g5641.t1  72.727  11   3   0   2  12  144  154  3.000  21.6\n",
       "35  15  g12562.t1  80.000  10   2   0   1  10   45   54  0.950  22.7\n",
       "36  15   g5641.t1  80.000  10   2   0   1  10   45   54  1.100  22.3\n",
       "37  15   g5616.t1  80.000  10   2   0   1  10   45   54  1.400  21.9\n",
       "38  15   g5503.t1  80.000  10   2   0   1  10   53   62  5.400  20.4\n",
       "39  15   g5502.t1  80.000  10   2   0   1  10   53   62  6.600  20.0\n",
       "44  20  g11320.t1  66.667  12   4   0   4  15  166  177  4.100  21.6\n",
       "45  20   g8312.t1  61.538  13   5   0   1  13  148  160  5.600  21.2\n",
       "47  21   g5927.t1  56.250  16   7   0   7  22  598  613  2.400  22.3\n",
       "48  21   g8100.t1  56.250  16   7   0   1  16  927  942  9.900  20.8\n",
       "55  24   g5616.t1  88.889   9   1   0   1   9  128  136  2.300  21.6\n",
       "56  24  g12388.t1  88.889   9   1   0   1   9  136  144  4.900  20.4\n",
       "57  24   g5467.t1  88.889   9   1   0   1   9  113  121  5.300  20.4\n",
       "61  34  g11960.t1  53.846  13   6   0   4  16  556  568  3.500  21.6\n",
       "64  37   g2203.t1  77.778   9   2   0   1   9  359  367  8.400  19.6"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[2] < 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "6ba807c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('VLLDNQDDYELQ', 'VLLDNQDDYELK')"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g5443.t1\"][55:67], peptides[13]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "7c3efedf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('LLDNQDDYELQ', 'VLLDNQDDYELK')"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g15153.t1\"][143:154], peptides[13]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "13603ead",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"proteins/filtered_by_peptides_blast.fa\", 'w') as f_out:\n",
    "    for protein_id in sorted(list(set(df[1])), key=lambda x: int(x[1:-3])):\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "c9732114",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(df[1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe971ee9",
   "metadata": {},
   "source": [
    "## Localization prediction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a358a8f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g702.t1</td>\n",
       "      <td>extr: 29, plas: 2, lyso: 1</td>\n",
       "      <td>extr: 6, vacu: 5, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>extr: 25, plas: 5, mito: 1, lyso: 1</td>\n",
       "      <td>vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1</td>\n",
       "      <td>extr: 24, golg: 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>plas: 29, nucl: 2, golg: 1</td>\n",
       "      <td>cyto: 5, nucl: 4, E.R.: 3, vacu: 2</td>\n",
       "      <td>plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>mito: 18, cyto: 11, extr: 2, nucl: 1</td>\n",
       "      <td>chlo: 6, mito: 6, nucl: 1, cyto: 1</td>\n",
       "      <td>cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1</td>\n",
       "      <td>chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...</td>\n",
       "      <td>chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1</td>\n",
       "      <td>pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>plas: 32</td>\n",
       "      <td>nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...</td>\n",
       "      <td>plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>plas: 24, mito: 8</td>\n",
       "      <td>chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1</td>\n",
       "      <td>extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>extr: 28, nucl: 3, cyto: 1</td>\n",
       "      <td>chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1</td>\n",
       "      <td>extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>extr: 27, plas: 4, mito: 1</td>\n",
       "      <td>extr: 6, vacu: 5, chlo: 2, golg: 1</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>chlo: 9, extr: 5</td>\n",
       "      <td>extr: 27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>extr: 29, plas: 1, mito: 1, lyso: 1</td>\n",
       "      <td>chlo: 10, extr: 3, cyto: 1</td>\n",
       "      <td>extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>plas: 23, mito: 7, E.R.: 1, golg: 1</td>\n",
       "      <td>chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...</td>\n",
       "      <td>plas: 21, mito: 3, E.R.: 2, pero: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>extr: 31, mito: 1</td>\n",
       "      <td>extr: 12, vacu: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>extr: 8, vacu: 4, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1</td>\n",
       "      <td>extr: 11, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...</td>\n",
       "      <td>nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...</td>\n",
       "      <td>cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>extr: 25, plas: 4, mito: 2, lyso: 1</td>\n",
       "      <td>chlo: 10, vacu: 2, cyto: 1, mito: 1</td>\n",
       "      <td>extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>plas: 29, cyto: 3</td>\n",
       "      <td>plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1</td>\n",
       "      <td>mito: 13, plas: 8, cyto: 5, vacu: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>extr: 30, lyso: 2</td>\n",
       "      <td>extr: 9, vacu: 3, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...</td>\n",
       "      <td>nucl: 11, chlo: 1, cyto: 1, cysk: 1</td>\n",
       "      <td>extr: 24, nucl: 1, mito: 1, cyto: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>extr: 32</td>\n",
       "      <td>extr: 11, vacu: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "0     g702.t1                         extr: 29, plas: 2, lyso: 1   \n",
       "1    g1285.t1                extr: 25, plas: 5, mito: 1, lyso: 1   \n",
       "2    g2203.t1                         plas: 29, nucl: 2, golg: 1   \n",
       "3    g3428.t1               mito: 18, cyto: 11, extr: 2, nucl: 1   \n",
       "4    g3679.t1       extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1   \n",
       "5    g4106.t1  E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...   \n",
       "6    g4970.t1                                           plas: 32   \n",
       "7    g5237.t1                                  plas: 24, mito: 8   \n",
       "8    g5443.t1                         extr: 28, nucl: 3, cyto: 1   \n",
       "9    g5467.t1                         extr: 27, plas: 4, mito: 1   \n",
       "10   g5502.t1                                  extr: 31, lyso: 1   \n",
       "11   g5503.t1                extr: 29, plas: 1, mito: 1, lyso: 1   \n",
       "12   g5510.t1                plas: 23, mito: 7, E.R.: 1, golg: 1   \n",
       "13   g5616.t1                                  extr: 31, mito: 1   \n",
       "14   g5641.t1                                  extr: 31, lyso: 1   \n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "21  g11320.t1      plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1   \n",
       "22  g11513.t1  cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "25  g12388.t1                extr: 25, plas: 4, mito: 2, lyso: 1   \n",
       "26  g12510.t1                                  plas: 29, cyto: 3   \n",
       "27  g12562.t1                                  extr: 30, lyso: 2   \n",
       "28  g13530.t1  extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "30  g15153.t1                                           extr: 32   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                          wolf_plants  \\\n",
       "0                  extr: 6, vacu: 5, golg: 2, E.R.: 1   \n",
       "1         vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1   \n",
       "2                  cyto: 5, nucl: 4, E.R.: 3, vacu: 2   \n",
       "3                  chlo: 6, mito: 6, nucl: 1, cyto: 1   \n",
       "4         chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1   \n",
       "5         chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1   \n",
       "6   nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...   \n",
       "7         chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1   \n",
       "8         chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1   \n",
       "9                  extr: 6, vacu: 5, chlo: 2, golg: 1   \n",
       "10                                   chlo: 9, extr: 5   \n",
       "11                         chlo: 10, extr: 3, cyto: 1   \n",
       "12  chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...   \n",
       "13                                  extr: 12, vacu: 2   \n",
       "14                          extr: 8, vacu: 4, golg: 2   \n",
       "15                                           nucl: 14   \n",
       "16        nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17        nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18                 nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                                  nucl: 13, chlo: 1   \n",
       "20                          nucl: 8, cyto: 5, plas: 1   \n",
       "21                         extr: 11, golg: 2, E.R.: 1   \n",
       "22  nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...   \n",
       "23       nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                         nucl: 13.5, cyto_nucl: 7.5   \n",
       "25                chlo: 10, vacu: 2, cyto: 1, mito: 1   \n",
       "26        plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1   \n",
       "27                          extr: 9, vacu: 3, golg: 2   \n",
       "28                nucl: 11, chlo: 1, cyto: 1, cysk: 1   \n",
       "29                                           nucl: 14   \n",
       "30                         extr: 11, vacu: 2, E.R.: 1   \n",
       "31                nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                                  nucl: 13, chlo: 1   \n",
       "33                                  nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi  \n",
       "0                                            extr: 26  \n",
       "1                                   extr: 24, golg: 2  \n",
       "2   plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...  \n",
       "3   cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...  \n",
       "4                                            extr: 26  \n",
       "5   pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...  \n",
       "6   plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...  \n",
       "7   extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...  \n",
       "8   extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...  \n",
       "9                                            extr: 25  \n",
       "10                                           extr: 27  \n",
       "11       extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1  \n",
       "12                plas: 21, mito: 3, E.R.: 2, pero: 1  \n",
       "13                                           extr: 25  \n",
       "14                                           extr: 25  \n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5  \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1  \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...  \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...  \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2  \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1  \n",
       "21                                           extr: 27  \n",
       "22  cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...  \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3  \n",
       "24                          nucl: 25, cyto_nucl: 15.5  \n",
       "25       extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1  \n",
       "26                mito: 13, plas: 8, cyto: 5, vacu: 1  \n",
       "27                                           extr: 25  \n",
       "28                extr: 24, nucl: 1, mito: 1, cyto: 1  \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...  \n",
       "30                                           extr: 27  \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3  \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...  \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "\n",
    "for org_type in [\"animals\", \"plants\", \"fungi\"]:\n",
    "    dfs.append(pd.read_csv(f\"preds_wolf/versus_{org_type}.txt\", \n",
    "                           sep=\" details \", \n",
    "                           engine=\"python\", names=[\"protein_id\", f\"wolf_{org_type}\"]))\n",
    "\n",
    "df_wolf = pd.concat([dfs[0], dfs[1][\"wolf_plants\"], dfs[2][\"wolf_fungi\"]], axis=1)\n",
    "df_wolf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "ddead69e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g702.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id targetp_nonplants targetp_plants\n",
       "0     g702.t1                SP             SP\n",
       "1    g1285.t1                SP             SP\n",
       "2    g2203.t1             OTHER          OTHER\n",
       "3    g3428.t1             OTHER          OTHER\n",
       "4    g3679.t1                SP             SP\n",
       "5    g4106.t1             OTHER             SP\n",
       "6    g4970.t1             OTHER          OTHER\n",
       "7    g5237.t1             OTHER          OTHER\n",
       "8    g5443.t1             OTHER          OTHER\n",
       "9    g5467.t1                SP             SP\n",
       "10   g5502.t1                SP             SP\n",
       "11   g5503.t1                SP             SP\n",
       "12   g5510.t1             OTHER          OTHER\n",
       "13   g5616.t1                SP             SP\n",
       "14   g5641.t1                SP             SP\n",
       "15   g5927.t1             OTHER          OTHER\n",
       "16   g7861.t1             OTHER          OTHER\n",
       "17   g8100.t1             OTHER          OTHER\n",
       "18   g8312.t1             OTHER          OTHER\n",
       "19  g10513.t1             OTHER          OTHER\n",
       "20  g10514.t1             OTHER          OTHER\n",
       "21  g11320.t1                SP             SP\n",
       "22  g11513.t1             OTHER          OTHER\n",
       "23  g11806.t1             OTHER          OTHER\n",
       "24  g11960.t1             OTHER          OTHER\n",
       "25  g12388.t1                SP             SP\n",
       "26  g12510.t1             OTHER          OTHER\n",
       "27  g12562.t1                SP             SP\n",
       "28  g13530.t1                SP             SP\n",
       "29  g14472.t1             OTHER          OTHER\n",
       "30  g15153.t1                SP             SP\n",
       "31  g15484.t1             OTHER          OTHER\n",
       "32  g16318.t1             OTHER          OTHER\n",
       "33  g16368.t1             OTHER          OTHER"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "\n",
    "for org_type in [\"nonplants\", \"plants\"]:\n",
    "    dfs.append(pd.read_csv(f\"preds_targetp/versus_{org_type}.txt\", \n",
    "                           sep='\\t', \n",
    "                           header=1, \n",
    "                          )\n",
    "              )\n",
    "    dfs[-1].rename(columns={\"# ID\": \"protein_id\", \"Prediction\": f\"targetp_{org_type}\"}, inplace=True)\n",
    "\n",
    "df_targetp = pd.concat([dfs[0][[\"protein_id\", \"targetp_nonplants\"]], dfs[1][\"targetp_plants\"]], axis=1)\n",
    "df_targetp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "992369ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g702.t1</td>\n",
       "      <td>extr: 29, plas: 2, lyso: 1</td>\n",
       "      <td>extr: 6, vacu: 5, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>extr: 25, plas: 5, mito: 1, lyso: 1</td>\n",
       "      <td>vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1</td>\n",
       "      <td>extr: 24, golg: 2</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>plas: 29, nucl: 2, golg: 1</td>\n",
       "      <td>cyto: 5, nucl: 4, E.R.: 3, vacu: 2</td>\n",
       "      <td>plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>mito: 18, cyto: 11, extr: 2, nucl: 1</td>\n",
       "      <td>chlo: 6, mito: 6, nucl: 1, cyto: 1</td>\n",
       "      <td>cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1</td>\n",
       "      <td>chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...</td>\n",
       "      <td>chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1</td>\n",
       "      <td>pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>plas: 32</td>\n",
       "      <td>nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...</td>\n",
       "      <td>plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>plas: 24, mito: 8</td>\n",
       "      <td>chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1</td>\n",
       "      <td>extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>extr: 28, nucl: 3, cyto: 1</td>\n",
       "      <td>chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1</td>\n",
       "      <td>extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>extr: 27, plas: 4, mito: 1</td>\n",
       "      <td>extr: 6, vacu: 5, chlo: 2, golg: 1</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>chlo: 9, extr: 5</td>\n",
       "      <td>extr: 27</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>extr: 29, plas: 1, mito: 1, lyso: 1</td>\n",
       "      <td>chlo: 10, extr: 3, cyto: 1</td>\n",
       "      <td>extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>plas: 23, mito: 7, E.R.: 1, golg: 1</td>\n",
       "      <td>chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...</td>\n",
       "      <td>plas: 21, mito: 3, E.R.: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>extr: 31, mito: 1</td>\n",
       "      <td>extr: 12, vacu: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>extr: 8, vacu: 4, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1</td>\n",
       "      <td>extr: 11, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...</td>\n",
       "      <td>nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...</td>\n",
       "      <td>cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>extr: 25, plas: 4, mito: 2, lyso: 1</td>\n",
       "      <td>chlo: 10, vacu: 2, cyto: 1, mito: 1</td>\n",
       "      <td>extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>plas: 29, cyto: 3</td>\n",
       "      <td>plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1</td>\n",
       "      <td>mito: 13, plas: 8, cyto: 5, vacu: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>extr: 30, lyso: 2</td>\n",
       "      <td>extr: 9, vacu: 3, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...</td>\n",
       "      <td>nucl: 11, chlo: 1, cyto: 1, cysk: 1</td>\n",
       "      <td>extr: 24, nucl: 1, mito: 1, cyto: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>extr: 32</td>\n",
       "      <td>extr: 11, vacu: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "0     g702.t1                         extr: 29, plas: 2, lyso: 1   \n",
       "1    g1285.t1                extr: 25, plas: 5, mito: 1, lyso: 1   \n",
       "2    g2203.t1                         plas: 29, nucl: 2, golg: 1   \n",
       "3    g3428.t1               mito: 18, cyto: 11, extr: 2, nucl: 1   \n",
       "4    g3679.t1       extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1   \n",
       "5    g4106.t1  E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...   \n",
       "6    g4970.t1                                           plas: 32   \n",
       "7    g5237.t1                                  plas: 24, mito: 8   \n",
       "8    g5443.t1                         extr: 28, nucl: 3, cyto: 1   \n",
       "9    g5467.t1                         extr: 27, plas: 4, mito: 1   \n",
       "10   g5502.t1                                  extr: 31, lyso: 1   \n",
       "11   g5503.t1                extr: 29, plas: 1, mito: 1, lyso: 1   \n",
       "12   g5510.t1                plas: 23, mito: 7, E.R.: 1, golg: 1   \n",
       "13   g5616.t1                                  extr: 31, mito: 1   \n",
       "14   g5641.t1                                  extr: 31, lyso: 1   \n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "21  g11320.t1      plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1   \n",
       "22  g11513.t1  cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "25  g12388.t1                extr: 25, plas: 4, mito: 2, lyso: 1   \n",
       "26  g12510.t1                                  plas: 29, cyto: 3   \n",
       "27  g12562.t1                                  extr: 30, lyso: 2   \n",
       "28  g13530.t1  extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "30  g15153.t1                                           extr: 32   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                          wolf_plants  \\\n",
       "0                  extr: 6, vacu: 5, golg: 2, E.R.: 1   \n",
       "1         vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1   \n",
       "2                  cyto: 5, nucl: 4, E.R.: 3, vacu: 2   \n",
       "3                  chlo: 6, mito: 6, nucl: 1, cyto: 1   \n",
       "4         chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1   \n",
       "5         chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1   \n",
       "6   nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...   \n",
       "7         chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1   \n",
       "8         chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1   \n",
       "9                  extr: 6, vacu: 5, chlo: 2, golg: 1   \n",
       "10                                   chlo: 9, extr: 5   \n",
       "11                         chlo: 10, extr: 3, cyto: 1   \n",
       "12  chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...   \n",
       "13                                  extr: 12, vacu: 2   \n",
       "14                          extr: 8, vacu: 4, golg: 2   \n",
       "15                                           nucl: 14   \n",
       "16        nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17        nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18                 nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                                  nucl: 13, chlo: 1   \n",
       "20                          nucl: 8, cyto: 5, plas: 1   \n",
       "21                         extr: 11, golg: 2, E.R.: 1   \n",
       "22  nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...   \n",
       "23       nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                         nucl: 13.5, cyto_nucl: 7.5   \n",
       "25                chlo: 10, vacu: 2, cyto: 1, mito: 1   \n",
       "26        plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1   \n",
       "27                          extr: 9, vacu: 3, golg: 2   \n",
       "28                nucl: 11, chlo: 1, cyto: 1, cysk: 1   \n",
       "29                                           nucl: 14   \n",
       "30                         extr: 11, vacu: 2, E.R.: 1   \n",
       "31                nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                                  nucl: 13, chlo: 1   \n",
       "33                                  nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi targetp_nonplants  \\\n",
       "0                                            extr: 26                SP   \n",
       "1                                   extr: 24, golg: 2                SP   \n",
       "2   plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...             OTHER   \n",
       "3   cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...             OTHER   \n",
       "4                                            extr: 26                SP   \n",
       "5   pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...             OTHER   \n",
       "6   plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...             OTHER   \n",
       "7   extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...             OTHER   \n",
       "8   extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...             OTHER   \n",
       "9                                            extr: 25                SP   \n",
       "10                                           extr: 27                SP   \n",
       "11       extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1                SP   \n",
       "12                plas: 21, mito: 3, E.R.: 2, pero: 1             OTHER   \n",
       "13                                           extr: 25                SP   \n",
       "14                                           extr: 25                SP   \n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5             OTHER   \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1             OTHER   \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...             OTHER   \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...             OTHER   \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2             OTHER   \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1             OTHER   \n",
       "21                                           extr: 27                SP   \n",
       "22  cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...             OTHER   \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3             OTHER   \n",
       "24                          nucl: 25, cyto_nucl: 15.5             OTHER   \n",
       "25       extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1                SP   \n",
       "26                mito: 13, plas: 8, cyto: 5, vacu: 1             OTHER   \n",
       "27                                           extr: 25                SP   \n",
       "28                extr: 24, nucl: 1, mito: 1, cyto: 1                SP   \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...             OTHER   \n",
       "30                                           extr: 27                SP   \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3             OTHER   \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...             OTHER   \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...             OTHER   \n",
       "\n",
       "   targetp_plants  \n",
       "0              SP  \n",
       "1              SP  \n",
       "2           OTHER  \n",
       "3           OTHER  \n",
       "4              SP  \n",
       "5              SP  \n",
       "6           OTHER  \n",
       "7           OTHER  \n",
       "8           OTHER  \n",
       "9              SP  \n",
       "10             SP  \n",
       "11             SP  \n",
       "12          OTHER  \n",
       "13             SP  \n",
       "14             SP  \n",
       "15          OTHER  \n",
       "16          OTHER  \n",
       "17          OTHER  \n",
       "18          OTHER  \n",
       "19          OTHER  \n",
       "20          OTHER  \n",
       "21             SP  \n",
       "22          OTHER  \n",
       "23          OTHER  \n",
       "24          OTHER  \n",
       "25             SP  \n",
       "26          OTHER  \n",
       "27             SP  \n",
       "28             SP  \n",
       "29          OTHER  \n",
       "30             SP  \n",
       "31          OTHER  \n",
       "32          OTHER  \n",
       "33          OTHER  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates = pd.concat([df_wolf, df_targetp.drop(columns=[\"protein_id\"])], axis=1)\n",
    "candidates.to_csv(\"proteins/wolf_and_targetp_predictions.tsv\", sep='\\t', index=False)\n",
    "candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "38e1b779",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num candidates: 12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                     wolf_plants  \\\n",
       "15                                      nucl: 14   \n",
       "16   nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17   nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18            nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                             nucl: 13, chlo: 1   \n",
       "20                     nucl: 8, cyto: 5, plas: 1   \n",
       "23  nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                    nucl: 13.5, cyto_nucl: 7.5   \n",
       "29                                      nucl: 14   \n",
       "31           nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                             nucl: 13, chlo: 1   \n",
       "33                             nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi targetp_nonplants  \\\n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5             OTHER   \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1             OTHER   \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...             OTHER   \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...             OTHER   \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2             OTHER   \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1             OTHER   \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3             OTHER   \n",
       "24                          nucl: 25, cyto_nucl: 15.5             OTHER   \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...             OTHER   \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3             OTHER   \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...             OTHER   \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...             OTHER   \n",
       "\n",
       "   targetp_plants  \n",
       "15          OTHER  \n",
       "16          OTHER  \n",
       "17          OTHER  \n",
       "18          OTHER  \n",
       "19          OTHER  \n",
       "20          OTHER  \n",
       "23          OTHER  \n",
       "24          OTHER  \n",
       "29          OTHER  \n",
       "31          OTHER  \n",
       "32          OTHER  \n",
       "33          OTHER  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates = candidates[(candidates[\"wolf_animals\"].str.startswith(\"nucl\")) &\n",
    "                        (candidates[\"targetp_nonplants\"] == \"OTHER\")\n",
    "                       ]\n",
    "print(f\"Num candidates: {len(candidates)}\")\n",
    "candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ed0f99d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num candidates: 18\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>plas: 29, nucl: 2, golg: 1</td>\n",
       "      <td>cyto: 5, nucl: 4, E.R.: 3, vacu: 2</td>\n",
       "      <td>plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>mito: 18, cyto: 11, extr: 2, nucl: 1</td>\n",
       "      <td>chlo: 6, mito: 6, nucl: 1, cyto: 1</td>\n",
       "      <td>cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>plas: 32</td>\n",
       "      <td>nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...</td>\n",
       "      <td>plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>plas: 24, mito: 8</td>\n",
       "      <td>chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1</td>\n",
       "      <td>extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>extr: 28, nucl: 3, cyto: 1</td>\n",
       "      <td>chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1</td>\n",
       "      <td>extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...</td>\n",
       "      <td>nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...</td>\n",
       "      <td>cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "2    g2203.t1                         plas: 29, nucl: 2, golg: 1   \n",
       "3    g3428.t1               mito: 18, cyto: 11, extr: 2, nucl: 1   \n",
       "6    g4970.t1                                           plas: 32   \n",
       "7    g5237.t1                                  plas: 24, mito: 8   \n",
       "8    g5443.t1                         extr: 28, nucl: 3, cyto: 1   \n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "22  g11513.t1  cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                          wolf_plants  \\\n",
       "2                  cyto: 5, nucl: 4, E.R.: 3, vacu: 2   \n",
       "3                  chlo: 6, mito: 6, nucl: 1, cyto: 1   \n",
       "6   nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...   \n",
       "7         chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1   \n",
       "8         chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1   \n",
       "15                                           nucl: 14   \n",
       "16        nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17        nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18                 nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                                  nucl: 13, chlo: 1   \n",
       "20                          nucl: 8, cyto: 5, plas: 1   \n",
       "22  nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...   \n",
       "23       nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                         nucl: 13.5, cyto_nucl: 7.5   \n",
       "29                                           nucl: 14   \n",
       "31                nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                                  nucl: 13, chlo: 1   \n",
       "33                                  nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi targetp_nonplants  \\\n",
       "2   plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...             OTHER   \n",
       "3   cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...             OTHER   \n",
       "6   plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...             OTHER   \n",
       "7   extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...             OTHER   \n",
       "8   extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...             OTHER   \n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5             OTHER   \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1             OTHER   \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...             OTHER   \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...             OTHER   \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2             OTHER   \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1             OTHER   \n",
       "22  cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...             OTHER   \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3             OTHER   \n",
       "24                          nucl: 25, cyto_nucl: 15.5             OTHER   \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...             OTHER   \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3             OTHER   \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...             OTHER   \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...             OTHER   \n",
       "\n",
       "   targetp_plants  \n",
       "2           OTHER  \n",
       "3           OTHER  \n",
       "6           OTHER  \n",
       "7           OTHER  \n",
       "8           OTHER  \n",
       "15          OTHER  \n",
       "16          OTHER  \n",
       "17          OTHER  \n",
       "18          OTHER  \n",
       "19          OTHER  \n",
       "20          OTHER  \n",
       "22          OTHER  \n",
       "23          OTHER  \n",
       "24          OTHER  \n",
       "29          OTHER  \n",
       "31          OTHER  \n",
       "32          OTHER  \n",
       "33          OTHER  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates = candidates[((candidates[\"wolf_animals\"].str.find(\"nucl\") != -1) | \n",
    "                        (candidates[\"wolf_plants\"].str.find(\"nucl\") != -1) |\n",
    "                        (candidates[\"wolf_fungi\"].str.find(\"nucl\") != -1)) &\n",
    "                        (candidates[\"targetp_nonplants\"] == \"OTHER\") &\n",
    "                        (candidates[\"targetp_plants\"] == \"OTHER\")\n",
    "               ]\n",
    "print(f\"Num candidates: {len(candidates)}\")\n",
    "candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "64490096",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"proteins/filtered_by_localization_short.fa\", 'w') as f_out:\n",
    "    for protein_id in candidates[\"protein_id\"]:\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78a50040",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e608feb4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6ab2f08",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
