{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e8fb75c1",
   "metadata": {},
   "source": [
    "# Project 4. \"Tardigrades: from genestealers to space marines\". Eukaryotic genome analysis.\n",
    "\n",
    "Lab notebook by Nikita Vyatkin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "984c1d9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def show_file_content(path, lines=10, headers_only=False):\n",
    "    if headers_only:\n",
    "        ! mkdir -p temp\n",
    "        ! touch temp/show_file_content.tmp\n",
    "        ! grep '>' {path} > temp/show_file_content.tmp\n",
    "        path = \"temp/show_file_content.tmp\"\n",
    "    \n",
    "    ! head -{lines} {path}\n",
    "    print(\"...\")\n",
    "    ! tail -{lines} {path}\n",
    "    \n",
    "    if headers_only:\n",
    "        ! rm {path}\n",
    "\n",
    "\n",
    "def count_seqs(path):\n",
    "    res = ! grep '>' {path} | wc -l\n",
    "    print(f\"Total: {res[0]} sequencies\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "668b748b",
   "metadata": {},
   "source": [
    "## Data source\n",
    "\n",
    "For this project we will be using a sequence of the _Ramazzottius varieornatus_, the YOKOZUNA-1 strain (sequenced in the University of Tokyo and named after the highest rank in professional sumo).\n",
    "\n",
    "One can download assembled genome [here](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b0b32a6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-12-05 18:12:32--  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 165.112.9.230, 2607:f220:41f:250::229, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 17621408 (17M) [application/x-gzip]\n",
      "Saving to: ‘data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz’\n",
      "\n",
      "GCA_001949185.1_Rva 100%[===================>]  16,80M   878KB/s    in 26s     \n",
      "\n",
      "2022-12-05 18:13:00 (674 KB/s) - ‘data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz’ saved [17621408/17621408]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! wget -P data/ https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
    "! gunzip data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
    "! mv data/GCA_001949185.1_Rvar_4.0_genomic.fna data/Ramazzottius_varieornatus.fna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a7a139c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 54M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n"
     ]
    }
   ],
   "source": [
    "! ls -lh data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4190903f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BDGG01000001.1 Ramazzottius varieornatus DNA, contig: Scaffold001, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      "TGCTGCGACATACTCGAAACACCCAAGGTCTTACCGGGAGAATATCGTCGTCTGGAGGATTTGTATGGGCCATTGTGATG\n",
      "GCCGCTACCTGGCATTGTACAACTTGAATGGTTGGTCTGAGCGGGACATGATCGACAGGTGGGCTTTGATGGTGACACTT\n",
      "TTGGTGCCGCCGGCATTCCTTCTGCCCGCACGTGACATCGACAACCTCTTGACTGCCGGCCATGAAACCTCACCCCAAAC\n",
      "AGGTATTCAGTGGGCAGAAAAAGCTACAAAAATCCTATCGCACGGGAATAAGTCCGTGCTAAACTAGTTAGAGCGACTTT\n",
      "TGAAAGGAAGATGTAGAGTTTAGTTCATTGGAGTAATCAAGTCTTCGGTTAgtttcgtgacgtcacgattGATCCGTATG\n",
      "GTGAAGCGTCTGAGCACAGAGTTTTCTTATTTTTCAACAGGTGGGAGCGACGTCATGAGCGAGTATGTGACGCCCTTCTG\n",
      "TTACGTCTGTGATTCTCTCGTGGATGCCGACTGCCGAGAATCGCAAGACTGGATGCCTCGACGAAGAAACGGCttaagga\n",
      "agaaagtgatgTGGACAGGCCAAGGGGAGGACGATCATGGCAGACTTGCCTTCGACTGGAAAGAAGTCTACATtcggctg\n",
      "gaaaagttcaaGCAGCGATGTCCGTCCAACAATGGCTGCACCACTCACGTCTTCCTCAATAGCGGTACGTATCTTGCTGT\n",
      "...\n",
      "TCTTCTTTTAAGACAAGCCGGCTTGTCTTATTTAATATTAATCTTTATTTTTCTTTCTTCTTTTATAGCTATATTCATTA\n",
      "ATATAAAACAAAATTTATTTAAAAGAATGTTGACAGTTTCAAGATTGTCCAACACCAGATGACTCTTGACCGCAGTTGTA\n",
      "ACAACATATTCTGGATGAATATTTTATTTTGCAGTGTACTCATTTTCTTTAGTATTAAGTTCAACAATAAATAATAAGTA\n",
      "TTTAGACCAAAATTACCGTAATAGAGACACGCTAGAAAACCCCCTTTTATTTTTAAGCATAGGAGGACTCCCCCCCATAC\n",
      "CAGGATTTTTCCCTAAAATTTTAATTCTGATAAATTTAAGATATTCTCGGTTAAGATTGCTAAGAATTTTTCTCCTTCTT\n",
      "GCCGCGTGTGTAGATATTTTTATTTATTTACGATTTTCCTTCACAAGAGCATTGAAAAATAATGGTAATCTTATATATTG\n",
      "TAAAGCTGTAACAACAAAGAGAATTAAAATACTGTCTTTAAGAGCAGTCACTATAGTCACAATAATAGCAATTATATTAT\n",
      "AAAAAGATTTTAAGTTAAAAAAACTAATGTGTTTCAAACACTTCAATAATGAAAATTTAAATCTTAATTTTAAAGAAAAA\n",
      "CTTTTTAAATATGCTTGCAACATATCATATAAATAAAATTTCATAACAAAAAGAAAAAATCTATATGTAAATTTACAATT\n",
      "TACCGCCTATACTCAGCCATTTTGACTA\n",
      "Total: 200 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/Ramazzottius_varieornatus.fna\")\n",
    "count_seqs(\"data/Ramazzottius_varieornatus.fna\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8ab4773f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BDGG01000001.1 Ramazzottius varieornatus DNA, contig: Scaffold001, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000002.1 Ramazzottius varieornatus DNA, contig: Scaffold002, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000003.1 Ramazzottius varieornatus DNA, contig: Scaffold003, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      "...\n",
      ">BDGG01000198.1 Ramazzottius varieornatus DNA, contig: Scaffold277, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000199.1 Ramazzottius varieornatus DNA, contig: Scaffold279, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">AP017609.1 Ramazzottius varieornatus mitochondrial DNA, complete genome, strain: YOKOZUNA-1\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/Ramazzottius_varieornatus.fna\", lines=3, headers_only=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b577d86",
   "metadata": {},
   "source": [
    "## Structural annotation\n",
    "\n",
    "One can download AUGUSTUS results [here](https://drive.google.com/file/d/1wBxf6cDgu22NbjAOgTe-8b3Zx60hNKY0/view?usp=drive_web).\n",
    "\n",
    "[Script](http://augustus.gobics.de/binaries/scripts/getAnnoFasta.pl) for GFF to FASTA converting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8f182b8e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".:\r\n",
      "total 52M\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  5 18:41 \u001b[0m\u001b[01;34mannotation\u001b[0m/\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 16:25 \u001b[01;34mdata\u001b[0m/\r\n",
      "-rwxr-xr-x 1 nikitos nikitos  38M апр 21  2022 \u001b[01;32mdiamond\u001b[0m*\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  13M апр 21  2022 \u001b[01;31mdiamond-linux64.tar.gz\u001b[0m\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  8 16:34 \u001b[01;34mdiamond_search\u001b[0m/\r\n",
      "-rwxrwxr-x 1 nikitos nikitos 6,6K дек  5 18:27 \u001b[01;32mgetAnnoFasta.pl\u001b[0m*\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  51K дек  8 17:50 Lab_Notebook.ipynb\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 6,7K дек  8 17:31 matched_proteins.fa\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 743K дек  8 17:46 \u001b[01;31moutput_all_results.zip\u001b[0m\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек 10 17:38 \u001b[01;34mtemp\u001b[0m/\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 3,3K дек  8 17:47 wolfpsort_res.html\r\n",
      "\r\n",
      "./annotation:\r\n",
      "total 7,0M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 7,0M дек  5 18:41 augustus.whole.aa\r\n",
      "\r\n",
      "./data:\r\n",
      "total 81M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 27M дек  5 18:21 augustus.whole.gff\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 737 дек  7 16:22 dna_associated_peptides.fa\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n",
      "\r\n",
      "./diamond_search:\r\n",
      "total 7,2M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  680 дек  8 16:34 matches.tsv\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 7,2M дек  8 16:26 reference.dmnd\r\n",
      "\r\n",
      "./temp:\r\n",
      "total 0\r\n"
     ]
    }
   ],
   "source": [
    "ls -lhR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "a0a0ebbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "! chmod +x getAnnoFasta.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fda5f0da",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "97cda1bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "! ./getAnnoFasta.pl data/augustus.whole.gff\n",
    "! mv data/augustus.whole.aa proteins/all_proteins.faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "497d3541",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">g1.t1\n",
      "AATYSKHPRSYRENIVVWRICMGHCDGRYLALYNLNGWSERDMIDRWALMVTLLVPPAFLLPARDIDNLLTAGHETSPQTGGSDVMSEYVTPFCYVCDSL\n",
      "VDADCRESQDWMPRRRNGLRKKVMWTGQGEDDHGRLAFDWKEVYIRLEKFKQRCPSNNGCTTHVFLNSDRVIRRCFDPKEPPCSGHEWMVNRRNPKKRSK\n",
      "VKVVSRLTCADWPFCNVDSAYYWPNGTVEALLSPHDDPQLKVRSAKLSCYECDSILQPDCTGQNFVKAVRKPNDAVAHSVDWVGLRKGMEKFRTRCPVES\n",
      "GCFFQGDHMTNRVIRGCRNMTTDLPRHLLVKDRRKESSVFRTEEVCRESECNSRVLFFSHSGAFLDSDWHFNNSSAGRDNGQGEAKKEVYQGDAPATEET\n",
      "GFLPHVKQSGQGKRKEGRLQKRTIGDVEREERRADEARQRRGKPKAEGRGGAGTRVLSFGHRSRKTSAGGKLDSCWLLLLLPFFLL\n",
      ">g2.t1\n",
      "MTFAVTVATTSQSCAIAFDVFWLTTIKLFFIAKYFGSEFTPFDKVQSLTCPHCAQQGIEYKDLLPHLETNHADTSDFLVVSDNITIIWDVAHCI\n",
      ">g3.t1\n",
      "MLDKAKLVLLNKGGRPSIRSAEDGWKVFSDSLLCRASSDGRVFIGKGGGNGGAVAKAEVAAIGVVLVHHMLQQVVRHGEVVIVSANTNWTPLPY\n",
      "...\n",
      "FPHCLGYVDGSLIGLQEAPSMCGEDYYTRKGSYAINAMIVCDDLKRIRHVLAGFPGCSHDHRVFMYSKLALNPFKYFSGHEYLLADSAYPVSDIIVPTFK\n",
      "KPLANSH\n",
      ">g16433.t1\n",
      "MHLDADIDESSKEEKPEIVMEYNRTKAGVDTLDQLTGNYSCRRKTSRWPMALFYDILDISTLDAYIIWCEINPGWNSTLPTKRRMFLQDVSKKMMQRQLL\n",
      "RRSTTPSISFSMKKFVSTVVPRVQVSSDQAVDEMANTSKPKALQKKPTR\n",
      ">g16434.t1\n",
      "RIHLLADCHLLKTLNLLQRDREVFPKVDLKSGVIWKVAFFSQALAAMERSVSQKNFGSPVLV\n",
      ">g16435.t1\n",
      "MKRPGGQKYHPWSDETKLKSGDVGGANHPWADEAKLNQHPKCAVLSSKLSNVTSMMKFLATGPTGECMSESRYTAFLAGQKNSLGVATKVAGGITSPSQL\n",
      "CLCKPLLYEEFLTFSNSMVMCETGLDDTCLTRKFK\n",
      "Total: 16435 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"proteins/all_proteins.faa\")\n",
    "count_seqs(\"proteins/all_proteins.faa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3d4968d",
   "metadata": {},
   "source": [
    "## Physical localization\n",
    "\n",
    "Considering that DNA is a major target of radiation damage, we may hypothesize that tardigrades might have unique proteins associated with their DNA to protect and/or effectively repair it. To explore this possibility, we will combine genomic and proteomic data.\n",
    "\n",
    "[List](https://disk.yandex.ru/d/xJqQMGX77Xueqg) of peptides that were associated with the Tardigrade's DNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8a8964f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".:\r\n",
      "total 48K\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  5 18:41 \u001b[0m\u001b[01;34mannotation\u001b[0m/\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 16:25 \u001b[01;34mdata\u001b[0m/\r\n",
      "-rwxrwxr-x 1 nikitos nikitos 6,6K дек  5 18:27 \u001b[01;32mgetAnnoFasta.pl\u001b[0m*\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  21K дек  8 15:39 Lab_Notebook.ipynb\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 17:10 \u001b[01;34mtemp\u001b[0m/\r\n",
      "-rw-rw-r-- 1 nikitos nikitos   76 дек  7 17:23 temp.txt\r\n",
      "\r\n",
      "./annotation:\r\n",
      "total 7,0M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 7,0M дек  5 18:41 augustus.whole.aa\r\n",
      "\r\n",
      "./data:\r\n",
      "total 81M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 27M дек  5 18:21 augustus.whole.gff\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 737 дек  7 16:22 dna_associated_peptides.fa\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n",
      "\r\n",
      "./temp:\r\n",
      "total 0\r\n"
     ]
    }
   ],
   "source": [
    "ls -lhR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "bd887bff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">1\n",
      "VFSALDVLR\n",
      ">2\n",
      "SPLQTDEIR\n",
      ">3\n",
      "AATGAVLSSADSR\n",
      "...\n",
      ">41\n",
      "GPEDEAGGPPK\n",
      ">42\n",
      "SSGGNSPDVVVR\n",
      ">43\n",
      "STVGQTPPQNLQR\n",
      "Total: 43 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/dna_associated_peptides.fa\", lines=6)\n",
    "count_seqs(\"data/dna_associated_peptides.fa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "d0ae5e22",
   "metadata": {},
   "outputs": [],
   "source": [
    "peptides = []\n",
    "\n",
    "with open(\"data/dna_associated_peptides.fa\", 'r') as f_in:\n",
    "    for line in f_in:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            peptides.append([line[1:], []])\n",
    "        elif line:\n",
    "            peptides[-1][1].append(line)\n",
    "\n",
    "    for peptide in peptides:\n",
    "        peptide[0] = int(peptide[0])\n",
    "        peptide[1] = ''.join(peptide[1])\n",
    "\n",
    "peptides = dict(peptides)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ef5d9ae2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: 'VFSALDVLR',\n",
       " 2: 'SPLQTDEIR',\n",
       " 3: 'AATGAVLSSADSR',\n",
       " 4: 'LLDDENDYELK',\n",
       " 5: 'VTGSSQGAINQQQAK',\n",
       " 6: 'VTGSSAQQIDINQAR',\n",
       " 7: 'LTSSGTGAGSAPAAAK',\n",
       " 8: 'NIPVGGVNTEATGDNYIR',\n",
       " 9: 'QGGMGMSGGMGGADR',\n",
       " 10: 'GPEDEAGGPPK',\n",
       " 11: 'LTDDELNQAMK',\n",
       " 12: 'AATGAVQSSASK',\n",
       " 13: 'VLLDNQDDYELK',\n",
       " 14: 'VFSALDVLR',\n",
       " 15: 'EIYANAQPGK',\n",
       " 16: 'QIMDNEGLR',\n",
       " 17: 'SAPLSASPISAR',\n",
       " 18: 'SSSSSSQESQSSSSQVR',\n",
       " 19: 'STVGQTPPQNLQR',\n",
       " 20: 'APVNGIVTDANGNQIQVR',\n",
       " 21: 'TNAAGQLIGPGGIAINDAGLTR',\n",
       " 22: 'VFSALDVLR',\n",
       " 23: 'SPLQTDEIR',\n",
       " 24: 'QFQDYSNSR',\n",
       " 25: 'AATGAVLSSADSR',\n",
       " 26: 'VFSALDVLR',\n",
       " 27: 'EIYANAQPGK',\n",
       " 28: 'QFQDYSNSR',\n",
       " 29: 'NIPVGGVNTEATGDNYIR',\n",
       " 30: 'AATGAVQSSASK',\n",
       " 31: 'VLLDNQDDYELK',\n",
       " 32: 'SGEQAMVDAK',\n",
       " 33: 'SSTTTSTMNAAK',\n",
       " 34: 'TTTTTGTTGTSGTTMPPK',\n",
       " 35: 'VVEQTGASSSR',\n",
       " 36: 'SSSSSSQESQSSSSQVR',\n",
       " 37: 'NTSFNTTGDR',\n",
       " 38: 'GYSYTNTTTYR',\n",
       " 39: 'AVTGAVAGVK',\n",
       " 40: 'QFQDYSNSR',\n",
       " 41: 'GPEDEAGGPPK',\n",
       " 42: 'SSGGNSPDVVVR',\n",
       " 43: 'STVGQTPPQNLQR'}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2e1f1585",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(peptides.values()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "3cdd9dbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_vals = set(peptides.values())\n",
    "new_peptides = dict()\n",
    "\n",
    "for key, val in peptides.items():\n",
    "    if val in unique_vals:\n",
    "        new_peptides[key] = val\n",
    "        unique_vals -= {val}\n",
    "\n",
    "peptides = new_peptides\n",
    "\n",
    "with open(\"data/dna_associated_peptides.fa\", 'w') as f_out:\n",
    "    for key, val in peptides.items():\n",
    "        f_out.write(f\">{key}\\n\")\n",
    "        f_out.write(val)\n",
    "        f_out.write('\\n')\n",
    "    \n",
    "len(peptides)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1effd975",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: 'VFSALDVLR',\n",
       " 2: 'SPLQTDEIR',\n",
       " 3: 'AATGAVLSSADSR',\n",
       " 4: 'LLDDENDYELK',\n",
       " 5: 'VTGSSQGAINQQQAK',\n",
       " 6: 'VTGSSAQQIDINQAR',\n",
       " 7: 'LTSSGTGAGSAPAAAK',\n",
       " 8: 'NIPVGGVNTEATGDNYIR',\n",
       " 9: 'QGGMGMSGGMGGADR',\n",
       " 10: 'GPEDEAGGPPK',\n",
       " 11: 'LTDDELNQAMK',\n",
       " 12: 'AATGAVQSSASK',\n",
       " 13: 'VLLDNQDDYELK',\n",
       " 15: 'EIYANAQPGK',\n",
       " 16: 'QIMDNEGLR',\n",
       " 17: 'SAPLSASPISAR',\n",
       " 18: 'SSSSSSQESQSSSSQVR',\n",
       " 19: 'STVGQTPPQNLQR',\n",
       " 20: 'APVNGIVTDANGNQIQVR',\n",
       " 21: 'TNAAGQLIGPGGIAINDAGLTR',\n",
       " 24: 'QFQDYSNSR',\n",
       " 32: 'SGEQAMVDAK',\n",
       " 33: 'SSTTTSTMNAAK',\n",
       " 34: 'TTTTTGTTGTSGTTMPPK',\n",
       " 35: 'VVEQTGASSSR',\n",
       " 37: 'NTSFNTTGDR',\n",
       " 38: 'GYSYTNTTTYR',\n",
       " 39: 'AVTGAVAGVK',\n",
       " 42: 'SSGGNSPDVVVR'}"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "9a3434e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MQRLGYPQREQYPHNQRQSILEKLIRAVDQALKNPDRKQRVEIESWLKRNLPKDPKLQLKLASDARGEVSLEDLLMTKEDLEAASKALLNESSKSGLHLPGLDESVQNIWPNDARGVLSLTELLVEAPAEPLRIFPSTYVADLIVTLVASLLEKEEVTLQKVLEMFPQLRNTFKPVETYLRKXXXXXXXXXXXXXXXXXXXXDEQRKADEASGKVSPANARSPSKSPVKLELQHKPSARSAKEIKSPH'"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins = []\n",
    "\n",
    "with open(\"proteins/all_proteins.faa\", 'r') as f_in:\n",
    "    for line in f_in:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            proteins.append([line[1:], []])\n",
    "        elif line:\n",
    "            proteins[-1][1].append(line)\n",
    "\n",
    "    for protein in proteins:\n",
    "        protein[1] = ''.join(protein[1])\n",
    "\n",
    "proteins = dict(proteins)\n",
    "proteins[\"g25.t1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "599747c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-12-08 16:12:47--  http://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
      "Resolving github.com (github.com)... 140.82.121.4\n",
      "Connecting to github.com (github.com)|140.82.121.4|:80... connected.\n",
      "HTTP request sent, awaiting response... 301 Moved Permanently\n",
      "Location: https://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz [following]\n",
      "--2022-12-08 16:12:47--  https://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
      "Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/31987083/4420191b-492f-4ec0-a80c-080436691f1e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221208T131247Z&X-Amz-Expires=300&X-Amz-Signature=b556836ddecfca6fd02dbf98564ee0598fd7f709f66560d9597b6863008f8adb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=31987083&response-content-disposition=attachment%3B%20filename%3Ddiamond-linux64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
      "--2022-12-08 16:12:47--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/31987083/4420191b-492f-4ec0-a80c-080436691f1e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221208T131247Z&X-Amz-Expires=300&X-Amz-Signature=b556836ddecfca6fd02dbf98564ee0598fd7f709f66560d9597b6863008f8adb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=31987083&response-content-disposition=attachment%3B%20filename%3Ddiamond-linux64.tar.gz&response-content-type=application%2Foctet-stream\n",
      "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.109.133, 185.199.108.133, ...\n",
      "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 13078237 (12M) [application/octet-stream]\n",
      "Saving to: ‘diamond-linux64.tar.gz’\n",
      "\n",
      "diamond-linux64.tar 100%[===================>]  12,47M  4,67MB/s    in 2,7s    \n",
      "\n",
      "2022-12-08 16:12:50 (4,67 MB/s) - ‘diamond-linux64.tar.gz’ saved [13078237/13078237]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# downloading the tool\n",
    "! wget http://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
    "! tar xzf diamond-linux64.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3911eaba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\r\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\r\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\r\n",
      "\r\n",
      "Syntax: diamond COMMAND [OPTIONS]\r\n",
      "\r\n",
      "Commands:\r\n",
      "makedb\tBuild DIAMOND database from a FASTA file\r\n",
      "prepdb\tPrepare BLAST database for use with Diamond\r\n",
      "blastp\tAlign amino acid query sequences against a protein reference database\r\n",
      "blastx\tAlign DNA query sequences against a protein reference database\r\n",
      "view\tView DIAMOND alignment archive (DAA) formatted file\r\n",
      "help\tProduce help message\r\n",
      "version\tDisplay version information\r\n",
      "getseq\tRetrieve sequences from a DIAMOND database file\r\n",
      "dbinfo\tPrint information about a DIAMOND database file\r\n",
      "test\tRun regression tests\r\n",
      "makeidx\tMake database index\r\n",
      "\r\n",
      "General options:\r\n",
      "--threads (-p)           number of CPU threads\r\n",
      "--db (-d)                database file\r\n",
      "--out (-o)               output file\r\n",
      "--outfmt (-f)            output format\r\n",
      "\t0   = BLAST pairwise\r\n",
      "\t5   = BLAST XML\r\n",
      "\t6   = BLAST tabular\r\n",
      "\t100 = DIAMOND alignment archive (DAA)\r\n",
      "\t101 = SAM\r\n",
      "\r\n",
      "\tValue 6 may be followed by a space-separated list of these keywords:\r\n",
      "\r\n",
      "\tqseqid means Query Seq - id\r\n",
      "\tqlen means Query sequence length\r\n",
      "\tsseqid means Subject Seq - id\r\n",
      "\tsallseqid means All subject Seq - id(s), separated by a ';'\r\n",
      "\tslen means Subject sequence length\r\n",
      "\tqstart means Start of alignment in query\r\n",
      "\tqend means End of alignment in query\r\n",
      "\tsstart means Start of alignment in subject\r\n",
      "\tsend means End of alignment in subject\r\n",
      "\tqseq means Aligned part of query sequence\r\n",
      "\tqseq_translated means Aligned part of query sequence (translated)\r\n",
      "\tfull_qseq means Query sequence\r\n",
      "\tfull_qseq_mate means Query sequence of the mate\r\n",
      "\tsseq means Aligned part of subject sequence\r\n",
      "\tfull_sseq means Subject sequence\r\n",
      "\tevalue means Expect value\r\n",
      "\tbitscore means Bit score\r\n",
      "\tscore means Raw score\r\n",
      "\tlength means Alignment length\r\n",
      "\tpident means Percentage of identical matches\r\n",
      "\tnident means Number of identical matches\r\n",
      "\tmismatch means Number of mismatches\r\n",
      "\tpositive means Number of positive - scoring matches\r\n",
      "\tgapopen means Number of gap openings\r\n",
      "\tgaps means Total number of gaps\r\n",
      "\tppos means Percentage of positive - scoring matches\r\n",
      "\tqframe means Query frame\r\n",
      "\tbtop means Blast traceback operations(BTOP)\r\n",
      "\tcigar means CIGAR string\r\n",
      "\tstaxids means unique Subject Taxonomy ID(s), separated by a ';' (in numerical order)\r\n",
      "\tsscinames means unique Subject Scientific Name(s), separated by a ';'\r\n",
      "\tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\r\n",
      "\tskingdoms means unique Subject Kingdom(s), separated by a ';'\r\n",
      "\tsphylums means unique Subject Phylum(s), separated by a ';'\r\n",
      "\tstitle means Subject Title\r\n",
      "\tsalltitles means All Subject Title(s), separated by a '<>'\r\n",
      "\tqcovhsp means Query Coverage Per HSP\r\n",
      "\tscovhsp means Subject Coverage Per HSP\r\n",
      "\tqtitle means Query title\r\n",
      "\tqqual means Query quality values for the aligned part of the query\r\n",
      "\tfull_qqual means Query quality values\r\n",
      "\tqstrand means Query strand\r\n",
      "\r\n",
      "\tDefault: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore\r\n",
      "--verbose (-v)           verbose console output\r\n",
      "--log                    enable debug log\r\n",
      "--quiet                  disable console output\r\n",
      "--header                 Write header lines to blast tabular format.\r\n",
      "\r\n",
      "Makedb options:\r\n",
      "--in                     input reference file in FASTA format\r\n",
      "--taxonmap               protein accession to taxid mapping file\r\n",
      "--taxonnodes             taxonomy nodes.dmp from NCBI\r\n",
      "--taxonnames             taxonomy names.dmp from NCBI\r\n",
      "\r\n",
      "Aligner options:\r\n",
      "--query (-q)             input query file\r\n",
      "--strand                 query strands to search (both/minus/plus)\r\n",
      "--un                     file for unaligned queries\r\n",
      "--al                     file or aligned queries\r\n",
      "--unfmt                  format of unaligned query file (fasta/fastq)\r\n",
      "--alfmt                  format of aligned query file (fasta/fastq)\r\n",
      "--unal                   report unaligned queries (0=no, 1=yes)\r\n",
      "--max-target-seqs (-k)   maximum number of target sequences to report alignments for (default=25)\r\n",
      "--top                    report alignments within this percentage range of top alignment score (overrides --max-target-seqs)\r\n",
      "--max-hsps               maximum number of HSPs per target sequence to report for each query (default=1)\r\n",
      "--range-culling          restrict hit culling to overlapping query ranges\r\n",
      "--compress               compression for output files (0=none, 1=gzip, zstd)\r\n",
      "--evalue (-e)            maximum e-value to report alignments (default=0.001)\r\n",
      "--min-score              minimum bit score to report alignments (overrides e-value setting)\r\n",
      "--id                     minimum identity% to report an alignment\r\n",
      "--query-cover            minimum query cover% to report an alignment\r\n",
      "--subject-cover          minimum subject cover% to report an alignment\r\n",
      "--fast                   enable fast mode\r\n",
      "--mid-sensitive          enable mid-sensitive mode\r\n",
      "--sensitive              enable sensitive mode)\r\n",
      "--more-sensitive         enable more sensitive mode\r\n",
      "--very-sensitive         enable very sensitive mode\r\n",
      "--ultra-sensitive        enable ultra sensitive mode\r\n",
      "--iterate                iterated search with increasing sensitivity\r\n",
      "--global-ranking (-g)    number of targets for global ranking\r\n",
      "--block-size (-b)        sequence block size in billions of letters (default=2.0)\r\n",
      "--index-chunks (-c)      number of chunks for index processing (default=4)\r\n",
      "--tmpdir (-t)            directory for temporary files\r\n",
      "--parallel-tmpdir        directory for temporary files used by multiprocessing\r\n",
      "--gapopen                gap open penalty\r\n",
      "--gapextend              gap extension penalty\r\n",
      "--frameshift (-F)        frame shift penalty (default=disabled)\r\n",
      "--long-reads             short for --range-culling --top 10 -F 15\r\n",
      "--matrix                 score matrix for protein alignment (default=BLOSUM62)\r\n",
      "--custom-matrix          file containing custom scoring matrix\r\n",
      "--comp-based-stats       composition based statistics mode (0-4)\r\n",
      "--masking                masking algorithm (none, seg, tantan=default)\r\n",
      "--query-gencode          genetic code to use to translate query (see user manual)\r\n",
      "--salltitles             include full subject titles in DAA file\r\n",
      "--sallseqid              include all subject ids in DAA file\r\n",
      "--no-self-hits           suppress reporting of identical self hits\r\n",
      "--taxonlist              restrict search to list of taxon ids (comma-separated)\r\n",
      "--taxon-exclude          exclude list of taxon ids (comma-separated)\r\n",
      "--seqidlist              filter the database by list of accessions\r\n",
      "--skip-missing-seqids    ignore accessions missing in the database\r\n",
      "\r\n",
      "Advanced options:\r\n",
      "--algo                   Seed search algorithm (0=double-indexed/1=query-indexed/ctg=contiguous-seed)\r\n",
      "--bin                    number of query bins for seed search\r\n",
      "--min-orf (-l)           ignore translated sequences without an open reading frame of at least this length\r\n",
      "--seed-cut               cutoff for seed complexity\r\n",
      "--freq-masking           mask seeds based on frequency\r\n",
      "--freq-sd                number of standard deviations for ignoring frequent seeds\r\n",
      "--motif-masking          softmask abundant motifs (0/1)\r\n",
      "--id2                    minimum number of identities for stage 1 hit\r\n",
      "--xdrop (-x)             xdrop for ungapped alignment\r\n",
      "--gapped-filter-evalue   E-value threshold for gapped filter (auto)\r\n",
      "--band                   band for dynamic programming computation\r\n",
      "--shapes (-s)            number of seed shapes (default=all available)\r\n",
      "--shape-mask             seed shapes\r\n",
      "--multiprocessing        enable distributed-memory parallel processing\r\n",
      "--mp-init                initialize multiprocessing run\r\n",
      "--mp-recover             enable continuation of interrupted multiprocessing run\r\n",
      "--mp-query-chunk         process only a single query chunk as specified\r\n",
      "--ext-chunk-size         chunk size for adaptive ranking (default=auto)\r\n",
      "--no-ranking             disable ranking heuristic\r\n",
      "--ext                    Extension mode (banded-fast/banded-slow/full)\r\n",
      "--culling-overlap        minimum range overlap with higher scoring hit to delete a hit (default=50%)\r\n",
      "--taxon-k                maximum number of targets to report per species\r\n",
      "--range-cover            percentage of query range to be covered for range culling (default=50%)\r\n",
      "--dbsize                 effective database size (in letters)\r\n",
      "--no-auto-append         disable auto appending of DAA and DMND file extensions\r\n",
      "--xml-blord-format       Use gnl|BL_ORD_ID| style format in XML output\r\n",
      "--stop-match-score       Set the match score of stop codons against each other.\r\n",
      "--tantan-minMaskProb     minimum repeat probability for masking (default=0.9)\r\n",
      "--file-buffer-size       file buffer size in bytes (default=67108864)\r\n",
      "--memory-limit (-M)      Memory limit for extension stage in GB\r\n",
      "--no-unlink              Do not unlink temporary files.\r\n",
      "--target-indexed         Enable target-indexed mode\r\n",
      "--ignore-warnings        Ignore warnings\r\n",
      "\r\n",
      "View options:\r\n",
      "--daa (-a)               DIAMOND alignment archive (DAA) file\r\n",
      "--forwardonly            only show alignments of forward strand\r\n",
      "\r\n",
      "Getseq options:\r\n",
      "--seq                    Space-separated list of sequence numbers to display.\r\n",
      "\r\n",
      "Online documentation at http://www.diamondsearch.org\r\n"
     ]
    }
   ],
   "source": [
    "! ./diamond help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "31cf4083",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir search_diamond"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "64d11657",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\n",
      "\n",
      "#CPU threads: 12\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "Database input file: proteins/all_proteins.faa\n",
      "Opening the database file...  [0s]\n",
      "Loading sequences...  [0.021s]\n",
      "Masking sequences...  [0.037s]\n",
      "Writing sequences...  [0.003s]\n",
      "Hashing sequences...  [0.001s]\n",
      "Loading sequences...  [0s]\n",
      "Writing trailer...  [0s]\n",
      "Closing the input file...  [0s]\n",
      "Closing the database file...  [0.001s]\n",
      "\n",
      "Database sequences  16435\n",
      "  Database letters  7015168\n",
      "     Database hash  590081acba74e419860a4dfdcf0de70e\n",
      "        Total time  0.065000s\n"
     ]
    }
   ],
   "source": [
    "# creating a diamond-formatted database file\n",
    "! ./diamond makedb --in proteins/all_proteins.faa -d search_diamond/reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "ea9462ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\n",
      "\n",
      "#CPU threads: 12\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "Temporary directory: search_diamond\n",
      "#Target sequences to report alignments for: 25\n",
      "Opening the database...  [0.004s]\n",
      "Database: search_diamond/reference.dmnd (type: Diamond database, sequences: 16435, letters: 7015168)\n",
      "Block size = 2000000000\n",
      "Opening the input file...  [0s]\n",
      "Opening the output file...  [0s]\n",
      "Loading query sequences...  [0s]\n",
      "Masking queries...  [0.001s]\n",
      "Algorithm: Double-indexed\n",
      "Building query histograms...  [0s]\n",
      "Allocating buffers...  [0s]\n",
      "Loading reference sequences...  [0.006s]\n",
      "Masking reference...  [0.036s]\n",
      "Initializing temporary storage...  [0s]\n",
      "Building reference histograms...  [0.029s]\n",
      "Allocating buffers...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 1/4.\n",
      "Building reference seed array...  [0.011s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0.011s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 2/4.\n",
      "Building reference seed array...  [0.017s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 3/4.\n",
      "Building reference seed array...  [0.013s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 4/4.\n",
      "Building reference seed array...  [0.01s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 1/4.\n",
      "Building reference seed array...  [0.013s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 2/4.\n",
      "Building reference seed array...  [0.014s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 3/4.\n",
      "Building reference seed array...  [0.012s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 4/4.\n",
      "Building reference seed array...  [0.011s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Deallocating buffers...  [0s]\n",
      "Clearing query masking...  [0s]\n",
      "Computing alignments...  [0.003s]\n",
      "Deallocating reference...  [0s]\n",
      "Loading reference sequences...  [0s]\n",
      "Deallocating buffers...  [0s]\n",
      "Deallocating queries...  [0s]\n",
      "Loading query sequences...  [0s]\n",
      "Closing the input file...  [0s]\n",
      "Closing the output file...  [0s]\n",
      "Cleaning up...  [0s]\n",
      "Total time = 0.262s\n",
      "Reported 3 pairwise alignments, 3 HSPs.\n",
      "3 queries aligned.\n"
     ]
    }
   ],
   "source": [
    "# running a search in blastp mode\n",
    "! ./diamond blastp -d search_diamond/reference.dmnd -q data/dna_associated_peptides.fa -o search_diamond/matches.tsv -f 6 --iterate --header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "302a4c97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# DIAMOND v2.0.15. http://github.com/bbuchfink/diamond\r\n",
      "# Invocation: ./diamond blastp -d search_diamond/reference.dmnd -q data/dna_associated_peptides.fa -o search_diamond/matches.tsv -f 6 --iterate --header\r\n",
      "# Fields: Query ID, Subject ID, Percentage of identical matches, Alignment length, Number of mismatches, Number of gap openings, Start of alignment in query, End of alignment in query, Start of alignment in subject, End of alignment in subject, Expected value, Bit score\r\n",
      "8\tg4106.t1\t100\t18\t0\t0\t1\t18\t222\t239\t5.55e-07\t40.8\r\n",
      "20\tg12510.t1\t100\t18\t0\t0\t1\t18\t425\t442\t1.43e-06\t39.7\r\n",
      "21\tg12510.t1\t100\t22\t0\t0\t1\t22\t443\t464\t2.46e-09\t47.8\r\n"
     ]
    }
   ],
   "source": [
    "! cat search_diamond/matches.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "980d5f2b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MCAIWENSLVGSTDNIVMRNVVAAEETISRLLCHLTSIDMWVSQNPSFDLRQKKARLPSQSRCIVRWVCLQASLQVVKDAQLAFVTVCFVLQLMEKKCAWNVTSRSIIMVTTTTKTLNYSIIRGNHTNSVNATITSTPGFNINYFGKIPGILKVLELVTIVVALGLAASRRREAPGLELPAADSWRIEYFLTGEVFFLCVMSAGLMWLVAILLTYMMHVVSSVITPKTTIHEVVFNFLFMLMFWTAGIVELAQTVPNSYPYNGRPGMEQGDWGCRVAAGALSLFIGTFFLISFALAVKELQGPKRSAPLSASPISARDHPIVTVVALGLAASRRREGTTTVVGWPVEYFLAGEVFFLCICAAALTYLFGQLTTFAWSVVSSVLIPKTTYFEIVINFLLFLAWATAGIVQLAMTCREAWNGREIRAPVNGIVTDANGNQIQVRTNAAGQLIGPGGIAINDAGLTRAVYRSPDMLFQDWGCRIAAGSLALFLSIFHLISFALAVKEFQGPRNEAR',\n",
       " 'APVNGIVTDANGNQIQVR',\n",
       " 'TNAAGQLIGPGGIAINDAGLTR')"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g12510.t1\"], peptides[20], peptides[21]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "cfd54fd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('g702.t1', '18', 174),\n",
       " ('g702.t1', '35', 157),\n",
       " ('g1285.t1', '24', 137),\n",
       " ('g3428.t1', '10', 159),\n",
       " ('g3428.t1', '11', 128),\n",
       " ('g3679.t1', '42', 20),\n",
       " ('g4106.t1', '8', 221),\n",
       " ('g5237.t1', '19', 90),\n",
       " ('g5502.t1', '24', 135),\n",
       " ('g5503.t1', '24', 135),\n",
       " ('g5510.t1', '16', 168),\n",
       " ('g5616.t1', '12', 154),\n",
       " ('g5616.t1', '13', 142),\n",
       " ('g5641.t1', '1', 24),\n",
       " ('g5641.t1', '2', 35),\n",
       " ('g5641.t1', '3', 154),\n",
       " ('g5641.t1', '4', 143),\n",
       " ('g5641.t1', '24', 127),\n",
       " ('g10513.t1', '9', 304),\n",
       " ('g10514.t1', '37', 220),\n",
       " ('g10514.t1', '38', 41),\n",
       " ('g12510.t1', '17', 305),\n",
       " ('g12510.t1', '20', 424),\n",
       " ('g12510.t1', '21', 442),\n",
       " ('g12562.t1', '24', 127),\n",
       " ('g12562.t1', '39', 160),\n",
       " ('g13530.t1', '5', 223),\n",
       " ('g13530.t1', '6', 87),\n",
       " ('g14472.t1', '7', 367),\n",
       " ('g15153.t1', '1', 24),\n",
       " ('g15153.t1', '15', 44),\n",
       " ('g15153.t1', '24', 127),\n",
       " ('g15484.t1', '32', 830),\n",
       " ('g15484.t1', '33', 790),\n",
       " ('g15484.t1', '34', 754)]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def hamming_distance(s1: str, s2: str) -> int:\n",
    "    assert len(s1) == len(s2), \"Strings have different lengths\"\n",
    "    for i in range(len(s1)):\n",
    "        if s1[i] != s2[i]:\n",
    "            return False\n",
    "    return True\n",
    "\n",
    "\n",
    "def pattern_matching(text: str, pattern: str, mismatches: int = 0) -> list[int]:\n",
    "    res = []\n",
    "    pattern_len = len(pattern)\n",
    "    for i in range(len(text) - pattern_len + 1):\n",
    "        if hamming_distance(text[i:i + pattern_len], pattern):\n",
    "            res.append(i)\n",
    "    return res\n",
    "\n",
    "matches = []\n",
    "for protein in proteins.items():\n",
    "    for peptide in peptides.items():\n",
    "        res = pattern_matching(protein[1], peptide[1])\n",
    "        for ind in res:\n",
    "            matches.append((protein[0], peptide[0], ind))\n",
    "\n",
    "matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "98076cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "35"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matches)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0d9a02b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MKCLIVFSAFVASALAANGDQQPKVWDALNVLRGKDALFADQIQELLMNVVPGQDFPLLSEVPDTAVDCSRFKQPGFYADTEQGRCQVFHRCDINGHLTGYLCPNMTLFNQITLVCDWFYNVDCSQARQFYDYSNSRLYQEGVALLDDQNDFEVRARVVEQTGASSSRRLSGSRSSSSSSQESQSSSSQVRRRKSRKSKXXXXXXXXQSVSQEFA',\n",
       " 'SSSSSSQESQSSSSQVR',\n",
       " 'VVEQTGASSSR')"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g702.t1\"], peptides[\"18\"], peptides[\"35\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d2d58cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['g702.t1',\n",
       " 'g1285.t1',\n",
       " 'g3428.t1',\n",
       " 'g3679.t1',\n",
       " 'g4106.t1',\n",
       " 'g5237.t1',\n",
       " 'g5502.t1',\n",
       " 'g5503.t1',\n",
       " 'g5510.t1',\n",
       " 'g5616.t1',\n",
       " 'g5641.t1',\n",
       " 'g10513.t1',\n",
       " 'g10514.t1',\n",
       " 'g12510.t1',\n",
       " 'g12562.t1',\n",
       " 'g13530.t1',\n",
       " 'g14472.t1',\n",
       " 'g15153.t1',\n",
       " 'g15484.t1']"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matched_proteins = [matches[0][0]]\n",
    "for prot, *_ in matches:\n",
    "    if matched_proteins[-1] != prot:\n",
    "        matched_proteins.append(prot)\n",
    "matched_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "0bcb96e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"matched_proteins.fa\", 'w') as f_out:\n",
    "    for protein_id in matched_proteins:\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "33ea64b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir filter_blast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b792e333",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 12/15/2022 14:40:37\n",
      "New DB name:   /home/nikitos/Bioinformatics/BI/Workshop_on_bioinformatics/Project4/filter_blast/reference\n",
      "New DB title:  proteins/all_proteins.faa\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 16435 sequences in 0.177495 seconds.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! makeblastdb -in proteins/all_proteins.faa -dbtype prot -out filter_blast/reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "f066b860",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\r\n",
      "  blastp [-h] [-help] [-import_search_strategy filename]\r\n",
      "    [-export_search_strategy filename] [-task task_name] [-db database_name]\r\n",
      "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\r\n",
      "    [-negative_gilist filename] [-negative_seqidlist filename]\r\n",
      "    [-taxids taxids] [-negative_taxids taxids] [-taxidlist filename]\r\n",
      "    [-negative_taxidlist filename] [-ipglist filename]\r\n",
      "    [-negative_ipglist filename] [-entrez_query entrez_query]\r\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\r\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\r\n",
      "    [-out output_file] [-evalue evalue] [-word_size int_value]\r\n",
      "    [-gapopen open_penalty] [-gapextend extend_penalty]\r\n",
      "    [-qcov_hsp_perc float_value] [-max_hsps int_value]\r\n",
      "    [-xdrop_ungap float_value] [-xdrop_gap float_value]\r\n",
      "    [-xdrop_gap_final float_value] [-searchsp int_value] [-seg SEG_options]\r\n",
      "    [-soft_masking soft_masking] [-matrix matrix_name]\r\n",
      "    [-threshold float_value] [-culling_limit int_value]\r\n",
      "    [-best_hit_overhang float_value] [-best_hit_score_edge float_value]\r\n",
      "    [-subject_besthit] [-window_size int_value] [-lcase_masking]\r\n",
      "    [-query_loc range] [-parse_deflines] [-outfmt format] [-show_gis]\r\n",
      "    [-num_descriptions int_value] [-num_alignments int_value]\r\n",
      "    [-line_length line_length] [-html] [-sorthits sort_hits]\r\n",
      "    [-sorthsps sort_hsps] [-max_target_seqs num_sequences]\r\n",
      "    [-num_threads int_value] [-mt_mode int_value] [-ungapped] [-remote]\r\n",
      "    [-comp_based_stats compo] [-use_sw_tback] [-version]\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "   Protein-Protein BLAST 2.12.0+\r\n",
      "\r\n",
      "OPTIONAL ARGUMENTS\r\n",
      " -h\r\n",
      "   Print USAGE and DESCRIPTION;  ignore all other parameters\r\n",
      " -help\r\n",
      "   Print USAGE, DESCRIPTION and ARGUMENTS; ignore all other parameters\r\n",
      " -version\r\n",
      "   Print version number;  ignore other arguments\r\n",
      "\r\n",
      " *** Input query options\r\n",
      " -query <File_In>\r\n",
      "   Input file name\r\n",
      "   Default = `-'\r\n",
      " -query_loc <String>\r\n",
      "   Location on the query sequence in 1-based offsets (Format: start-stop)\r\n",
      "\r\n",
      " *** General search options\r\n",
      " -task <String, Permissible values: 'blastp' 'blastp-fast' 'blastp-short' >\r\n",
      "   Task to execute\r\n",
      "   Default = `blastp'\r\n",
      " -db <String>\r\n",
      "   BLAST database name\r\n",
      "    * Incompatible with:  subject, subject_loc\r\n",
      " -out <File_Out>\r\n",
      "   Output file name\r\n",
      "   Default = `-'\r\n",
      " -evalue <Real>\r\n",
      "   Expectation value (E) threshold for saving hits \r\n",
      "   Default = `10'\r\n",
      " -word_size <Integer, >=2>\r\n",
      "   Word size for wordfinder algorithm\r\n",
      " -gapopen <Integer>\r\n",
      "   Cost to open a gap\r\n",
      " -gapextend <Integer>\r\n",
      "   Cost to extend a gap\r\n",
      " -matrix <String>\r\n",
      "   Scoring matrix name (normally BLOSUM62)\r\n",
      " -threshold <Real, >=0>\r\n",
      "   Minimum word score such that the word is added to the BLAST lookup table\r\n",
      " -comp_based_stats <String>\r\n",
      "   Use composition-based statistics:\r\n",
      "       D or d: default (equivalent to 2 )\r\n",
      "       0 or F or f: No composition-based statistics\r\n",
      "       1: Composition-based statistics as in NAR 29:2994-3005, 2001\r\n",
      "       2 or T or t : Composition-based score adjustment as in Bioinformatics\r\n",
      "   21:902-911,\r\n",
      "       2005, conditioned on sequence properties\r\n",
      "       3: Composition-based score adjustment as in Bioinformatics 21:902-911,\r\n",
      "       2005, unconditionally\r\n",
      "   Default = `2'\r\n",
      "\r\n",
      " *** BLAST-2-Sequences options\r\n",
      " -subject <File_In>\r\n",
      "   Subject sequence(s) to search\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   negative_seqidlist, taxids, taxidlist, negative_taxids, negative_taxidlist,\r\n",
      "   ipglist, negative_ipglist, db_soft_mask, db_hard_mask\r\n",
      " -subject_loc <String>\r\n",
      "   Location on the subject sequence in 1-based offsets (Format: start-stop)\r\n",
      "    * Incompatible with:  db, gilist, seqidlist, negative_gilist,\r\n",
      "   negative_seqidlist, taxids, taxidlist, negative_taxids, negative_taxidlist,\r\n",
      "   ipglist, negative_ipglist, db_soft_mask, db_hard_mask, remote\r\n",
      "\r\n",
      " *** Formatting options\r\n",
      " -outfmt <String>\r\n",
      "   alignment view options:\r\n",
      "     0 = Pairwise,\r\n",
      "     1 = Query-anchored showing identities,\r\n",
      "     2 = Query-anchored no identities,\r\n",
      "     3 = Flat query-anchored showing identities,\r\n",
      "     4 = Flat query-anchored no identities,\r\n",
      "     5 = BLAST XML,\r\n",
      "     6 = Tabular,\r\n",
      "     7 = Tabular with comment lines,\r\n",
      "     8 = Seqalign (Text ASN.1),\r\n",
      "     9 = Seqalign (Binary ASN.1),\r\n",
      "    10 = Comma-separated values,\r\n",
      "    11 = BLAST archive (ASN.1),\r\n",
      "    12 = Seqalign (JSON),\r\n",
      "    13 = Multiple-file BLAST JSON,\r\n",
      "    14 = Multiple-file BLAST XML2,\r\n",
      "    15 = Single-file BLAST JSON,\r\n",
      "    16 = Single-file BLAST XML2,\r\n",
      "    18 = Organism Report\r\n",
      "   \r\n",
      "   Options 6, 7 and 10 can be additionally configured to produce\r\n",
      "   a custom format specified by space delimited format specifiers,\r\n",
      "   or by a token specified by the delim keyword.\r\n",
      "    E.g.: \"10 delim=@ qacc sacc score\".\r\n",
      "   The delim keyword must appear after the numeric output format\r\n",
      "   specification.\r\n",
      "   The supported format specifiers are:\r\n",
      "   \t    qseqid means Query Seq-id\r\n",
      "   \t       qgi means Query GI\r\n",
      "   \t      qacc means Query accesion\r\n",
      "   \t   qaccver means Query accesion.version\r\n",
      "   \t      qlen means Query sequence length\r\n",
      "   \t    sseqid means Subject Seq-id\r\n",
      "   \t sallseqid means All subject Seq-id(s), separated by a ';'\r\n",
      "   \t       sgi means Subject GI\r\n",
      "   \t    sallgi means All subject GIs\r\n",
      "   \t      sacc means Subject accession\r\n",
      "   \t   saccver means Subject accession.version\r\n",
      "   \t   sallacc means All subject accessions\r\n",
      "   \t      slen means Subject sequence length\r\n",
      "   \t    qstart means Start of alignment in query\r\n",
      "   \t      qend means End of alignment in query\r\n",
      "   \t    sstart means Start of alignment in subject\r\n",
      "   \t      send means End of alignment in subject\r\n",
      "   \t      qseq means Aligned part of query sequence\r\n",
      "   \t      sseq means Aligned part of subject sequence\r\n",
      "   \t    evalue means Expect value\r\n",
      "   \t  bitscore means Bit score\r\n",
      "   \t     score means Raw score\r\n",
      "   \t    length means Alignment length\r\n",
      "   \t    pident means Percentage of identical matches\r\n",
      "   \t    nident means Number of identical matches\r\n",
      "   \t  mismatch means Number of mismatches\r\n",
      "   \t  positive means Number of positive-scoring matches\r\n",
      "   \t   gapopen means Number of gap openings\r\n",
      "   \t      gaps means Total number of gaps\r\n",
      "   \t      ppos means Percentage of positive-scoring matches\r\n",
      "   \t    frames means Query and subject frames separated by a '/'\r\n",
      "   \t    qframe means Query frame\r\n",
      "   \t    sframe means Subject frame\r\n",
      "   \t      btop means Blast traceback operations (BTOP)\r\n",
      "   \t    staxid means Subject Taxonomy ID\r\n",
      "   \t  ssciname means Subject Scientific Name\r\n",
      "   \t  scomname means Subject Common Name\r\n",
      "   \tsblastname means Subject Blast Name\r\n",
      "   \t sskingdom means Subject Super Kingdom\r\n",
      "   \t   staxids means unique Subject Taxonomy ID(s), separated by a ';'\r\n",
      "   \t\t\t (in numerical order)\r\n",
      "   \t sscinames means unique Subject Scientific Name(s), separated by a ';'\r\n",
      "   \t scomnames means unique Subject Common Name(s), separated by a ';'\r\n",
      "   \tsblastnames means unique Subject Blast Name(s), separated by a ';'\r\n",
      "   \t\t\t (in alphabetical order)\r\n",
      "   \tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\r\n",
      "   \t\t\t (in alphabetical order) \r\n",
      "   \t    stitle means Subject Title\r\n",
      "   \tsalltitles means All Subject Title(s), separated by a '<>'\r\n",
      "   \t   sstrand means Subject Strand\r\n",
      "   \t     qcovs means Query Coverage Per Subject\r\n",
      "   \t   qcovhsp means Query Coverage Per HSP\r\n",
      "   \t    qcovus means Query Coverage Per Unique Subject (blastn only)\r\n",
      "   When not provided, the default value is:\r\n",
      "   'qaccver saccver pident length mismatch gapopen qstart qend sstart send\r\n",
      "   evalue bitscore', which is equivalent to the keyword 'std'\r\n",
      "   Default = `0'\r\n",
      " -show_gis\r\n",
      "   Show NCBI GIs in deflines?\r\n",
      " -num_descriptions <Integer, >=0>\r\n",
      "   Number of database sequences to show one-line descriptions for\r\n",
      "   Not applicable for outfmt > 4\r\n",
      "   Default = `500'\r\n",
      "    * Incompatible with:  max_target_seqs\r\n",
      " -num_alignments <Integer, >=0>\r\n",
      "   Number of database sequences to show alignments for\r\n",
      "   Default = `250'\r\n",
      "    * Incompatible with:  max_target_seqs\r\n",
      " -line_length <Integer, >=1>\r\n",
      "   Line length for formatting alignments\r\n",
      "   Not applicable for outfmt > 4\r\n",
      "   Default = `60'\r\n",
      " -html\r\n",
      "   Produce HTML output?\r\n",
      " -sorthits <Integer, (>=0 and =<4)>\r\n",
      "   Sorting option for hits:\r\n",
      "   alignment view options:\r\n",
      "     0 = Sort by evalue,\r\n",
      "     1 = Sort by bit score,\r\n",
      "     2 = Sort by total score,\r\n",
      "     3 = Sort by percent identity,\r\n",
      "     4 = Sort by query coverage\r\n",
      "   Not applicable for outfmt > 4\r\n",
      " -sorthsps <Integer, (>=0 and =<4)>\r\n",
      "   Sorting option for hps:\r\n",
      "     0 = Sort by hsp evalue,\r\n",
      "     1 = Sort by hsp score,\r\n",
      "     2 = Sort by hsp query start,\r\n",
      "     3 = Sort by hsp percent identity,\r\n",
      "     4 = Sort by hsp subject start\r\n",
      "   Not applicable for outfmt != 0\r\n",
      "\r\n",
      " *** Query filtering options\r\n",
      " -seg <String>\r\n",
      "   Filter query sequence with SEG (Format: 'yes', 'window locut hicut', or\r\n",
      "   'no' to disable)\r\n",
      "   Default = `no'\r\n",
      " -soft_masking <Boolean>\r\n",
      "   Apply filtering locations as soft masks\r\n",
      "   Default = `false'\r\n",
      " -lcase_masking\r\n",
      "   Use lower case filtering in query and subject sequence(s)?\r\n",
      "\r\n",
      " *** Restrict search or results\r\n",
      " -gilist <String>\r\n",
      "   Restrict search of database to list of GIs\r\n",
      "    * Incompatible with:  seqidlist, taxids, taxidlist, negative_gilist,\r\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -seqidlist <String>\r\n",
      "   Restrict search of database to list of SeqIDs\r\n",
      "    * Incompatible with:  gilist, taxids, taxidlist, negative_gilist,\r\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_gilist <String>\r\n",
      "   Restrict search of database to everything except the specified GIs\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\r\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_seqidlist <String>\r\n",
      "   Restrict search of database to everything except the specified SeqIDs\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\r\n",
      "   negative_gilist, negative_taxids, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -taxids <String>\r\n",
      "   Restrict search of database to include only the specified taxonomy IDs\r\n",
      "   (multiple IDs delimited by ',')\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxidlist, negative_gilist,\r\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_taxids <String>\r\n",
      "   Restrict search of database to everything except the specified taxonomy IDs\r\n",
      "   (multiple IDs delimited by ',')\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\r\n",
      "   negative_gilist, negative_seqidlist, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -taxidlist <String>\r\n",
      "   Restrict search of database to include only the specified taxonomy IDs\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, negative_gilist,\r\n",
      "   negative_seqidlist, negative_taxids, negative_taxidlist, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -negative_taxidlist <String>\r\n",
      "   Restrict search of database to everything except the specified taxonomy IDs\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\r\n",
      "   negative_gilist, negative_seqidlist, negative_taxids, remote, subject,\r\n",
      "   subject_loc\r\n",
      " -ipglist <String>\r\n",
      "   Restrict search of database to list of IPGs\r\n",
      "    * Incompatible with:  subject, subject_loc\r\n",
      " -negative_ipglist <String>\r\n",
      "   Restrict search of database to everything except the specified IPGs\r\n",
      "    * Incompatible with:  subject, subject_loc\r\n",
      " -entrez_query <String>\r\n",
      "   Restrict search with the given Entrez query\r\n",
      "    * Requires:  remote\r\n",
      " -db_soft_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as soft masking\r\n",
      "    * Incompatible with:  db_hard_mask, subject, subject_loc\r\n",
      " -db_hard_mask <String>\r\n",
      "   Filtering algorithm ID to apply to the BLAST database as hard masking\r\n",
      "    * Incompatible with:  db_soft_mask, subject, subject_loc\r\n",
      " -qcov_hsp_perc <Real, 0..100>\r\n",
      "   Percent query coverage per hsp\r\n",
      " -max_hsps <Integer, >=1>\r\n",
      "   Set maximum number of HSPs per subject sequence to save for each query\r\n",
      " -culling_limit <Integer, >=0>\r\n",
      "   If the query range of a hit is enveloped by that of at least this many\r\n",
      "   higher-scoring hits, delete the hit\r\n",
      "    * Incompatible with:  best_hit_overhang, best_hit_score_edge\r\n",
      " -best_hit_overhang <Real, (>0 and <0.5)>\r\n",
      "   Best Hit algorithm overhang value (recommended value: 0.1)\r\n",
      "    * Incompatible with:  culling_limit\r\n",
      " -best_hit_score_edge <Real, (>0 and <0.5)>\r\n",
      "   Best Hit algorithm score edge value (recommended value: 0.1)\r\n",
      "    * Incompatible with:  culling_limit\r\n",
      " -subject_besthit\r\n",
      "   Turn on best hit per subject sequence\r\n",
      " -max_target_seqs <Integer, >=1>\r\n",
      "   Maximum number of aligned sequences to keep \r\n",
      "   (value of 5 or more is recommended)\r\n",
      "   Default = `500'\r\n",
      "    * Incompatible with:  num_descriptions, num_alignments\r\n",
      "\r\n",
      " *** Statistical options\r\n",
      " -dbsize <Int8>\r\n",
      "   Effective length of the database \r\n",
      " -searchsp <Int8, >=0>\r\n",
      "   Effective length of the search space\r\n",
      "\r\n",
      " *** Search strategy options\r\n",
      " -import_search_strategy <File_In>\r\n",
      "   Search strategy to use\r\n",
      "    * Incompatible with:  export_search_strategy\r\n",
      " -export_search_strategy <File_Out>\r\n",
      "   File name to record the search strategy used\r\n",
      "    * Incompatible with:  import_search_strategy\r\n",
      "\r\n",
      " *** Extension options\r\n",
      " -xdrop_ungap <Real>\r\n",
      "   X-dropoff value (in bits) for ungapped extensions\r\n",
      " -xdrop_gap <Real>\r\n",
      "   X-dropoff value (in bits) for preliminary gapped extensions\r\n",
      " -xdrop_gap_final <Real>\r\n",
      "   X-dropoff value (in bits) for final gapped alignment\r\n",
      " -window_size <Integer, >=0>\r\n",
      "   Multiple hits window size, use 0 to specify 1-hit algorithm\r\n",
      " -ungapped\r\n",
      "   Perform ungapped alignment only?\r\n",
      "\r\n",
      " *** Miscellaneous options\r\n",
      " -parse_deflines\r\n",
      "   Should the query and subject defline(s) be parsed?\r\n",
      " -num_threads <Integer, >=1>\r\n",
      "   Number of threads (CPUs) to use in the BLAST search\r\n",
      "   Default = `1'\r\n",
      "    * Incompatible with:  remote\r\n",
      " -mt_mode <Integer, (>=0 and =<1)>\r\n",
      "   Multi-thread mode to use in BLAST search:\r\n",
      "    0 (auto) split by database \r\n",
      "    1 split by queries\r\n",
      "   Default = `0'\r\n",
      "    * Requires:  num_threads\r\n",
      " -remote\r\n",
      "   Execute search remotely?\r\n",
      "    * Incompatible with:  gilist, seqidlist, taxids, taxidlist,\r\n",
      "   negative_gilist, negative_seqidlist, negative_taxids, negative_taxidlist,\r\n",
      "   subject_loc, num_threads\r\n",
      " -use_sw_tback\r\n",
      "   Compute locally optimal Smith-Waterman alignments?\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "! blastp -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1e18e797",
   "metadata": {},
   "outputs": [],
   "source": [
    "! blastp -db filter_blast/reference -query data/dna_associated_peptides.fa -outfmt 6 -out filter_blast/matches.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "00469688",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>33</td>\n",
       "      <td>2.000</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>33</td>\n",
       "      <td>2.100</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>1.500</td>\n",
       "      <td>21.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>3.300</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>6.700</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>37</td>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>221</td>\n",
       "      <td>230</td>\n",
       "      <td>0.970</td>\n",
       "      <td>22.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>37</td>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>77.778</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>359</td>\n",
       "      <td>367</td>\n",
       "      <td>8.400</td>\n",
       "      <td>19.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>38</td>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>42</td>\n",
       "      <td>52</td>\n",
       "      <td>0.600</td>\n",
       "      <td>23.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>39</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>161</td>\n",
       "      <td>170</td>\n",
       "      <td>0.820</td>\n",
       "      <td>22.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>42</td>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>100.000</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>21</td>\n",
       "      <td>32</td>\n",
       "      <td>0.046</td>\n",
       "      <td>26.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    0          1        2   3   4   5   6   7    8    9      10    11\n",
       "0    1   g5641.t1  100.000   9   0   0   1   9   25   33  2.000  21.6\n",
       "1    1  g15153.t1  100.000   9   0   0   1   9   25   33  2.100  21.6\n",
       "2    2   g5641.t1  100.000   9   0   0   1   9   36   44  1.500  21.9\n",
       "3    2  g12562.t1   88.889   9   1   0   1   9   36   44  3.300  20.8\n",
       "4    2   g5616.t1   88.889   9   1   0   1   9   36   44  6.700  20.0\n",
       "..  ..        ...      ...  ..  ..  ..  ..  ..  ...  ...    ...   ...\n",
       "63  37  g10514.t1  100.000  10   0   0   1  10  221  230  0.970  22.7\n",
       "64  37   g2203.t1   77.778   9   2   0   1   9  359  367  8.400  19.6\n",
       "65  38  g10514.t1  100.000  11   0   0   1  11   42   52  0.600  23.1\n",
       "66  39  g12562.t1  100.000  10   0   0   1  10  161  170  0.820  22.7\n",
       "67  42   g3679.t1  100.000  12   0   0   1  12   21   32  0.046  26.6\n",
       "\n",
       "[68 rows x 12 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv(\"filter_blast/matches.tsv\", sep='\\t', header=None)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a9121596",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>3.300</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>6.700</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>77.778</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>36</td>\n",
       "      <td>44</td>\n",
       "      <td>7.500</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>72.727</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>1.600</td>\n",
       "      <td>21.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>72.727</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>3.100</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4</td>\n",
       "      <td>g702.t1</td>\n",
       "      <td>63.636</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>145</td>\n",
       "      <td>155</td>\n",
       "      <td>3.700</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>4</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>63.636</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>5.400</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>8</td>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>62.500</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>64</td>\n",
       "      <td>79</td>\n",
       "      <td>0.650</td>\n",
       "      <td>23.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>8</td>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>57.143</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>134</td>\n",
       "      <td>147</td>\n",
       "      <td>8.800</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9</td>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>152</td>\n",
       "      <td>166</td>\n",
       "      <td>3.900</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>9</td>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>224</td>\n",
       "      <td>238</td>\n",
       "      <td>4.900</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>9</td>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>204</td>\n",
       "      <td>218</td>\n",
       "      <td>5.200</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10</td>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>87.500</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>872</td>\n",
       "      <td>879</td>\n",
       "      <td>8.300</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>13</td>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>91.667</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>56</td>\n",
       "      <td>67</td>\n",
       "      <td>0.094</td>\n",
       "      <td>25.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>13</td>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>91.667</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>151</td>\n",
       "      <td>162</td>\n",
       "      <td>0.140</td>\n",
       "      <td>25.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>13</td>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>90.909</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>0.210</td>\n",
       "      <td>24.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>13</td>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>91.667</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>151</td>\n",
       "      <td>162</td>\n",
       "      <td>0.280</td>\n",
       "      <td>24.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>13</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>72.727</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>144</td>\n",
       "      <td>154</td>\n",
       "      <td>3.000</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>15</td>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>54</td>\n",
       "      <td>0.950</td>\n",
       "      <td>22.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>15</td>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>54</td>\n",
       "      <td>1.100</td>\n",
       "      <td>22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>15</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "      <td>54</td>\n",
       "      <td>1.400</td>\n",
       "      <td>21.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>15</td>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>53</td>\n",
       "      <td>62</td>\n",
       "      <td>5.400</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>15</td>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>80.000</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>53</td>\n",
       "      <td>62</td>\n",
       "      <td>6.600</td>\n",
       "      <td>20.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>20</td>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>66.667</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>166</td>\n",
       "      <td>177</td>\n",
       "      <td>4.100</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>20</td>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>61.538</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>148</td>\n",
       "      <td>160</td>\n",
       "      <td>5.600</td>\n",
       "      <td>21.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>21</td>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>56.250</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>22</td>\n",
       "      <td>598</td>\n",
       "      <td>613</td>\n",
       "      <td>2.400</td>\n",
       "      <td>22.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>21</td>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>56.250</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>927</td>\n",
       "      <td>942</td>\n",
       "      <td>9.900</td>\n",
       "      <td>20.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>24</td>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>128</td>\n",
       "      <td>136</td>\n",
       "      <td>2.300</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>24</td>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>136</td>\n",
       "      <td>144</td>\n",
       "      <td>4.900</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>24</td>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>88.889</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>113</td>\n",
       "      <td>121</td>\n",
       "      <td>5.300</td>\n",
       "      <td>20.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>34</td>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>53.846</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>556</td>\n",
       "      <td>568</td>\n",
       "      <td>3.500</td>\n",
       "      <td>21.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>37</td>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>77.778</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>359</td>\n",
       "      <td>367</td>\n",
       "      <td>8.400</td>\n",
       "      <td>19.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    0          1       2   3   4   5   6   7    8    9      10    11\n",
       "3    2  g12562.t1  88.889   9   1   0   1   9   36   44  3.300  20.8\n",
       "4    2   g5616.t1  88.889   9   1   0   1   9   36   44  6.700  20.0\n",
       "5    2  g15153.t1  77.778   9   2   0   1   9   36   44  7.500  20.0\n",
       "8    4  g12562.t1  72.727  11   3   0   1  11  144  154  1.600  21.9\n",
       "9    4   g5616.t1  72.727  11   3   0   1  11  144  154  3.100  21.2\n",
       "10   4    g702.t1  63.636  11   4   0   1  11  145  155  3.700  21.2\n",
       "11   4  g15153.t1  63.636  11   4   0   1  11  144  154  5.400  20.4\n",
       "16   8   g7861.t1  62.500  16   6   0   1  16   64   79  0.650  23.5\n",
       "17   8   g4970.t1  57.143  14   6   0   5  18  134  147  8.800  20.4\n",
       "19   9  g11806.t1  66.667  15   5   0   1  15  152  166  3.900  21.2\n",
       "20   9  g16318.t1  66.667  15   5   0   1  15  224  238  4.900  21.2\n",
       "21   9  g16368.t1  66.667  15   5   0   1  15  204  218  5.200  20.8\n",
       "23  10  g11513.t1  87.500   8   1   0   2   9  872  879  8.300  20.0\n",
       "28  13   g5443.t1  91.667  12   1   0   1  12   56   67  0.094  25.8\n",
       "29  13   g5502.t1  91.667  12   1   0   1  12  151  162  0.140  25.0\n",
       "30  13  g15153.t1  90.909  11   1   0   2  12  144  154  0.210  24.6\n",
       "31  13   g5503.t1  91.667  12   1   0   1  12  151  162  0.280  24.3\n",
       "33  13   g5641.t1  72.727  11   3   0   2  12  144  154  3.000  21.6\n",
       "35  15  g12562.t1  80.000  10   2   0   1  10   45   54  0.950  22.7\n",
       "36  15   g5641.t1  80.000  10   2   0   1  10   45   54  1.100  22.3\n",
       "37  15   g5616.t1  80.000  10   2   0   1  10   45   54  1.400  21.9\n",
       "38  15   g5503.t1  80.000  10   2   0   1  10   53   62  5.400  20.4\n",
       "39  15   g5502.t1  80.000  10   2   0   1  10   53   62  6.600  20.0\n",
       "44  20  g11320.t1  66.667  12   4   0   4  15  166  177  4.100  21.6\n",
       "45  20   g8312.t1  61.538  13   5   0   1  13  148  160  5.600  21.2\n",
       "47  21   g5927.t1  56.250  16   7   0   7  22  598  613  2.400  22.3\n",
       "48  21   g8100.t1  56.250  16   7   0   1  16  927  942  9.900  20.8\n",
       "55  24   g5616.t1  88.889   9   1   0   1   9  128  136  2.300  21.6\n",
       "56  24  g12388.t1  88.889   9   1   0   1   9  136  144  4.900  20.4\n",
       "57  24   g5467.t1  88.889   9   1   0   1   9  113  121  5.300  20.4\n",
       "61  34  g11960.t1  53.846  13   6   0   4  16  556  568  3.500  21.6\n",
       "64  37   g2203.t1  77.778   9   2   0   1   9  359  367  8.400  19.6"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[2] < 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "6ba807c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('VLLDNQDDYELQ', 'VLLDNQDDYELK')"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g5443.t1\"][55:67], peptides[13]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "7c3efedf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('LLDNQDDYELQ', 'VLLDNQDDYELK')"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g15153.t1\"][143:154], peptides[13]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "13603ead",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"proteins/filtered_by_peptides_blast.fa\", 'w') as f_out:\n",
    "    for protein_id in sorted(list(set(df[1])), key=lambda x: int(x[1:-3])):\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "c9732114",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(df[1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe971ee9",
   "metadata": {},
   "source": [
    "## Localization prediction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a358a8f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g702.t1</td>\n",
       "      <td>extr: 29, plas: 2, lyso: 1</td>\n",
       "      <td>extr: 6, vacu: 5, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>extr: 25, plas: 5, mito: 1, lyso: 1</td>\n",
       "      <td>vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1</td>\n",
       "      <td>extr: 24, golg: 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>plas: 29, nucl: 2, golg: 1</td>\n",
       "      <td>cyto: 5, nucl: 4, E.R.: 3, vacu: 2</td>\n",
       "      <td>plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>mito: 18, cyto: 11, extr: 2, nucl: 1</td>\n",
       "      <td>chlo: 6, mito: 6, nucl: 1, cyto: 1</td>\n",
       "      <td>cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1</td>\n",
       "      <td>chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...</td>\n",
       "      <td>chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1</td>\n",
       "      <td>pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>plas: 32</td>\n",
       "      <td>nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...</td>\n",
       "      <td>plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>plas: 24, mito: 8</td>\n",
       "      <td>chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1</td>\n",
       "      <td>extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>extr: 28, nucl: 3, cyto: 1</td>\n",
       "      <td>chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1</td>\n",
       "      <td>extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>extr: 27, plas: 4, mito: 1</td>\n",
       "      <td>extr: 6, vacu: 5, chlo: 2, golg: 1</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>chlo: 9, extr: 5</td>\n",
       "      <td>extr: 27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>extr: 29, plas: 1, mito: 1, lyso: 1</td>\n",
       "      <td>chlo: 10, extr: 3, cyto: 1</td>\n",
       "      <td>extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>plas: 23, mito: 7, E.R.: 1, golg: 1</td>\n",
       "      <td>chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...</td>\n",
       "      <td>plas: 21, mito: 3, E.R.: 2, pero: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>extr: 31, mito: 1</td>\n",
       "      <td>extr: 12, vacu: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>extr: 8, vacu: 4, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1</td>\n",
       "      <td>extr: 11, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...</td>\n",
       "      <td>nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...</td>\n",
       "      <td>cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>extr: 25, plas: 4, mito: 2, lyso: 1</td>\n",
       "      <td>chlo: 10, vacu: 2, cyto: 1, mito: 1</td>\n",
       "      <td>extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>plas: 29, cyto: 3</td>\n",
       "      <td>plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1</td>\n",
       "      <td>mito: 13, plas: 8, cyto: 5, vacu: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>extr: 30, lyso: 2</td>\n",
       "      <td>extr: 9, vacu: 3, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...</td>\n",
       "      <td>nucl: 11, chlo: 1, cyto: 1, cysk: 1</td>\n",
       "      <td>extr: 24, nucl: 1, mito: 1, cyto: 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>extr: 32</td>\n",
       "      <td>extr: 11, vacu: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "0     g702.t1                         extr: 29, plas: 2, lyso: 1   \n",
       "1    g1285.t1                extr: 25, plas: 5, mito: 1, lyso: 1   \n",
       "2    g2203.t1                         plas: 29, nucl: 2, golg: 1   \n",
       "3    g3428.t1               mito: 18, cyto: 11, extr: 2, nucl: 1   \n",
       "4    g3679.t1       extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1   \n",
       "5    g4106.t1  E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...   \n",
       "6    g4970.t1                                           plas: 32   \n",
       "7    g5237.t1                                  plas: 24, mito: 8   \n",
       "8    g5443.t1                         extr: 28, nucl: 3, cyto: 1   \n",
       "9    g5467.t1                         extr: 27, plas: 4, mito: 1   \n",
       "10   g5502.t1                                  extr: 31, lyso: 1   \n",
       "11   g5503.t1                extr: 29, plas: 1, mito: 1, lyso: 1   \n",
       "12   g5510.t1                plas: 23, mito: 7, E.R.: 1, golg: 1   \n",
       "13   g5616.t1                                  extr: 31, mito: 1   \n",
       "14   g5641.t1                                  extr: 31, lyso: 1   \n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "21  g11320.t1      plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1   \n",
       "22  g11513.t1  cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "25  g12388.t1                extr: 25, plas: 4, mito: 2, lyso: 1   \n",
       "26  g12510.t1                                  plas: 29, cyto: 3   \n",
       "27  g12562.t1                                  extr: 30, lyso: 2   \n",
       "28  g13530.t1  extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "30  g15153.t1                                           extr: 32   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                          wolf_plants  \\\n",
       "0                  extr: 6, vacu: 5, golg: 2, E.R.: 1   \n",
       "1         vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1   \n",
       "2                  cyto: 5, nucl: 4, E.R.: 3, vacu: 2   \n",
       "3                  chlo: 6, mito: 6, nucl: 1, cyto: 1   \n",
       "4         chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1   \n",
       "5         chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1   \n",
       "6   nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...   \n",
       "7         chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1   \n",
       "8         chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1   \n",
       "9                  extr: 6, vacu: 5, chlo: 2, golg: 1   \n",
       "10                                   chlo: 9, extr: 5   \n",
       "11                         chlo: 10, extr: 3, cyto: 1   \n",
       "12  chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...   \n",
       "13                                  extr: 12, vacu: 2   \n",
       "14                          extr: 8, vacu: 4, golg: 2   \n",
       "15                                           nucl: 14   \n",
       "16        nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17        nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18                 nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                                  nucl: 13, chlo: 1   \n",
       "20                          nucl: 8, cyto: 5, plas: 1   \n",
       "21                         extr: 11, golg: 2, E.R.: 1   \n",
       "22  nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...   \n",
       "23       nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                         nucl: 13.5, cyto_nucl: 7.5   \n",
       "25                chlo: 10, vacu: 2, cyto: 1, mito: 1   \n",
       "26        plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1   \n",
       "27                          extr: 9, vacu: 3, golg: 2   \n",
       "28                nucl: 11, chlo: 1, cyto: 1, cysk: 1   \n",
       "29                                           nucl: 14   \n",
       "30                         extr: 11, vacu: 2, E.R.: 1   \n",
       "31                nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                                  nucl: 13, chlo: 1   \n",
       "33                                  nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi  \n",
       "0                                            extr: 26  \n",
       "1                                   extr: 24, golg: 2  \n",
       "2   plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...  \n",
       "3   cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...  \n",
       "4                                            extr: 26  \n",
       "5   pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...  \n",
       "6   plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...  \n",
       "7   extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...  \n",
       "8   extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...  \n",
       "9                                            extr: 25  \n",
       "10                                           extr: 27  \n",
       "11       extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1  \n",
       "12                plas: 21, mito: 3, E.R.: 2, pero: 1  \n",
       "13                                           extr: 25  \n",
       "14                                           extr: 25  \n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5  \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1  \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...  \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...  \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2  \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1  \n",
       "21                                           extr: 27  \n",
       "22  cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...  \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3  \n",
       "24                          nucl: 25, cyto_nucl: 15.5  \n",
       "25       extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1  \n",
       "26                mito: 13, plas: 8, cyto: 5, vacu: 1  \n",
       "27                                           extr: 25  \n",
       "28                extr: 24, nucl: 1, mito: 1, cyto: 1  \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...  \n",
       "30                                           extr: 27  \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3  \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...  \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "\n",
    "for org_type in [\"animals\", \"plants\", \"fungi\"]:\n",
    "    dfs.append(pd.read_csv(f\"preds_wolf/versus_{org_type}.txt\", \n",
    "                           sep=\" details \", \n",
    "                           engine=\"python\", names=[\"protein_id\", f\"wolf_{org_type}\"]))\n",
    "\n",
    "df_wolf = pd.concat([dfs[0], dfs[1][\"wolf_plants\"], dfs[2][\"wolf_fungi\"]], axis=1)\n",
    "df_wolf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ddead69e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g702.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id targetp_nonplants targetp_plants\n",
       "0     g702.t1                SP             SP\n",
       "1    g1285.t1                SP             SP\n",
       "2    g2203.t1             OTHER          OTHER\n",
       "3    g3428.t1             OTHER          OTHER\n",
       "4    g3679.t1                SP             SP\n",
       "5    g4106.t1             OTHER             SP\n",
       "6    g4970.t1             OTHER          OTHER\n",
       "7    g5237.t1             OTHER          OTHER\n",
       "8    g5443.t1             OTHER          OTHER\n",
       "9    g5467.t1                SP             SP\n",
       "10   g5502.t1                SP             SP\n",
       "11   g5503.t1                SP             SP\n",
       "12   g5510.t1             OTHER          OTHER\n",
       "13   g5616.t1                SP             SP\n",
       "14   g5641.t1                SP             SP\n",
       "15   g5927.t1             OTHER          OTHER\n",
       "16   g7861.t1             OTHER          OTHER\n",
       "17   g8100.t1             OTHER          OTHER\n",
       "18   g8312.t1             OTHER          OTHER\n",
       "19  g10513.t1             OTHER          OTHER\n",
       "20  g10514.t1             OTHER          OTHER\n",
       "21  g11320.t1                SP             SP\n",
       "22  g11513.t1             OTHER          OTHER\n",
       "23  g11806.t1             OTHER          OTHER\n",
       "24  g11960.t1             OTHER          OTHER\n",
       "25  g12388.t1                SP             SP\n",
       "26  g12510.t1             OTHER          OTHER\n",
       "27  g12562.t1                SP             SP\n",
       "28  g13530.t1                SP             SP\n",
       "29  g14472.t1             OTHER          OTHER\n",
       "30  g15153.t1                SP             SP\n",
       "31  g15484.t1             OTHER          OTHER\n",
       "32  g16318.t1             OTHER          OTHER\n",
       "33  g16368.t1             OTHER          OTHER"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "\n",
    "for org_type in [\"nonplants\", \"plants\"]:\n",
    "    dfs.append(pd.read_csv(f\"preds_targetp/versus_{org_type}.txt\", \n",
    "                           sep='\\t', \n",
    "                           header=1, \n",
    "                          )\n",
    "              )\n",
    "    dfs[-1].rename(columns={\"# ID\": \"protein_id\", \"Prediction\": f\"targetp_{org_type}\"}, inplace=True)\n",
    "\n",
    "df_targetp = pd.concat([dfs[0][[\"protein_id\", \"targetp_nonplants\"]], dfs[1][\"targetp_plants\"]], axis=1)\n",
    "df_targetp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "992369ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g702.t1</td>\n",
       "      <td>extr: 29, plas: 2, lyso: 1</td>\n",
       "      <td>extr: 6, vacu: 5, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>extr: 25, plas: 5, mito: 1, lyso: 1</td>\n",
       "      <td>vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1</td>\n",
       "      <td>extr: 24, golg: 2</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>plas: 29, nucl: 2, golg: 1</td>\n",
       "      <td>cyto: 5, nucl: 4, E.R.: 3, vacu: 2</td>\n",
       "      <td>plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>mito: 18, cyto: 11, extr: 2, nucl: 1</td>\n",
       "      <td>chlo: 6, mito: 6, nucl: 1, cyto: 1</td>\n",
       "      <td>cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1</td>\n",
       "      <td>chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...</td>\n",
       "      <td>chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1</td>\n",
       "      <td>pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>plas: 32</td>\n",
       "      <td>nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...</td>\n",
       "      <td>plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>plas: 24, mito: 8</td>\n",
       "      <td>chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1</td>\n",
       "      <td>extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>extr: 28, nucl: 3, cyto: 1</td>\n",
       "      <td>chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1</td>\n",
       "      <td>extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>g5467.t1</td>\n",
       "      <td>extr: 27, plas: 4, mito: 1</td>\n",
       "      <td>extr: 6, vacu: 5, chlo: 2, golg: 1</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>g5502.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>chlo: 9, extr: 5</td>\n",
       "      <td>extr: 27</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>g5503.t1</td>\n",
       "      <td>extr: 29, plas: 1, mito: 1, lyso: 1</td>\n",
       "      <td>chlo: 10, extr: 3, cyto: 1</td>\n",
       "      <td>extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>plas: 23, mito: 7, E.R.: 1, golg: 1</td>\n",
       "      <td>chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...</td>\n",
       "      <td>plas: 21, mito: 3, E.R.: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>g5616.t1</td>\n",
       "      <td>extr: 31, mito: 1</td>\n",
       "      <td>extr: 12, vacu: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>g5641.t1</td>\n",
       "      <td>extr: 31, lyso: 1</td>\n",
       "      <td>extr: 8, vacu: 4, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>g11320.t1</td>\n",
       "      <td>plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1</td>\n",
       "      <td>extr: 11, golg: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...</td>\n",
       "      <td>nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...</td>\n",
       "      <td>cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>g12388.t1</td>\n",
       "      <td>extr: 25, plas: 4, mito: 2, lyso: 1</td>\n",
       "      <td>chlo: 10, vacu: 2, cyto: 1, mito: 1</td>\n",
       "      <td>extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>plas: 29, cyto: 3</td>\n",
       "      <td>plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1</td>\n",
       "      <td>mito: 13, plas: 8, cyto: 5, vacu: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>g12562.t1</td>\n",
       "      <td>extr: 30, lyso: 2</td>\n",
       "      <td>extr: 9, vacu: 3, golg: 2</td>\n",
       "      <td>extr: 25</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...</td>\n",
       "      <td>nucl: 11, chlo: 1, cyto: 1, cysk: 1</td>\n",
       "      <td>extr: 24, nucl: 1, mito: 1, cyto: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>g15153.t1</td>\n",
       "      <td>extr: 32</td>\n",
       "      <td>extr: 11, vacu: 2, E.R.: 1</td>\n",
       "      <td>extr: 27</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "0     g702.t1                         extr: 29, plas: 2, lyso: 1   \n",
       "1    g1285.t1                extr: 25, plas: 5, mito: 1, lyso: 1   \n",
       "2    g2203.t1                         plas: 29, nucl: 2, golg: 1   \n",
       "3    g3428.t1               mito: 18, cyto: 11, extr: 2, nucl: 1   \n",
       "4    g3679.t1       extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1   \n",
       "5    g4106.t1  E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...   \n",
       "6    g4970.t1                                           plas: 32   \n",
       "7    g5237.t1                                  plas: 24, mito: 8   \n",
       "8    g5443.t1                         extr: 28, nucl: 3, cyto: 1   \n",
       "9    g5467.t1                         extr: 27, plas: 4, mito: 1   \n",
       "10   g5502.t1                                  extr: 31, lyso: 1   \n",
       "11   g5503.t1                extr: 29, plas: 1, mito: 1, lyso: 1   \n",
       "12   g5510.t1                plas: 23, mito: 7, E.R.: 1, golg: 1   \n",
       "13   g5616.t1                                  extr: 31, mito: 1   \n",
       "14   g5641.t1                                  extr: 31, lyso: 1   \n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "21  g11320.t1      plas: 24.5, extr_plas: 16, extr: 6.5, lyso: 1   \n",
       "22  g11513.t1  cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "25  g12388.t1                extr: 25, plas: 4, mito: 2, lyso: 1   \n",
       "26  g12510.t1                                  plas: 29, cyto: 3   \n",
       "27  g12562.t1                                  extr: 30, lyso: 2   \n",
       "28  g13530.t1  extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "30  g15153.t1                                           extr: 32   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                          wolf_plants  \\\n",
       "0                  extr: 6, vacu: 5, golg: 2, E.R.: 1   \n",
       "1         vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1   \n",
       "2                  cyto: 5, nucl: 4, E.R.: 3, vacu: 2   \n",
       "3                  chlo: 6, mito: 6, nucl: 1, cyto: 1   \n",
       "4         chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1   \n",
       "5         chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1   \n",
       "6   nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...   \n",
       "7         chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1   \n",
       "8         chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1   \n",
       "9                  extr: 6, vacu: 5, chlo: 2, golg: 1   \n",
       "10                                   chlo: 9, extr: 5   \n",
       "11                         chlo: 10, extr: 3, cyto: 1   \n",
       "12  chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...   \n",
       "13                                  extr: 12, vacu: 2   \n",
       "14                          extr: 8, vacu: 4, golg: 2   \n",
       "15                                           nucl: 14   \n",
       "16        nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17        nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18                 nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                                  nucl: 13, chlo: 1   \n",
       "20                          nucl: 8, cyto: 5, plas: 1   \n",
       "21                         extr: 11, golg: 2, E.R.: 1   \n",
       "22  nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...   \n",
       "23       nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                         nucl: 13.5, cyto_nucl: 7.5   \n",
       "25                chlo: 10, vacu: 2, cyto: 1, mito: 1   \n",
       "26        plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1   \n",
       "27                          extr: 9, vacu: 3, golg: 2   \n",
       "28                nucl: 11, chlo: 1, cyto: 1, cysk: 1   \n",
       "29                                           nucl: 14   \n",
       "30                         extr: 11, vacu: 2, E.R.: 1   \n",
       "31                nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                                  nucl: 13, chlo: 1   \n",
       "33                                  nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi targetp_nonplants  \\\n",
       "0                                            extr: 26                SP   \n",
       "1                                   extr: 24, golg: 2                SP   \n",
       "2   plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...             OTHER   \n",
       "3   cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...             OTHER   \n",
       "4                                            extr: 26                SP   \n",
       "5   pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...             OTHER   \n",
       "6   plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...             OTHER   \n",
       "7   extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...             OTHER   \n",
       "8   extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...             OTHER   \n",
       "9                                            extr: 25                SP   \n",
       "10                                           extr: 27                SP   \n",
       "11       extr: 19, mito: 4, golg: 2, pero: 1, E.R.: 1                SP   \n",
       "12                plas: 21, mito: 3, E.R.: 2, pero: 1             OTHER   \n",
       "13                                           extr: 25                SP   \n",
       "14                                           extr: 25                SP   \n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5             OTHER   \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1             OTHER   \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...             OTHER   \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...             OTHER   \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2             OTHER   \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1             OTHER   \n",
       "21                                           extr: 27                SP   \n",
       "22  cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...             OTHER   \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3             OTHER   \n",
       "24                          nucl: 25, cyto_nucl: 15.5             OTHER   \n",
       "25       extr: 20, mito: 3, E.R.: 2, cyto: 1, pero: 1                SP   \n",
       "26                mito: 13, plas: 8, cyto: 5, vacu: 1             OTHER   \n",
       "27                                           extr: 25                SP   \n",
       "28                extr: 24, nucl: 1, mito: 1, cyto: 1                SP   \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...             OTHER   \n",
       "30                                           extr: 27                SP   \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3             OTHER   \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...             OTHER   \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...             OTHER   \n",
       "\n",
       "   targetp_plants  \n",
       "0              SP  \n",
       "1              SP  \n",
       "2           OTHER  \n",
       "3           OTHER  \n",
       "4              SP  \n",
       "5              SP  \n",
       "6           OTHER  \n",
       "7           OTHER  \n",
       "8           OTHER  \n",
       "9              SP  \n",
       "10             SP  \n",
       "11             SP  \n",
       "12          OTHER  \n",
       "13             SP  \n",
       "14             SP  \n",
       "15          OTHER  \n",
       "16          OTHER  \n",
       "17          OTHER  \n",
       "18          OTHER  \n",
       "19          OTHER  \n",
       "20          OTHER  \n",
       "21             SP  \n",
       "22          OTHER  \n",
       "23          OTHER  \n",
       "24          OTHER  \n",
       "25             SP  \n",
       "26          OTHER  \n",
       "27             SP  \n",
       "28             SP  \n",
       "29          OTHER  \n",
       "30             SP  \n",
       "31          OTHER  \n",
       "32          OTHER  \n",
       "33          OTHER  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates = pd.concat([df_wolf, df_targetp.drop(columns=[\"protein_id\"])], axis=1)\n",
    "candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ed0f99d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num candidates: 24\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>wolf_animals</th>\n",
       "      <th>wolf_plants</th>\n",
       "      <th>wolf_fungi</th>\n",
       "      <th>targetp_nonplants</th>\n",
       "      <th>targetp_plants</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1285.t1</td>\n",
       "      <td>extr: 25, plas: 5, mito: 1, lyso: 1</td>\n",
       "      <td>vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1</td>\n",
       "      <td>extr: 24, golg: 2</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g2203.t1</td>\n",
       "      <td>plas: 29, nucl: 2, golg: 1</td>\n",
       "      <td>cyto: 5, nucl: 4, E.R.: 3, vacu: 2</td>\n",
       "      <td>plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g3428.t1</td>\n",
       "      <td>mito: 18, cyto: 11, extr: 2, nucl: 1</td>\n",
       "      <td>chlo: 6, mito: 6, nucl: 1, cyto: 1</td>\n",
       "      <td>cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g3679.t1</td>\n",
       "      <td>extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1</td>\n",
       "      <td>chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1</td>\n",
       "      <td>extr: 26</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>g4106.t1</td>\n",
       "      <td>E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...</td>\n",
       "      <td>chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1</td>\n",
       "      <td>pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>g4970.t1</td>\n",
       "      <td>plas: 32</td>\n",
       "      <td>nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...</td>\n",
       "      <td>plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>g5237.t1</td>\n",
       "      <td>plas: 24, mito: 8</td>\n",
       "      <td>chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1</td>\n",
       "      <td>extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>g5443.t1</td>\n",
       "      <td>extr: 28, nucl: 3, cyto: 1</td>\n",
       "      <td>chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1</td>\n",
       "      <td>extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>g5510.t1</td>\n",
       "      <td>plas: 23, mito: 7, E.R.: 1, golg: 1</td>\n",
       "      <td>chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...</td>\n",
       "      <td>plas: 21, mito: 3, E.R.: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>g5927.t1</td>\n",
       "      <td>nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 22.5, cyto_nucl: 14, cyto: 4.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>g7861.t1</td>\n",
       "      <td>nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...</td>\n",
       "      <td>nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1</td>\n",
       "      <td>nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>g8100.t1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...</td>\n",
       "      <td>nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2</td>\n",
       "      <td>cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>g8312.t1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...</td>\n",
       "      <td>nucl: 7, cyto: 5, chlo: 1, vacu: 1</td>\n",
       "      <td>nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>g10513.t1</td>\n",
       "      <td>nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>g10514.t1</td>\n",
       "      <td>nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...</td>\n",
       "      <td>nucl: 8, cyto: 5, plas: 1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>g11513.t1</td>\n",
       "      <td>cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...</td>\n",
       "      <td>nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...</td>\n",
       "      <td>cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>g11806.t1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...</td>\n",
       "      <td>nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1</td>\n",
       "      <td>nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>g11960.t1</td>\n",
       "      <td>nucl: 32</td>\n",
       "      <td>nucl: 13.5, cyto_nucl: 7.5</td>\n",
       "      <td>nucl: 25, cyto_nucl: 15.5</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>g12510.t1</td>\n",
       "      <td>plas: 29, cyto: 3</td>\n",
       "      <td>plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1</td>\n",
       "      <td>mito: 13, plas: 8, cyto: 5, vacu: 1</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>g13530.t1</td>\n",
       "      <td>extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...</td>\n",
       "      <td>nucl: 11, chlo: 1, cyto: 1, cysk: 1</td>\n",
       "      <td>extr: 24, nucl: 1, mito: 1, cyto: 1</td>\n",
       "      <td>SP</td>\n",
       "      <td>SP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>g14472.t1</td>\n",
       "      <td>nucl: 28, plas: 2, cyto: 1, cysk: 1</td>\n",
       "      <td>nucl: 14</td>\n",
       "      <td>nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>g15484.t1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...</td>\n",
       "      <td>nucl: 11, plas: 1, vacu: 1, golg: 1</td>\n",
       "      <td>nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>g16318.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>g16368.t1</td>\n",
       "      <td>nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...</td>\n",
       "      <td>nucl: 13, chlo: 1</td>\n",
       "      <td>nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>OTHER</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   protein_id                                       wolf_animals  \\\n",
       "1    g1285.t1                extr: 25, plas: 5, mito: 1, lyso: 1   \n",
       "2    g2203.t1                         plas: 29, nucl: 2, golg: 1   \n",
       "3    g3428.t1               mito: 18, cyto: 11, extr: 2, nucl: 1   \n",
       "4    g3679.t1       extr: 26, mito: 2, lyso: 2, plas: 1, E.R.: 1   \n",
       "5    g4106.t1  E.R.: 14.5, E.R._golg: 9.5, extr: 7, golg: 3.5...   \n",
       "6    g4970.t1                                           plas: 32   \n",
       "7    g5237.t1                                  plas: 24, mito: 8   \n",
       "8    g5443.t1                         extr: 28, nucl: 3, cyto: 1   \n",
       "12   g5510.t1                plas: 23, mito: 7, E.R.: 1, golg: 1   \n",
       "15   g5927.t1             nucl: 30.5, cyto_nucl: 16.5, cyto: 1.5   \n",
       "16   g7861.t1  nucl: 16, cyto_nucl: 14, cyto: 8, plas: 5, per...   \n",
       "17   g8100.t1  nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, plas: ...   \n",
       "18   g8312.t1  nucl: 15.5, cyto_nucl: 15.5, cyto: 12.5, mito:...   \n",
       "19  g10513.t1  nucl: 20, cyto_nucl: 14.5, cyto: 7, extr: 3, E...   \n",
       "20  g10514.t1  nucl: 19, cyto_nucl: 15, cyto: 9, extr: 3, mit...   \n",
       "22  g11513.t1  cyto: 17, cyto_nucl: 12.8333, cyto_mito: 9.833...   \n",
       "23  g11806.t1  nucl: 18, cyto_nucl: 11.8333, mito: 5, extr: 4...   \n",
       "24  g11960.t1                                           nucl: 32   \n",
       "26  g12510.t1                                  plas: 29, cyto: 3   \n",
       "28  g13530.t1  extr: 13, nucl: 6.5, lyso: 5, cyto_nucl: 4.5, ...   \n",
       "29  g14472.t1                nucl: 28, plas: 2, cyto: 1, cysk: 1   \n",
       "31  g15484.t1  nucl: 17.5, cyto_nucl: 15.3333, cyto: 12, cyto...   \n",
       "32  g16318.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "33  g16368.t1  nucl: 20.5, cyto_nucl: 13, extr: 5, cyto: 4.5,...   \n",
       "\n",
       "                                          wolf_plants  \\\n",
       "1         vacu: 5, chlo: 3, extr: 3, golg: 2, nucl: 1   \n",
       "2                  cyto: 5, nucl: 4, E.R.: 3, vacu: 2   \n",
       "3                  chlo: 6, mito: 6, nucl: 1, cyto: 1   \n",
       "4         chlo: 6, nucl: 3, mito: 2, vacu: 2, extr: 1   \n",
       "5         chlo: 6, E.R.: 3, plas: 2, pero: 2, vacu: 1   \n",
       "6   nucl: 4, cyto: 3, E.R.: 3, vacu: 2, mito: 1, p...   \n",
       "7         chlo: 5, golg: 4, nucl: 2, extr: 2, plas: 1   \n",
       "8         chlo: 9, nucl: 2, cyto: 1, extr: 1, E.R.: 1   \n",
       "12  chlo: 5, plas: 3.5, cyto_plas: 2.5, E.R.: 2, m...   \n",
       "15                                           nucl: 14   \n",
       "16        nucl: 4, vacu: 4, cyto: 3, E.R.: 2, chlo: 1   \n",
       "17        nucl: 4, E.R.: 4, cyto: 2, mito: 2, vacu: 2   \n",
       "18                 nucl: 7, cyto: 5, chlo: 1, vacu: 1   \n",
       "19                                  nucl: 13, chlo: 1   \n",
       "20                          nucl: 8, cyto: 5, plas: 1   \n",
       "22  nucl: 4, plas: 3, cyto: 2, E.R.: 2, chlo: 1, v...   \n",
       "23       nucl: 11.5, cyto_nucl: 6.5, plas: 1, extr: 1   \n",
       "24                         nucl: 13.5, cyto_nucl: 7.5   \n",
       "26        plas: 5, vacu: 3, E.R.: 3, cyto: 2, golg: 1   \n",
       "28                nucl: 11, chlo: 1, cyto: 1, cysk: 1   \n",
       "29                                           nucl: 14   \n",
       "31                nucl: 11, plas: 1, vacu: 1, golg: 1   \n",
       "32                                  nucl: 13, chlo: 1   \n",
       "33                                  nucl: 13, chlo: 1   \n",
       "\n",
       "                                           wolf_fungi targetp_nonplants  \\\n",
       "1                                   extr: 24, golg: 2                SP   \n",
       "2   plas: 9, cyto: 8.5, cyto_nucl: 5, mito: 4, ext...             OTHER   \n",
       "3   cyto: 12.5, cyto_nucl: 11, nucl: 8.5, mito: 5,...             OTHER   \n",
       "4                                            extr: 26                SP   \n",
       "5   pero: 10, cyto: 5.5, E.R.: 5, cyto_nucl: 3.5, ...             OTHER   \n",
       "6   plas: 16, cyto: 3, nucl: 2, mito: 2, pero: 2, ...             OTHER   \n",
       "7   extr: 10, nucl: 4, plas: 4, mito: 3, cyto: 3, ...             OTHER   \n",
       "8   extr: 17, nucl: 5, cyto: 2, mito: 1, E.R.: 1, ...             OTHER   \n",
       "12                plas: 21, mito: 3, E.R.: 2, pero: 1             OTHER   \n",
       "15               nucl: 22.5, cyto_nucl: 14, cyto: 4.5             OTHER   \n",
       "16       nucl: 11, plas: 9, cyto: 4, mito: 2, pero: 1             OTHER   \n",
       "17  cyto: 11.5, cyto_nucl: 8.5, mito: 5, nucl: 4.5...             OTHER   \n",
       "18  nucl: 15.5, cyto_nucl: 12, cyto: 5.5, pero: 5,...             OTHER   \n",
       "19      nucl: 18.5, cyto_nucl: 15, cyto: 6.5, mito: 2             OTHER   \n",
       "20    nucl: 20.5, cyto_nucl: 14.5, cyto: 5.5, mito: 1             OTHER   \n",
       "22  cyto: 13, cyto_nucl: 9.5, mito: 6, nucl: 4, pe...             OTHER   \n",
       "23          nucl: 18, cyto_nucl: 14, cyto: 6, mito: 3             OTHER   \n",
       "24                          nucl: 25, cyto_nucl: 15.5             OTHER   \n",
       "26                mito: 13, plas: 8, cyto: 5, vacu: 1             OTHER   \n",
       "28                extr: 24, nucl: 1, mito: 1, cyto: 1                SP   \n",
       "29  nucl: 14, cyto_nucl: 11.5, mito: 5, cyto: 5, e...             OTHER   \n",
       "31    nucl: 16.5, cyto_nucl: 12.5, cyto: 7.5, pero: 3             OTHER   \n",
       "32  nucl: 17, cyto_nucl: 14, cyto: 7, mito: 1, ext...             OTHER   \n",
       "33  nucl: 17.5, cyto_nucl: 13.5, cyto: 6.5, mito: ...             OTHER   \n",
       "\n",
       "   targetp_plants  \n",
       "1              SP  \n",
       "2           OTHER  \n",
       "3           OTHER  \n",
       "4              SP  \n",
       "5              SP  \n",
       "6           OTHER  \n",
       "7           OTHER  \n",
       "8           OTHER  \n",
       "12          OTHER  \n",
       "15          OTHER  \n",
       "16          OTHER  \n",
       "17          OTHER  \n",
       "18          OTHER  \n",
       "19          OTHER  \n",
       "20          OTHER  \n",
       "22          OTHER  \n",
       "23          OTHER  \n",
       "24          OTHER  \n",
       "26          OTHER  \n",
       "28             SP  \n",
       "29          OTHER  \n",
       "31          OTHER  \n",
       "32          OTHER  \n",
       "33          OTHER  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "candidates = candidates[(candidates[\"wolf_animals\"].str.find(\"nucl\") != -1) | \n",
    "                        (candidates[\"wolf_plants\"].str.find(\"nucl\") != -1) |\n",
    "                        (candidates[\"wolf_fungi\"].str.find(\"nucl\") != -1) |\n",
    "                        (candidates[\"targetp_nonplants\"] == \"OTHER\") |\n",
    "                        (candidates[\"targetp_plants\"] == \"OTHER\")\n",
    "               ]\n",
    "print(f\"Num candidates: {len(candidates)}\")\n",
    "candidates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "64490096",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"proteins/filtered_by_localization.fa\", 'w') as f_out:\n",
    "    for protein_id in candidates[\"protein_id\"]:\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78a50040",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e608feb4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6ab2f08",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
