{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e8fb75c1",
   "metadata": {},
   "source": [
    "# Project 4. \"Tardigrades: from genestealers to space marines\". Eukaryotic genome analysis.\n",
    "\n",
    "Lab notebook by Nikita Vyatkin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "984c1d9f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def show_file_content(path, lines=10, headers_only=False):\n",
    "    if headers_only:\n",
    "        ! mkdir -p temp\n",
    "        ! touch temp/show_file_content.tmp\n",
    "        ! grep '>' {path} > temp/show_file_content.tmp\n",
    "        path = \"temp/show_file_content.tmp\"\n",
    "    \n",
    "    ! head -{lines} {path}\n",
    "    print(\"...\")\n",
    "    ! tail -{lines} {path}\n",
    "    \n",
    "    if headers_only:\n",
    "        ! rm {path}\n",
    "\n",
    "\n",
    "def count_seqs(path):\n",
    "    res = ! grep '>' {path} | wc -l\n",
    "    print(f\"Total: {res[0]} sequencies\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "acbd5772",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: grep [OPTION]... PATTERNS [FILE]...\r\n",
      "Search for PATTERNS in each FILE.\r\n",
      "Example: grep -i 'hello world' menu.h main.c\r\n",
      "PATTERNS can contain multiple patterns separated by newlines.\r\n",
      "\r\n",
      "Pattern selection and interpretation:\r\n",
      "  -E, --extended-regexp     PATTERNS are extended regular expressions\r\n",
      "  -F, --fixed-strings       PATTERNS are strings\r\n",
      "  -G, --basic-regexp        PATTERNS are basic regular expressions\r\n",
      "  -P, --perl-regexp         PATTERNS are Perl regular expressions\r\n",
      "  -e, --regexp=PATTERNS     use PATTERNS for matching\r\n",
      "  -f, --file=FILE           take PATTERNS from FILE\r\n",
      "  -i, --ignore-case         ignore case distinctions in patterns and data\r\n",
      "      --no-ignore-case      do not ignore case distinctions (default)\r\n",
      "  -w, --word-regexp         match only whole words\r\n",
      "  -x, --line-regexp         match only whole lines\r\n",
      "  -z, --null-data           a data line ends in 0 byte, not newline\r\n",
      "\r\n",
      "Miscellaneous:\r\n",
      "  -s, --no-messages         suppress error messages\r\n",
      "  -v, --invert-match        select non-matching lines\r\n",
      "  -V, --version             display version information and exit\r\n",
      "      --help                display this help text and exit\r\n",
      "\r\n",
      "Output control:\r\n",
      "  -m, --max-count=NUM       stop after NUM selected lines\r\n",
      "  -b, --byte-offset         print the byte offset with output lines\r\n",
      "  -n, --line-number         print line number with output lines\r\n",
      "      --line-buffered       flush output on every line\r\n",
      "  -H, --with-filename       print file name with output lines\r\n",
      "  -h, --no-filename         suppress the file name prefix on output\r\n",
      "      --label=LABEL         use LABEL as the standard input file name prefix\r\n",
      "  -o, --only-matching       show only nonempty parts of lines that match\r\n",
      "  -q, --quiet, --silent     suppress all normal output\r\n",
      "      --binary-files=TYPE   assume that binary files are TYPE;\r\n",
      "                            TYPE is 'binary', 'text', or 'without-match'\r\n",
      "  -a, --text                equivalent to --binary-files=text\r\n",
      "  -I                        equivalent to --binary-files=without-match\r\n",
      "  -d, --directories=ACTION  how to handle directories;\r\n",
      "                            ACTION is 'read', 'recurse', or 'skip'\r\n",
      "  -D, --devices=ACTION      how to handle devices, FIFOs and sockets;\r\n",
      "                            ACTION is 'read' or 'skip'\r\n",
      "  -r, --recursive           like --directories=recurse\r\n",
      "  -R, --dereference-recursive  likewise, but follow all symlinks\r\n",
      "      --include=GLOB        search only files that match GLOB (a file pattern)\r\n",
      "      --exclude=GLOB        skip files that match GLOB\r\n",
      "      --exclude-from=FILE   skip files that match any file pattern from FILE\r\n",
      "      --exclude-dir=GLOB    skip directories that match GLOB\r\n",
      "  -L, --files-without-match  print only names of FILEs with no selected lines\r\n",
      "  -l, --files-with-matches  print only names of FILEs with selected lines\r\n",
      "  -c, --count               print only a count of selected lines per FILE\r\n",
      "  -T, --initial-tab         make tabs line up (if needed)\r\n",
      "  -Z, --null                print 0 byte after FILE name\r\n",
      "\r\n",
      "Context control:\r\n",
      "  -B, --before-context=NUM  print NUM lines of leading context\r\n",
      "  -A, --after-context=NUM   print NUM lines of trailing context\r\n",
      "  -C, --context=NUM         print NUM lines of output context\r\n",
      "  -NUM                      same as --context=NUM\r\n",
      "      --color[=WHEN],\r\n",
      "      --colour[=WHEN]       use markers to highlight the matching strings;\r\n",
      "                            WHEN is 'always', 'never', or 'auto'\r\n",
      "  -U, --binary              do not strip CR characters at EOL (MSDOS/Windows)\r\n",
      "\r\n",
      "When FILE is '-', read standard input.  With no FILE, read '.' if\r\n",
      "recursive, '-' otherwise.  With fewer than two FILEs, assume -h.\r\n",
      "Exit status is 0 if any line (or file if -L) is selected, 1 otherwise;\r\n",
      "if any error occurs and -q is not given, the exit status is 2.\r\n",
      "\r\n",
      "Report bugs to: bug-grep@gnu.org\r\n",
      "GNU grep home page: <http://www.gnu.org/software/grep/>\r\n",
      "General help using GNU software: <https://www.gnu.org/gethelp/>\r\n"
     ]
    }
   ],
   "source": [
    "! grep --help"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "668b748b",
   "metadata": {},
   "source": [
    "## Data source\n",
    "\n",
    "For this project we will be using a sequence of the _Ramazzottius varieornatus_, the YOKOZUNA-1 strain (sequenced in the University of Tokyo and named after the highest rank in professional sumo).\n",
    "\n",
    "One can download assembled genome [here](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b0b32a6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-12-05 18:12:32--  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.7, 165.112.9.230, 2607:f220:41f:250::229, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.7|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 17621408 (17M) [application/x-gzip]\n",
      "Saving to: ‘data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz’\n",
      "\n",
      "GCA_001949185.1_Rva 100%[===================>]  16,80M   878KB/s    in 26s     \n",
      "\n",
      "2022-12-05 18:13:00 (674 KB/s) - ‘data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz’ saved [17621408/17621408]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "! wget -P data/ https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/001/949/185/GCA_001949185.1_Rvar_4.0/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
    "! gunzip data/GCA_001949185.1_Rvar_4.0_genomic.fna.gz\n",
    "! mv data/GCA_001949185.1_Rvar_4.0_genomic.fna data/Ramazzottius_varieornatus.fna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a7a139c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 54M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n"
     ]
    }
   ],
   "source": [
    "! ls -lh data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "4190903f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BDGG01000001.1 Ramazzottius varieornatus DNA, contig: Scaffold001, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      "TGCTGCGACATACTCGAAACACCCAAGGTCTTACCGGGAGAATATCGTCGTCTGGAGGATTTGTATGGGCCATTGTGATG\n",
      "GCCGCTACCTGGCATTGTACAACTTGAATGGTTGGTCTGAGCGGGACATGATCGACAGGTGGGCTTTGATGGTGACACTT\n",
      "TTGGTGCCGCCGGCATTCCTTCTGCCCGCACGTGACATCGACAACCTCTTGACTGCCGGCCATGAAACCTCACCCCAAAC\n",
      "AGGTATTCAGTGGGCAGAAAAAGCTACAAAAATCCTATCGCACGGGAATAAGTCCGTGCTAAACTAGTTAGAGCGACTTT\n",
      "TGAAAGGAAGATGTAGAGTTTAGTTCATTGGAGTAATCAAGTCTTCGGTTAgtttcgtgacgtcacgattGATCCGTATG\n",
      "GTGAAGCGTCTGAGCACAGAGTTTTCTTATTTTTCAACAGGTGGGAGCGACGTCATGAGCGAGTATGTGACGCCCTTCTG\n",
      "TTACGTCTGTGATTCTCTCGTGGATGCCGACTGCCGAGAATCGCAAGACTGGATGCCTCGACGAAGAAACGGCttaagga\n",
      "agaaagtgatgTGGACAGGCCAAGGGGAGGACGATCATGGCAGACTTGCCTTCGACTGGAAAGAAGTCTACATtcggctg\n",
      "gaaaagttcaaGCAGCGATGTCCGTCCAACAATGGCTGCACCACTCACGTCTTCCTCAATAGCGGTACGTATCTTGCTGT\n",
      "...\n",
      "TCTTCTTTTAAGACAAGCCGGCTTGTCTTATTTAATATTAATCTTTATTTTTCTTTCTTCTTTTATAGCTATATTCATTA\n",
      "ATATAAAACAAAATTTATTTAAAAGAATGTTGACAGTTTCAAGATTGTCCAACACCAGATGACTCTTGACCGCAGTTGTA\n",
      "ACAACATATTCTGGATGAATATTTTATTTTGCAGTGTACTCATTTTCTTTAGTATTAAGTTCAACAATAAATAATAAGTA\n",
      "TTTAGACCAAAATTACCGTAATAGAGACACGCTAGAAAACCCCCTTTTATTTTTAAGCATAGGAGGACTCCCCCCCATAC\n",
      "CAGGATTTTTCCCTAAAATTTTAATTCTGATAAATTTAAGATATTCTCGGTTAAGATTGCTAAGAATTTTTCTCCTTCTT\n",
      "GCCGCGTGTGTAGATATTTTTATTTATTTACGATTTTCCTTCACAAGAGCATTGAAAAATAATGGTAATCTTATATATTG\n",
      "TAAAGCTGTAACAACAAAGAGAATTAAAATACTGTCTTTAAGAGCAGTCACTATAGTCACAATAATAGCAATTATATTAT\n",
      "AAAAAGATTTTAAGTTAAAAAAACTAATGTGTTTCAAACACTTCAATAATGAAAATTTAAATCTTAATTTTAAAGAAAAA\n",
      "CTTTTTAAATATGCTTGCAACATATCATATAAATAAAATTTCATAACAAAAAGAAAAAATCTATATGTAAATTTACAATT\n",
      "TACCGCCTATACTCAGCCATTTTGACTA\n",
      "Total: 200 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/Ramazzottius_varieornatus.fna\")\n",
    "count_seqs(\"data/Ramazzottius_varieornatus.fna\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "3536e905",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "touch: cannot touch 'temp/show_file_content.tmp': No such file or directory\n",
      "/bin/bash: temp/show_file_content.tmp: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "path = \"data/Ramazzottius_varieornatus.fna\"\n",
    "! touch temp/show_file_content.tmp\n",
    "! grep '>' {path} > temp/show_file_content.tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "8ab4773f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BDGG01000001.1 Ramazzottius varieornatus DNA, contig: Scaffold001, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000002.1 Ramazzottius varieornatus DNA, contig: Scaffold002, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000003.1 Ramazzottius varieornatus DNA, contig: Scaffold003, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      "...\n",
      ">BDGG01000198.1 Ramazzottius varieornatus DNA, contig: Scaffold277, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">BDGG01000199.1 Ramazzottius varieornatus DNA, contig: Scaffold279, strain: YOKOZUNA-1, whole genome shotgun sequence\n",
      ">AP017609.1 Ramazzottius varieornatus mitochondrial DNA, complete genome, strain: YOKOZUNA-1\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/Ramazzottius_varieornatus.fna\", lines=3, headers_only=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "4eaa153c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total: 200 sequencies\n"
     ]
    }
   ],
   "source": [
    "count_seqs(\"data/Ramazzottius_varieornatus.fna\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b577d86",
   "metadata": {},
   "source": [
    "## Functional annotation\n",
    "\n",
    "One can download AUGUSTUS results [here](https://drive.google.com/file/d/1wBxf6cDgu22NbjAOgTe-8b3Zx60hNKY0/view?usp=drive_web).\n",
    "\n",
    "[Script](http://augustus.gobics.de/binaries/scripts/getAnnoFasta.pl) for GFF to FASTA converting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "8f182b8e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".:\r\n",
      "total 48K\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  5 18:27 \u001b[0m\u001b[01;34mdata\u001b[0m/\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 6,6K дек  5 18:27 getAnnoFasta.pl\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  33K дек  5 18:33 Lab_Notebook.ipynb\r\n",
      "\r\n",
      "./data:\r\n",
      "total 81M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 27M дек  5 18:21 augustus.whole.gff\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n"
     ]
    }
   ],
   "source": [
    "ls -lhR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "a0a0ebbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "! chmod +x getAnnoFasta.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "8e739bc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "97cda1bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "! ./getAnnoFasta.pl data/augustus.whole.gff\n",
    "! mv data/augustus.whole.aa annotation/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "497d3541",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">g1.t1\n",
      "AATYSKHPRSYRENIVVWRICMGHCDGRYLALYNLNGWSERDMIDRWALMVTLLVPPAFLLPARDIDNLLTAGHETSPQTGGSDVMSEYVTPFCYVCDSL\n",
      "VDADCRESQDWMPRRRNGLRKKVMWTGQGEDDHGRLAFDWKEVYIRLEKFKQRCPSNNGCTTHVFLNSDRVIRRCFDPKEPPCSGHEWMVNRRNPKKRSK\n",
      "VKVVSRLTCADWPFCNVDSAYYWPNGTVEALLSPHDDPQLKVRSAKLSCYECDSILQPDCTGQNFVKAVRKPNDAVAHSVDWVGLRKGMEKFRTRCPVES\n",
      "GCFFQGDHMTNRVIRGCRNMTTDLPRHLLVKDRRKESSVFRTEEVCRESECNSRVLFFSHSGAFLDSDWHFNNSSAGRDNGQGEAKKEVYQGDAPATEET\n",
      "GFLPHVKQSGQGKRKEGRLQKRTIGDVEREERRADEARQRRGKPKAEGRGGAGTRVLSFGHRSRKTSAGGKLDSCWLLLLLPFFLL\n",
      ">g2.t1\n",
      "MTFAVTVATTSQSCAIAFDVFWLTTIKLFFIAKYFGSEFTPFDKVQSLTCPHCAQQGIEYKDLLPHLETNHADTSDFLVVSDNITIIWDVAHCI\n",
      ">g3.t1\n",
      "MLDKAKLVLLNKGGRPSIRSAEDGWKVFSDSLLCRASSDGRVFIGKGGGNGGAVAKAEVAAIGVVLVHHMLQQVVRHGEVVIVSANTNWTPLPY\n",
      "...\n",
      "FPHCLGYVDGSLIGLQEAPSMCGEDYYTRKGSYAINAMIVCDDLKRIRHVLAGFPGCSHDHRVFMYSKLALNPFKYFSGHEYLLADSAYPVSDIIVPTFK\n",
      "KPLANSH\n",
      ">g16433.t1\n",
      "MHLDADIDESSKEEKPEIVMEYNRTKAGVDTLDQLTGNYSCRRKTSRWPMALFYDILDISTLDAYIIWCEINPGWNSTLPTKRRMFLQDVSKKMMQRQLL\n",
      "RRSTTPSISFSMKKFVSTVVPRVQVSSDQAVDEMANTSKPKALQKKPTR\n",
      ">g16434.t1\n",
      "RIHLLADCHLLKTLNLLQRDREVFPKVDLKSGVIWKVAFFSQALAAMERSVSQKNFGSPVLV\n",
      ">g16435.t1\n",
      "MKRPGGQKYHPWSDETKLKSGDVGGANHPWADEAKLNQHPKCAVLSSKLSNVTSMMKFLATGPTGECMSESRYTAFLAGQKNSLGVATKVAGGITSPSQL\n",
      "CLCKPLLYEEFLTFSNSMVMCETGLDDTCLTRKFK\n",
      "Total: 16435 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"annotation/augustus.whole.aa\")\n",
    "count_seqs(\"annotation/augustus.whole.aa\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3d4968d",
   "metadata": {},
   "source": [
    "## Physical localization\n",
    "\n",
    "Considering that DNA is a major target of radiation damage, we may hypothesize that tardigrades might have unique proteins associated with their DNA to protect and/or effectively repair it. To explore this possibility, we will combine genomic and proteomic data.\n",
    "\n",
    "[List](https://disk.yandex.ru/d/xJqQMGX77Xueqg) of peptides that were associated with the Tardigrade's DNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8a8964f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".:\r\n",
      "total 48K\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  5 18:41 \u001b[0m\u001b[01;34mannotation\u001b[0m/\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 16:25 \u001b[01;34mdata\u001b[0m/\r\n",
      "-rwxrwxr-x 1 nikitos nikitos 6,6K дек  5 18:27 \u001b[01;32mgetAnnoFasta.pl\u001b[0m*\r\n",
      "-rw-rw-r-- 1 nikitos nikitos  21K дек  8 15:39 Lab_Notebook.ipynb\r\n",
      "drwxrwxr-x 2 nikitos nikitos 4,0K дек  7 17:10 \u001b[01;34mtemp\u001b[0m/\r\n",
      "-rw-rw-r-- 1 nikitos nikitos   76 дек  7 17:23 temp.txt\r\n",
      "\r\n",
      "./annotation:\r\n",
      "total 7,0M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 7,0M дек  5 18:41 augustus.whole.aa\r\n",
      "\r\n",
      "./data:\r\n",
      "total 81M\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 27M дек  5 18:21 augustus.whole.gff\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 737 дек  7 16:22 dna_associated_peptides.fa\r\n",
      "-rw-rw-r-- 1 nikitos nikitos 54M янв 12  2017 Ramazzottius_varieornatus.fna\r\n",
      "\r\n",
      "./temp:\r\n",
      "total 0\r\n"
     ]
    }
   ],
   "source": [
    "ls -lhR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bd887bff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">1\n",
      "VFSALDVLR\n",
      ">2\n",
      "SPLQTDEIR\n",
      ">3\n",
      "AATGAVLSSADSR\n",
      "...\n",
      ">41\n",
      "GPEDEAGGPPK\n",
      ">42\n",
      "SSGGNSPDVVVR\n",
      ">43\n",
      "STVGQTPPQNLQR\n",
      "Total: 43 sequencies\n"
     ]
    }
   ],
   "source": [
    "show_file_content(\"data/dna_associated_peptides.fa\", lines=6)\n",
    "count_seqs(\"data/dna_associated_peptides.fa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9a3434e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MYPLQEAYNRYKKSIKIKEVKHFGLPIIFVVGKPGCGKASMCHGLERLHDFSYLSIGQMLRDEASTNTKRGKKIAEFVYHQPFSKPVPQELVLELLREVLGRFLYIHKPIIIQGFPRTYAQAAQLEKEHMVYLTISNEKAEERYLDKIKYVEHLPDQTEHIRRASEAFDKMTVPVIDGLKKENRVVEVDAEQGSEKVLQEAVKIYEKWLLRSKQGIDFLENYKNLAGVQSYENTRRSI'"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins = []\n",
    "\n",
    "with open(\"annotation/augustus.whole.aa\", 'r') as f_in:\n",
    "    for line in f_in:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            proteins.append([line[1:], []])\n",
    "        elif line:\n",
    "            proteins[-1][1].append(line)\n",
    "\n",
    "    for protein in proteins:\n",
    "        protein[1] = ''.join(protein[1])\n",
    "\n",
    "proteins = dict(proteins)\n",
    "proteins[\"g24.t1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "e2e84d05",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'VFSALDVLR'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides = []\n",
    "\n",
    "with open(\"data/dna_associated_peptides.fa\", 'r') as f_in:\n",
    "    for line in f_in:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            peptides.append([line[1:], []])\n",
    "        elif line:\n",
    "            peptides[-1][1].append(line)\n",
    "\n",
    "    for peptide in peptides:\n",
    "        peptide[1] = ''.join(peptide[1])\n",
    "\n",
    "peptides = dict(peptides)\n",
    "peptides[\"1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "f097cbcc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('GPEDEAGGPPK', 'GPEDEAGGPPK')"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides[\"10\"], peptides[\"41\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "dec3c80a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(peptides.values()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "5fad3621",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_vals = set(peptides.values())\n",
    "new_peptides = dict()\n",
    "\n",
    "for key, val in peptides.items():\n",
    "    if val in unique_vals:\n",
    "        new_peptides[key] = val\n",
    "        unique_vals -= {val}\n",
    "\n",
    "peptides = new_peptides\n",
    "len(peptides)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "599747c3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-12-08 16:12:47--  http://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
      "Resolving github.com (github.com)... 140.82.121.4\n",
      "Connecting to github.com (github.com)|140.82.121.4|:80... connected.\n",
      "HTTP request sent, awaiting response... 301 Moved Permanently\n",
      "Location: https://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz [following]\n",
      "--2022-12-08 16:12:47--  https://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
      "Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/31987083/4420191b-492f-4ec0-a80c-080436691f1e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221208T131247Z&X-Amz-Expires=300&X-Amz-Signature=b556836ddecfca6fd02dbf98564ee0598fd7f709f66560d9597b6863008f8adb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=31987083&response-content-disposition=attachment%3B%20filename%3Ddiamond-linux64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
      "--2022-12-08 16:12:47--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/31987083/4420191b-492f-4ec0-a80c-080436691f1e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221208T131247Z&X-Amz-Expires=300&X-Amz-Signature=b556836ddecfca6fd02dbf98564ee0598fd7f709f66560d9597b6863008f8adb&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=31987083&response-content-disposition=attachment%3B%20filename%3Ddiamond-linux64.tar.gz&response-content-type=application%2Foctet-stream\n",
      "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.109.133, 185.199.108.133, ...\n",
      "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 13078237 (12M) [application/octet-stream]\n",
      "Saving to: ‘diamond-linux64.tar.gz’\n",
      "\n",
      "diamond-linux64.tar 100%[===================>]  12,47M  4,67MB/s    in 2,7s    \n",
      "\n",
      "2022-12-08 16:12:50 (4,67 MB/s) - ‘diamond-linux64.tar.gz’ saved [13078237/13078237]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# downloading the tool\n",
    "! wget http://github.com/bbuchfink/diamond/releases/download/v2.0.15/diamond-linux64.tar.gz\n",
    "! tar xzf diamond-linux64.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3911eaba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\r\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\r\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\r\n",
      "\r\n",
      "Syntax: diamond COMMAND [OPTIONS]\r\n",
      "\r\n",
      "Commands:\r\n",
      "makedb\tBuild DIAMOND database from a FASTA file\r\n",
      "prepdb\tPrepare BLAST database for use with Diamond\r\n",
      "blastp\tAlign amino acid query sequences against a protein reference database\r\n",
      "blastx\tAlign DNA query sequences against a protein reference database\r\n",
      "view\tView DIAMOND alignment archive (DAA) formatted file\r\n",
      "help\tProduce help message\r\n",
      "version\tDisplay version information\r\n",
      "getseq\tRetrieve sequences from a DIAMOND database file\r\n",
      "dbinfo\tPrint information about a DIAMOND database file\r\n",
      "test\tRun regression tests\r\n",
      "makeidx\tMake database index\r\n",
      "\r\n",
      "General options:\r\n",
      "--threads (-p)           number of CPU threads\r\n",
      "--db (-d)                database file\r\n",
      "--out (-o)               output file\r\n",
      "--outfmt (-f)            output format\r\n",
      "\t0   = BLAST pairwise\r\n",
      "\t5   = BLAST XML\r\n",
      "\t6   = BLAST tabular\r\n",
      "\t100 = DIAMOND alignment archive (DAA)\r\n",
      "\t101 = SAM\r\n",
      "\r\n",
      "\tValue 6 may be followed by a space-separated list of these keywords:\r\n",
      "\r\n",
      "\tqseqid means Query Seq - id\r\n",
      "\tqlen means Query sequence length\r\n",
      "\tsseqid means Subject Seq - id\r\n",
      "\tsallseqid means All subject Seq - id(s), separated by a ';'\r\n",
      "\tslen means Subject sequence length\r\n",
      "\tqstart means Start of alignment in query\r\n",
      "\tqend means End of alignment in query\r\n",
      "\tsstart means Start of alignment in subject\r\n",
      "\tsend means End of alignment in subject\r\n",
      "\tqseq means Aligned part of query sequence\r\n",
      "\tqseq_translated means Aligned part of query sequence (translated)\r\n",
      "\tfull_qseq means Query sequence\r\n",
      "\tfull_qseq_mate means Query sequence of the mate\r\n",
      "\tsseq means Aligned part of subject sequence\r\n",
      "\tfull_sseq means Subject sequence\r\n",
      "\tevalue means Expect value\r\n",
      "\tbitscore means Bit score\r\n",
      "\tscore means Raw score\r\n",
      "\tlength means Alignment length\r\n",
      "\tpident means Percentage of identical matches\r\n",
      "\tnident means Number of identical matches\r\n",
      "\tmismatch means Number of mismatches\r\n",
      "\tpositive means Number of positive - scoring matches\r\n",
      "\tgapopen means Number of gap openings\r\n",
      "\tgaps means Total number of gaps\r\n",
      "\tppos means Percentage of positive - scoring matches\r\n",
      "\tqframe means Query frame\r\n",
      "\tbtop means Blast traceback operations(BTOP)\r\n",
      "\tcigar means CIGAR string\r\n",
      "\tstaxids means unique Subject Taxonomy ID(s), separated by a ';' (in numerical order)\r\n",
      "\tsscinames means unique Subject Scientific Name(s), separated by a ';'\r\n",
      "\tsskingdoms means unique Subject Super Kingdom(s), separated by a ';'\r\n",
      "\tskingdoms means unique Subject Kingdom(s), separated by a ';'\r\n",
      "\tsphylums means unique Subject Phylum(s), separated by a ';'\r\n",
      "\tstitle means Subject Title\r\n",
      "\tsalltitles means All Subject Title(s), separated by a '<>'\r\n",
      "\tqcovhsp means Query Coverage Per HSP\r\n",
      "\tscovhsp means Subject Coverage Per HSP\r\n",
      "\tqtitle means Query title\r\n",
      "\tqqual means Query quality values for the aligned part of the query\r\n",
      "\tfull_qqual means Query quality values\r\n",
      "\tqstrand means Query strand\r\n",
      "\r\n",
      "\tDefault: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore\r\n",
      "--verbose (-v)           verbose console output\r\n",
      "--log                    enable debug log\r\n",
      "--quiet                  disable console output\r\n",
      "--header                 Write header lines to blast tabular format.\r\n",
      "\r\n",
      "Makedb options:\r\n",
      "--in                     input reference file in FASTA format\r\n",
      "--taxonmap               protein accession to taxid mapping file\r\n",
      "--taxonnodes             taxonomy nodes.dmp from NCBI\r\n",
      "--taxonnames             taxonomy names.dmp from NCBI\r\n",
      "\r\n",
      "Aligner options:\r\n",
      "--query (-q)             input query file\r\n",
      "--strand                 query strands to search (both/minus/plus)\r\n",
      "--un                     file for unaligned queries\r\n",
      "--al                     file or aligned queries\r\n",
      "--unfmt                  format of unaligned query file (fasta/fastq)\r\n",
      "--alfmt                  format of aligned query file (fasta/fastq)\r\n",
      "--unal                   report unaligned queries (0=no, 1=yes)\r\n",
      "--max-target-seqs (-k)   maximum number of target sequences to report alignments for (default=25)\r\n",
      "--top                    report alignments within this percentage range of top alignment score (overrides --max-target-seqs)\r\n",
      "--max-hsps               maximum number of HSPs per target sequence to report for each query (default=1)\r\n",
      "--range-culling          restrict hit culling to overlapping query ranges\r\n",
      "--compress               compression for output files (0=none, 1=gzip, zstd)\r\n",
      "--evalue (-e)            maximum e-value to report alignments (default=0.001)\r\n",
      "--min-score              minimum bit score to report alignments (overrides e-value setting)\r\n",
      "--id                     minimum identity% to report an alignment\r\n",
      "--query-cover            minimum query cover% to report an alignment\r\n",
      "--subject-cover          minimum subject cover% to report an alignment\r\n",
      "--fast                   enable fast mode\r\n",
      "--mid-sensitive          enable mid-sensitive mode\r\n",
      "--sensitive              enable sensitive mode)\r\n",
      "--more-sensitive         enable more sensitive mode\r\n",
      "--very-sensitive         enable very sensitive mode\r\n",
      "--ultra-sensitive        enable ultra sensitive mode\r\n",
      "--iterate                iterated search with increasing sensitivity\r\n",
      "--global-ranking (-g)    number of targets for global ranking\r\n",
      "--block-size (-b)        sequence block size in billions of letters (default=2.0)\r\n",
      "--index-chunks (-c)      number of chunks for index processing (default=4)\r\n",
      "--tmpdir (-t)            directory for temporary files\r\n",
      "--parallel-tmpdir        directory for temporary files used by multiprocessing\r\n",
      "--gapopen                gap open penalty\r\n",
      "--gapextend              gap extension penalty\r\n",
      "--frameshift (-F)        frame shift penalty (default=disabled)\r\n",
      "--long-reads             short for --range-culling --top 10 -F 15\r\n",
      "--matrix                 score matrix for protein alignment (default=BLOSUM62)\r\n",
      "--custom-matrix          file containing custom scoring matrix\r\n",
      "--comp-based-stats       composition based statistics mode (0-4)\r\n",
      "--masking                masking algorithm (none, seg, tantan=default)\r\n",
      "--query-gencode          genetic code to use to translate query (see user manual)\r\n",
      "--salltitles             include full subject titles in DAA file\r\n",
      "--sallseqid              include all subject ids in DAA file\r\n",
      "--no-self-hits           suppress reporting of identical self hits\r\n",
      "--taxonlist              restrict search to list of taxon ids (comma-separated)\r\n",
      "--taxon-exclude          exclude list of taxon ids (comma-separated)\r\n",
      "--seqidlist              filter the database by list of accessions\r\n",
      "--skip-missing-seqids    ignore accessions missing in the database\r\n",
      "\r\n",
      "Advanced options:\r\n",
      "--algo                   Seed search algorithm (0=double-indexed/1=query-indexed/ctg=contiguous-seed)\r\n",
      "--bin                    number of query bins for seed search\r\n",
      "--min-orf (-l)           ignore translated sequences without an open reading frame of at least this length\r\n",
      "--seed-cut               cutoff for seed complexity\r\n",
      "--freq-masking           mask seeds based on frequency\r\n",
      "--freq-sd                number of standard deviations for ignoring frequent seeds\r\n",
      "--motif-masking          softmask abundant motifs (0/1)\r\n",
      "--id2                    minimum number of identities for stage 1 hit\r\n",
      "--xdrop (-x)             xdrop for ungapped alignment\r\n",
      "--gapped-filter-evalue   E-value threshold for gapped filter (auto)\r\n",
      "--band                   band for dynamic programming computation\r\n",
      "--shapes (-s)            number of seed shapes (default=all available)\r\n",
      "--shape-mask             seed shapes\r\n",
      "--multiprocessing        enable distributed-memory parallel processing\r\n",
      "--mp-init                initialize multiprocessing run\r\n",
      "--mp-recover             enable continuation of interrupted multiprocessing run\r\n",
      "--mp-query-chunk         process only a single query chunk as specified\r\n",
      "--ext-chunk-size         chunk size for adaptive ranking (default=auto)\r\n",
      "--no-ranking             disable ranking heuristic\r\n",
      "--ext                    Extension mode (banded-fast/banded-slow/full)\r\n",
      "--culling-overlap        minimum range overlap with higher scoring hit to delete a hit (default=50%)\r\n",
      "--taxon-k                maximum number of targets to report per species\r\n",
      "--range-cover            percentage of query range to be covered for range culling (default=50%)\r\n",
      "--dbsize                 effective database size (in letters)\r\n",
      "--no-auto-append         disable auto appending of DAA and DMND file extensions\r\n",
      "--xml-blord-format       Use gnl|BL_ORD_ID| style format in XML output\r\n",
      "--stop-match-score       Set the match score of stop codons against each other.\r\n",
      "--tantan-minMaskProb     minimum repeat probability for masking (default=0.9)\r\n",
      "--file-buffer-size       file buffer size in bytes (default=67108864)\r\n",
      "--memory-limit (-M)      Memory limit for extension stage in GB\r\n",
      "--no-unlink              Do not unlink temporary files.\r\n",
      "--target-indexed         Enable target-indexed mode\r\n",
      "--ignore-warnings        Ignore warnings\r\n",
      "\r\n",
      "View options:\r\n",
      "--daa (-a)               DIAMOND alignment archive (DAA) file\r\n",
      "--forwardonly            only show alignments of forward strand\r\n",
      "\r\n",
      "Getseq options:\r\n",
      "--seq                    Space-separated list of sequence numbers to display.\r\n",
      "\r\n",
      "Online documentation at http://www.diamondsearch.org\r\n"
     ]
    }
   ],
   "source": [
    "! ./diamond help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "31cf4083",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir diamond_search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "64d11657",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\n",
      "\n",
      "#CPU threads: 12\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "Database input file: annotation/augustus.whole.aa\n",
      "Opening the database file...  [0s]\n",
      "Loading sequences...  [0.022s]\n",
      "Masking sequences...  [0.041s]\n",
      "Writing sequences...  [0.003s]\n",
      "Hashing sequences...  [0.001s]\n",
      "Loading sequences...  [0s]\n",
      "Writing trailer...  [0s]\n",
      "Closing the input file...  [0s]\n",
      "Closing the database file...  [0.001s]\n",
      "\n",
      "Database sequences  16435\n",
      "  Database letters  7015168\n",
      "     Database hash  590081acba74e419860a4dfdcf0de70e\n",
      "        Total time  0.070000s\n"
     ]
    }
   ],
   "source": [
    "# creating a diamond-formatted database file\n",
    "! ./diamond makedb --in annotation/augustus.whole.aa -d diamond_search/reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ea9462ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science\n",
      "Documentation, support and updates available at http://www.diamondsearch.org\n",
      "Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)\n",
      "\n",
      "#CPU threads: 12\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "Temporary directory: diamond_search\n",
      "#Target sequences to report alignments for: 25\n",
      "Opening the database...  [0.004s]\n",
      "Database: diamond_search/reference.dmnd (type: Diamond database, sequences: 16435, letters: 7015168)\n",
      "Block size = 2000000000\n",
      "Opening the input file...  [0s]\n",
      "Opening the output file...  [0s]\n",
      "Loading query sequences...  [0s]\n",
      "Masking queries...  [0.001s]\n",
      "Algorithm: Double-indexed\n",
      "Building query histograms...  [0s]\n",
      "Allocating buffers...  [0s]\n",
      "Loading reference sequences...  [0.006s]\n",
      "Masking reference...  [0.04s]\n",
      "Initializing temporary storage...  [0s]\n",
      "Building reference histograms...  [0.034s]\n",
      "Allocating buffers...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 1/4.\n",
      "Building reference seed array...  [0.017s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0.001s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 2/4.\n",
      "Building reference seed array...  [0.023s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 3/4.\n",
      "Building reference seed array...  [0.022s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 1/2, index chunk 4/4.\n",
      "Building reference seed array...  [0.016s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 1/4.\n",
      "Building reference seed array...  [0.019s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 2/4.\n",
      "Building reference seed array...  [0.021s]\n",
      "Building query seed array...  [0.001s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 3/4.\n",
      "Building reference seed array...  [0.021s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Processing query block 1, reference block 1/1, shape 2/2, index chunk 4/4.\n",
      "Building reference seed array...  [0.015s]\n",
      "Building query seed array...  [0s]\n",
      "Computing hash join...  [0s]\n",
      "Masking low complexity seeds...  [0s]\n",
      "Searching alignments...  [0s]\n",
      "Deallocating buffers...  [0s]\n",
      "Clearing query masking...  [0s]\n",
      "Computing alignments...  [0s]\n",
      "Deallocating reference...  [0s]\n",
      "Loading reference sequences...  [0s]\n",
      "Deallocating buffers...  [0s]\n",
      "Deallocating queries...  [0s]\n",
      "Loading query sequences...  [0s]\n",
      "Closing the input file...  [0s]\n",
      "Closing the output file...  [0s]\n",
      "Cleaning up...  [0s]\n",
      "Total time = 0.318s\n",
      "Reported 4 pairwise alignments, 4 HSPs.\n",
      "4 queries aligned.\n"
     ]
    }
   ],
   "source": [
    "# running a search in blastp mode\n",
    "! ./diamond blastp -d diamond_search/reference.dmnd -q data/dna_associated_peptides.fa -o diamond_search/matches.tsv -f 6 --iterate --header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "302a4c97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# DIAMOND v2.0.15. http://github.com/bbuchfink/diamond\r\n",
      "# Invocation: ./diamond blastp -d diamond_search/reference.dmnd -q data/dna_associated_peptides.fa -o diamond_search/matches.tsv -f 6 --iterate --header\r\n",
      "# Fields: Query ID, Subject ID, Percentage of identical matches, Alignment length, Number of mismatches, Number of gap openings, Start of alignment in query, End of alignment in query, Start of alignment in subject, End of alignment in subject, Expected value, Bit score\r\n",
      "8\tg4106.t1\t100\t18\t0\t0\t1\t18\t222\t239\t5.55e-07\t40.8\r\n",
      "20\tg12510.t1\t100\t18\t0\t0\t1\t18\t425\t442\t1.43e-06\t39.7\r\n",
      "21\tg12510.t1\t100\t22\t0\t0\t1\t22\t443\t464\t2.46e-09\t47.8\r\n",
      "29\tg4106.t1\t100\t18\t0\t0\t1\t18\t222\t239\t5.55e-07\t40.8\r\n"
     ]
    }
   ],
   "source": [
    "! cat diamond_search/matches.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "980d5f2b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MSTTTKCQHGLVFSTLALLILIVALVVFLSPITRRSGAYVTPVTRINNVTVTTPYSIDATTVLVVVQDAAIPVDTVVVTNTNADNTGVRTTSTGTTATGSTAVQYVRGSDGVVYAVPTATATAVVDGTSRDNYRGRGKRDADRSRNRDADRMRNRDADRMRNRGSGSIVVTVYPVSAYGSGYMVSGNVSSLVPAETYVLTFHQFGDMSGGCSQLGQGMRLRNIPVGGVNTEATGDNYIRHALVDSNGTIIGVANPNGNTNTGSTSGATAYVIENAVPVAVSNYQADYTNYAPSNYGYGNYGRGGNTAVLQPVSNGSNYVVVNQDGTNQTYYSGARAFGQGYNQNGLPSNVAFFIPGVDYSGNDTSYYRNDPNGVYNKLNPVSVRGTGNNDHVVYQNNTQNDESNVKFLPLQTVVNATTNTNYYNGGYVPVNYVPVSGYPTDNVVVVNPRSNVTTGYTQYLGDNYVPRERDGRRGRNKRQTGPLSAAASRRDARDYNGAVIGTFVTDENGRGNFDFYIPEESACGNPRYLYGRTVAIRPVYGVRRNKRQIVAGQPVDGVYGLDISGSDGSGTTYQTVNNTSTAAVQGVTIIRPDRLMPVDANYAGRMRDEGMRRRFQRMSKRPTMAPGMRMGQPISCGILGRA',\n",
       " 'NIPVGGVNTEATGDNYIR',\n",
       " 'NIPVGGVNTEATGDNYIR')"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g4106.t1\"], peptides[\"8\"], peptides[\"29\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c95476a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MCAIWENSLVGSTDNIVMRNVVAAEETISRLLCHLTSIDMWVSQNPSFDLRQKKARLPSQSRCIVRWVCLQASLQVVKDAQLAFVTVCFVLQLMEKKCAWNVTSRSIIMVTTTTKTLNYSIIRGNHTNSVNATITSTPGFNINYFGKIPGILKVLELVTIVVALGLAASRRREAPGLELPAADSWRIEYFLTGEVFFLCVMSAGLMWLVAILLTYMMHVVSSVITPKTTIHEVVFNFLFMLMFWTAGIVELAQTVPNSYPYNGRPGMEQGDWGCRVAAGALSLFIGTFFLISFALAVKELQGPKRSAPLSASPISARDHPIVTVVALGLAASRRREGTTTVVGWPVEYFLAGEVFFLCICAAALTYLFGQLTTFAWSVVSSVLIPKTTYFEIVINFLLFLAWATAGIVQLAMTCREAWNGREIRAPVNGIVTDANGNQIQVRTNAAGQLIGPGGIAINDAGLTRAVYRSPDMLFQDWGCRIAAGSLALFLSIFHLISFALAVKEFQGPRNEAR',\n",
       " 'APVNGIVTDANGNQIQVR',\n",
       " 'TNAAGQLIGPGGIAINDAGLTR')"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g12510.t1\"], peptides[\"20\"], peptides[\"21\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "cfd54fd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('g702.t1', '18', 174),\n",
       " ('g702.t1', '35', 157),\n",
       " ('g1285.t1', '24', 137),\n",
       " ('g3428.t1', '10', 159),\n",
       " ('g3428.t1', '11', 128),\n",
       " ('g3679.t1', '42', 20),\n",
       " ('g4106.t1', '8', 221),\n",
       " ('g5237.t1', '19', 90),\n",
       " ('g5502.t1', '24', 135),\n",
       " ('g5503.t1', '24', 135),\n",
       " ('g5510.t1', '16', 168),\n",
       " ('g5616.t1', '12', 154),\n",
       " ('g5616.t1', '13', 142),\n",
       " ('g5641.t1', '1', 24),\n",
       " ('g5641.t1', '2', 35),\n",
       " ('g5641.t1', '3', 154),\n",
       " ('g5641.t1', '4', 143),\n",
       " ('g5641.t1', '24', 127),\n",
       " ('g10513.t1', '9', 304),\n",
       " ('g10514.t1', '37', 220),\n",
       " ('g10514.t1', '38', 41),\n",
       " ('g12510.t1', '17', 305),\n",
       " ('g12510.t1', '20', 424),\n",
       " ('g12510.t1', '21', 442),\n",
       " ('g12562.t1', '24', 127),\n",
       " ('g12562.t1', '39', 160),\n",
       " ('g13530.t1', '5', 223),\n",
       " ('g13530.t1', '6', 87),\n",
       " ('g14472.t1', '7', 367),\n",
       " ('g15153.t1', '1', 24),\n",
       " ('g15153.t1', '15', 44),\n",
       " ('g15153.t1', '24', 127),\n",
       " ('g15484.t1', '32', 830),\n",
       " ('g15484.t1', '33', 790),\n",
       " ('g15484.t1', '34', 754)]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def hamming_distance(s1: str, s2: str) -> int:\n",
    "    assert len(s1) == len(s2), \"Strings have different lengths\"\n",
    "    for i in range(len(s1)):\n",
    "        if s1[i] != s2[i]:\n",
    "            return False\n",
    "    return True\n",
    "\n",
    "\n",
    "def pattern_matching(text: str, pattern: str, mismatches: int = 0) -> list[int]:\n",
    "    res = []\n",
    "    pattern_len = len(pattern)\n",
    "    for i in range(len(text) - pattern_len + 1):\n",
    "        if hamming_distance(text[i:i + pattern_len], pattern):\n",
    "            res.append(i)\n",
    "    return res\n",
    "\n",
    "matches = []\n",
    "for protein in proteins.items():\n",
    "    for peptide in peptides.items():\n",
    "        res = pattern_matching(protein[1], peptide[1])\n",
    "        for ind in res:\n",
    "            matches.append((protein[0], peptide[0], ind))\n",
    "\n",
    "matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "98076cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "35"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matches)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0d9a02b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('MKCLIVFSAFVASALAANGDQQPKVWDALNVLRGKDALFADQIQELLMNVVPGQDFPLLSEVPDTAVDCSRFKQPGFYADTEQGRCQVFHRCDINGHLTGYLCPNMTLFNQITLVCDWFYNVDCSQARQFYDYSNSRLYQEGVALLDDQNDFEVRARVVEQTGASSSRRLSGSRSSSSSSQESQSSSSQVRRRKSRKSKXXXXXXXXQSVSQEFA',\n",
       " 'SSSSSSQESQSSSSQVR',\n",
       " 'VVEQTGASSSR')"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteins[\"g702.t1\"], peptides[\"18\"], peptides[\"35\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d2d58cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['g702.t1',\n",
       " 'g1285.t1',\n",
       " 'g3428.t1',\n",
       " 'g3679.t1',\n",
       " 'g4106.t1',\n",
       " 'g5237.t1',\n",
       " 'g5502.t1',\n",
       " 'g5503.t1',\n",
       " 'g5510.t1',\n",
       " 'g5616.t1',\n",
       " 'g5641.t1',\n",
       " 'g10513.t1',\n",
       " 'g10514.t1',\n",
       " 'g12510.t1',\n",
       " 'g12562.t1',\n",
       " 'g13530.t1',\n",
       " 'g14472.t1',\n",
       " 'g15153.t1',\n",
       " 'g15484.t1']"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matched_proteins = [matches[0][0]]\n",
    "for prot, *_ in matches:\n",
    "    if matched_proteins[-1] != prot:\n",
    "        matched_proteins.append(prot)\n",
    "matched_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "0bcb96e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"matched_proteins.fa\", 'w') as f_out:\n",
    "    for protein_id in matched_proteins:\n",
    "        f_out.write('>' + protein_id + '\\n')\n",
    "        f_out.write(proteins[protein_id] + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b792e333",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e18e797",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00469688",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13603ead",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6ab2f08",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
