<!DOCTYPE html>
<html lang="en-gb"><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <title>Variant Effect Predictor results - Homo_sapiens - GRCh37 Archive browser 109</title>

<!--[if lte IE 7]><link rel="stylesheet" type="text/css" media="all" href="/minified/711e1712d7260698916043858921312e.ie7.css"/><![endif]--><!--[if gt IE 7]><link rel="stylesheet" type="text/css" media="all" href="/minified/711e1712d7260698916043858921312e.css"/><![endif]--><!--[if !IE]><!--><link rel="stylesheet" type="text/css" media="all" href="VEP_results_MikeRaiko_files/711e1712d7260698916043858921312e.css"><!--<![endif]--><!--[if lte IE 7]><link rel="stylesheet" type="text/css" media="all" href="/minified/0a9a03ef432183b962e4c70811450ba9.image.ie7.css"/><![endif]--><!--[if gt IE 7]><link rel="stylesheet" type="text/css" media="all" href="/minified/0a9a03ef432183b962e4c70811450ba9.image.css"/><![endif]--><!--[if !IE]><!--><link rel="stylesheet" type="text/css" media="all" href="VEP_results_MikeRaiko_files/0a9a03ef432183b962e4c70811450ba9.image.css"><!--<![endif]-->
  <link type="image/png" rel="icon" href="http://grch37.ensembl.org/i/ensembl-favicon.png">
  <link href="http://grch37.ensembl.org/apple-touch-icon.png" rel="apple-touch-icon" type="image/png">
  <link rel="search" href="http://grch37.ensembl.org/opensearch/all.xml" type="application/opensearchdescription+xml" title="GRCh37 (All)">
  <link type="application/opensearchdescription+xml" title="GRCh37 (H.sap)" href="http://grch37.ensembl.org/opensearch/Homo_sapiens.xml" rel="search">

  <meta name="google-site-verification" content="tG3WW0bg9gNaIHH002gJl-fV2k0bzf-ZxC8pBqXTaNo">
<meta name="viewport" content="target-densitydpi=device-dpi, width=device-width, initial-scale=1.0, maximum-scale=2.0, user-scalable=yes">
<script type="text/javascript" async="" src="VEP_results_MikeRaiko_files/linkid.js"></script><script async="" src="VEP_results_MikeRaiko_files/analytics.js"></script><script>var prefetch = ['/minified/dc70e5b4bed7893237228c404e2992b3.jpg','/minified/02a60942bb4c3f6fa98d4c07eeb89af7.png','/minified/92a5cd07513546550b5d607d385f401f.png','/minified/b07ae2d13f623431d4c5599a9ed36286.png','/minified/80781a8a7c2a789249be5426971bb266.png','/minified/c426c975e6ef844f7ebdfb248abce5af.png','/minified/2233bda0867971b025fbe642f50aef47.png'];</script>
<script defer="defer" type="text/javascript" src="VEP_results_MikeRaiko_files/c19e09172f7022adf7616ef6eb7c80a9.js"></script>
<script defer="defer" type="text/javascript" src="VEP_results_MikeRaiko_files/cfb6ab653b60030fd8a8d9952e3fa830.js"></script>

</head>
<body id="ensembl-webpage" data-pace="2" data-new-gr-c-s-check-loaded="8.904.0" data-gr-ext-installed="" class="">

  <div id="min_width_container">
    <div id="min_width_holder">
    
    <!-- Announcement Banner -->    
        
    <!-- /Announcement Banner -->

      <div id="masthead" class="js_panel bar_and_tabs">
        <input type="hidden" class="panel_type" value="Masthead">
        <div class="logo_holder"><a href="http://grch37.ensembl.org/"><div class="logo-header print_hide" title="Ensembl Home">&nbsp;</div></a><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png screen_hide_inline" style="height: 45px;background-position: -7.4px -325.125px;width: 185px;background-size: 1233.33333333333px 878.625px" alt="Ensembl Home" title="Ensembl Home"></div>
        <div class="mh print_hide">
          <div class="account_holder"><div class="_account_holder"><a class="constant account-link _accounts_no_user modal_link" href="http://grch37.ensembl.org/Multi/Account/Login" title="Login/Register">Login/Register</a></div></div>
          <div class="tools_holder"><ul class="tools" style="width: auto;"><li><a class="constant" href="http://grch37.ensembl.org/Homo_sapiens/Tools/Blast?tl=G9fJSvE7sdcvGHwt-8953987">BLAST/BLAT</a></li><li><a class="constant" href="http://grch37.ensembl.org/info/docs/tools/vep/">VEP</a></li><li><a class="constant" href="http://grch37.ensembl.org/info/docs/tools/index.html">Tools</a></li><li><a class="constant" href="http://grch37.ensembl.org/biomart/martview">BioMart</a></li><li><a class="constant" rel="nofollow" href="http://grch37.ensembl.org/info/data/">Downloads</a></li><li class="last"><a class="constant" rel="nofollow" href="http://grch37.ensembl.org/info/">Help &amp; Docs</a></li></ul><div class="more" style="display: none;"><a href="#">More <span class="arrow">▼</span></a><ul class="more_tools"></ul></div></div>
          <div class="search_holder print_hide">
    <div id="searchPanel" class="js_panel">
      <input type="hidden" class="panel_type" value="SearchBox">
      <form action="/Homo_sapiens/Psychic">
        <div class="search print_hide">
          <div class="sites button">
            <img class="search_image no-sprite badge-16" src="VEP_results_MikeRaiko_files/Homo_sapiens.png" alt="">
            <img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="height: 16px;background-position: -798px -47px;width: 7px;background-size: 1000px 781px">
            <input type="hidden" name="site" value="ensembl">
          </div>
          <div>
            <label class="hidden" for="se_q">Search terms</label>
            <div style="position: relative; display: inline-block; vertical-align: bottom; width: 256px; height: 16px; margin: 0px; padding: 0px; border-style: none; border-color: rgb(51, 51, 51); border-width: 0px; background-color: rgb(255, 255, 255);"><input class="query inactive solr_ghost" id="" type="text" name="" data-role="none" onkeydown="if (event.keyCode == 13) { $(this).closest('form').submit(); return false; }" style="padding: 0px; margin: 0px; border: medium none; outline: none; left: 0px; top: 0px; position: absolute; z-index: 1; background: none;" placeholder="" tabindex="5000"><span role="status" aria-live="polite" class="ui-helper-hidden-accessible"></span><input class="query inactive ui-autocomplete-input" id="se_q" type="text" name="q" value="Search Human…" data-role="none" onkeydown="if (event.keyCode == 13) { $(this).closest('form').submit(); return false; }" style="background-image: url(&quot;data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==&quot;); padding: 0px; margin: 0px; border: medium none; outline: none; background-color: transparent; z-index: 2; position: absolute;" autocomplete="off"></div>
          </div>
          <div class="button"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 1000px 781px;cursor: pointer;width: 16px;background-position: -292px -47px;height: 16px" alt="Search&nbsp;»" onclick="$(this).closest('form').submit()"></div>
        </div>
        <div class="site_menu hidden">
          <div class="ensembl"><img class="no-sprite" src="VEP_results_MikeRaiko_files/Homo_sapiens.png" alt="Search Human">Search Human<input type="hidden" value="Search Human…"></div>
<div class="ensembl_all"><img class="no-sprite" src="VEP_results_MikeRaiko_files/ensembl.gif" alt="Search all species">Search all species<input type="hidden" value="Search all species…"></div>
<div class="ensembl_genomes"><img class="no-sprite" src="VEP_results_MikeRaiko_files/ensembl_genomes.gif" alt="Search Ensembl genomes">Search Ensembl genomes<input type="hidden" value="Search Ensembl genomes…"></div>
<div class="vega"><img class="no-sprite" src="VEP_results_MikeRaiko_files/vega.gif" alt="Search Vega">Search Vega<input type="hidden" value="Search Vega…"></div>
<div class="ebi"><img class="no-sprite" src="VEP_results_MikeRaiko_files/ebi.gif" alt="Search EMBL-EBI">Search EMBL-EBI<input type="hidden" value="Search EMBL-EBI…"></div>
<div class="sanger"><img class="no-sprite" src="VEP_results_MikeRaiko_files/sanger.gif" alt="Search Sanger">Search Sanger<input type="hidden" value="Search Sanger…"></div>

        </div>
      </form>
    </div>
  </div>
        </div>
  
        
        <div class="tabs_holder print_hide"><ul class="tabs"><li class="tools active final short_tab" style="display:none"><span class="dropdown"><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?tl=G9fJSvE7sdcvGHwt-8953987" title="VEP">VEP</a><a class="toggle" href="#" rel="tools">▼</a></span></li><li class="tools active final long_tab" style="display: block;"><span class="dropdown"><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?tl=G9fJSvE7sdcvGHwt-8953987">VEP</a><a class="toggle" href="#" rel="tools">▼</a></span></li></ul>
    <div class="dropdown history location">
      <h4>Recent locations</h4>
      <ul class="recent"><li><a class="constant clear_history bold" href="http://grch37.ensembl.org/Account/ClearHistory?object=Location">Clear history</a></li></ul>
    </div></div>
        
      </div>

      <div id="main_holder">
      
      <div id="page_nav_wrapper">
        <div id="page_nav" class="nav print_hide js_panel floating">
          
    <input type="hidden" class="panel_type" value="LocalContext">
    <div class="header">Web Tools</div>
    <ul class="local_context"><li class="top_level parent last"><img class="Web_Tools toggle" src="VEP_results_MikeRaiko_files/open.gif" alt=""><a class="Web_Tools" href="http://grch37.ensembl.org/Homo_sapiens/Tools/Summary?tl=G9fJSvE7sdcvGHwt-8953987" title="Web Tools">Web Tools</a><ul><li class=""><div><div><div></div><div></div><div></div><div></div></div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 1000px 781px;height: 14px;width: 14px;background-position: -452px -3px"><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/Blast?tl=G9fJSvE7sdcvGHwt-8953987" class="BLAST/BLAT" title="BLAST/BLAT">BLAST/BLAT</a></li><li class="parent"><div></div><div></div><img class="Variant_Effect_Predictor toggle" src="VEP_results_MikeRaiko_files/open.gif" alt=""><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP?tl=G9fJSvE7sdcvGHwt-8953987" class="Variant_Effect_Predictor" title="Variant Effect Predictor">Variant Effect Predictor</a><ul><li class="last active"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 14px;background-position: -452px -3px;height: 14px;background-size: 1000px 781px"><a title="VEP analysis of snps_clean_MikeRaiko.vcf in Homo_sapiens" href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?tl=G9fJSvE7sdcvGHwt-8953987" class="VEP_analysis_of_snps_clean_MikeRaiko.vcf_in_Homo_sapiens">VEP analysis of snps_clean_MikeRaiko.vcf in Homo_sapiens</a></li></ul></li><li class=""><div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 14px;background-position: -452px -3px;height: 14px;background-size: 1000px 781px"><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/LD?tl=G9fJSvE7sdcvGHwt-8953987" class="Linkage_Disequilibrium_Calculator" title="Linkage Disequilibrium Calculator">Linkage Disequilibrium Calculator</a></li><li class=""><div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="height: 14px;background-position: -452px -3px;width: 14px;background-size: 1000px 781px"><a title="Variant Recoder" href="http://grch37.ensembl.org/Homo_sapiens/Tools/VR?tl=G9fJSvE7sdcvGHwt-8953987" class="Variant_Recoder">Variant Recoder</a></li><li class=""><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 1000px 781px;height: 14px;width: 14px;background-position: -452px -3px"><a class="File_Chameleon" href="http://grch37.ensembl.org/Homo_sapiens/Tools/FileChameleon?tl=G9fJSvE7sdcvGHwt-8953987" title="File Chameleon">File Chameleon</a></li><li class=""><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 14px;background-position: -452px -3px;height: 14px;background-size: 1000px 781px"><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/AssemblyConverter?tl=G9fJSvE7sdcvGHwt-8953987" class="Assembly_Converter" title="Assembly Converter">Assembly Converter</a></li><li class=""><div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 14px;background-position: -452px -3px;height: 14px;background-size: 1000px 781px"><a title="ID History Converter" class="ID_History_Converter" href="http://grch37.ensembl.org/Homo_sapiens/Tools/IDMapper?tl=G9fJSvE7sdcvGHwt-8953987">ID History Converter</a></li><li class=""><div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 1000px 781px;height: 14px;width: 14px;background-position: -452px -3px"><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VcftoPed?tl=G9fJSvE7sdcvGHwt-8953987" class="VCF_to_PED_Converter" title="VCF to PED Converter">VCF to PED Converter</a></li><li class=""><div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 14px;background-position: -452px -3px;height: 14px;background-size: 1000px 781px"><a class="Allele_Frequency_Calculator" href="http://grch37.ensembl.org/Homo_sapiens/Tools/AlleleFrequency?tl=G9fJSvE7sdcvGHwt-8953987" title="Allele Frequency Calculator">Allele Frequency Calculator</a></li><li class=""><div></div><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="height: 14px;background-position: -452px -3px;width: 14px;background-size: 1000px 781px"><a title="Data Slicer" class="Data_Slicer" href="http://grch37.ensembl.org/Homo_sapiens/Tools/DataSlicer?tl=G9fJSvE7sdcvGHwt-8953987">Data Slicer</a></li><li class="last"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 14px;background-position: -452px -3px;height: 14px;background-size: 1000px 781px"><a title="Variation Pattern Finder" href="http://grch37.ensembl.org/Homo_sapiens/Tools/VariationPattern?tl=G9fJSvE7sdcvGHwt-8953987" class="Variation_Pattern_Finder">Variation Pattern Finder</a></li></ul></li></ul>
          <div class="tool_buttons"><p class="disabled config" title="There are no options for this page" style="display: block;">Configure this page</p><p class="disabled  data" title="" style="display: block;">Custom tracks</p><p class="disabled export" title="Generic export has been disabled on this page. Check individual images, tables, etc for download buttons" style="display: block;">Export data</p><p style="display: block;"><a href="http://grch37.ensembl.org/Homo_sapiens/Share/Tools/VEP/Results?create=1;time=1676479822" class="share">Share this page</a></p><p style="display: block;"><a href="http://grch37.ensembl.org/Multi/Account/Login?then=/Multi/Account/Bookmark/Add%3Fdescription%3DVariant%20Effect%20Predictor%20results%20-%20Homo_sapiens%20-%20GRCh37%20Archive%20browser%20109%3Burl%3D//grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results%253Ftl%253DG9fJSvE7sdcvGHwt-8953987" class="modal_link bookmark">Bookmark this page</a></p></div>
          
          <p class="invisible">.</p>
        </div>
      </div>
    

      <div id="main">
          
          
          
    <div class="panel js_panel" id="ensembl_panel_1">
      <input type="hidden" class="panel_type" value="Content">
      
        <div class="nav-heading">
          <h1 class="caption"> <a href="http://grch37.ensembl.org/Help/View?id=483" class="popup constant help-header _ht"><span>Variant Effect Predictor results<span class="sprite help_icon"></span></span></a></h1>
          <p class="invisible">.</p>
        </div>
        <div class="content"><div class="session_messages js_panel" id="ensembl_panel_2"><input type="hidden" class="panel_type" value="Message"></div><div class="ajax initial_panel"><div class="js_panel" id="TicketDetails"><input type="hidden" class="panel_type" value="TicketDetails"><h3><a rel="_ticket_details" class="toggle _slide_toggle closed" href="#">Job details</a></h3><div class="_ticket_details toggleable hidden twocol"><div class="row"><div class="lhs">Job summary</div><div class="rhs"><div><p>VEP analysis of snps_clean_MikeRaiko.vcf in Homo_sapiens</p><p class="job-icons"><a class="_ticket_edit _change_location" href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Edit?db=core;tl=G9fJSvE7sdcvGHwt-8953987"><span class="edit_icon _ht sprite"></span></a><a class="_json_link _ticket_delete" href="http://grch37.ensembl.org/Homo_sapiens/Json/Tools/VEP/delete?db=core;tl=G9fJSvE7sdcvGHwt-8953987" data-redirect-url="/Homo_sapiens/Tools/VEP?db=core"><span class="_ht sprite delete_icon"></span><span class="_confirm hidden">This will delete this job permanently.</span></a></p></div></div></div><div class="row"><div class="lhs">Species</div><div class="rhs"><p><img class="job-species" src="VEP_results_MikeRaiko_files/Homo_sapiens.png" alt="" width="16" height="16">Human</p></div></div><div class="row"><div class="lhs">Assembly</div><div class="rhs"><p>GRCh37</p></div></div><div class="row"><div class="lhs">Options summary</div><div class="rhs"><div class="vep-job twocol"><div class="row bg1"><div class="lhs"><p><span class="ht _ht">1000 Genomes global minor allele frequency:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="bg2 row"><div class="lhs"><p><span class="ht _ht">APPRIS:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="row bg1"><div class="lhs"><p><span class="ht _ht">Buffer size:</span></p></div><div class="rhs"><p>5000</p></div></div><div class="row bg2"><div class="lhs"><p><span class="ht _ht">Filter by frequency:</span></p></div><div class="rhs"><p>Disabled</p></div></div><div class="bg1 row"><div class="lhs"><p><span class="ht _ht">Find co-located known variants:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="bg2 row"><div class="lhs"><p><span class="ht _ht">Gene symbol:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="bg1 row"><div class="lhs"><p><span class="ht _ht">MANE:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="bg2 row"><div class="lhs"><p><span class="ht _ht">PolyPhen:</span></p></div><div class="rhs"><p>Prediction and score</p></div></div><div class="bg1 row"><div class="lhs"><p><span class="ht _ht">PubMed IDs for citations of co-located variants:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="row bg2"><div class="lhs"><p><span class="ht _ht">Get regulatory region consequences:</span></p></div><div class="rhs"><p>Yes</p></div></div><div class="row bg1"><div class="lhs"><p><span class="ht _ht">Restrict results:</span></p></div><div class="rhs"><p>Disabled</p></div></div><div class="bg2 row"><div class="lhs"><p><span class="ht _ht">Right align variants prior to consequence calculation:</span></p></div><div class="rhs"><p>Disabled</p></div></div><div class="row bg1"><div class="lhs"><p><span class="ht _ht">SIFT:</span></p></div><div class="rhs"><p>Prediction and score</p></div></div><div class="bg2 row"><div class="lhs"><p><span class="ht _ht">Transcript biotype:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="row bg1"><div class="lhs"><p><span class="ht _ht">Transcript database to use:</span></p></div><div class="rhs"><p>Ensembl transcripts</p></div></div><div class="bg2 row"><div class="lhs"><p><span>Transcript support level:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="bg1 row"><div class="lhs"><p><span class="ht _ht">Transcript version:</span></p></div><div class="rhs"><p>Enabled</p></div></div><div class="bg2 row"><div class="lhs"><p><span class="ht _ht">Upstream/Downstream distance (bp):</span></p></div><div class="rhs"><p>5000</p></div></div></div></div></div><div class="row"><div class="lhs">VEP and data version</div><div class="rhs"><div class="vep-job twocol"><div class="row bg1"><div class="lhs">1000genomes</div><div class="rhs"><p>phase3</p></div></div><div class="row bg2"><div class="lhs">Assembly</div><div class="rhs"><p>GRCh37.p13</p></div></div><div class="bg1 row"><div class="lhs">Cache</div><div class="rhs"><p>109_GRCh37</p></div></div><div class="bg2 row"><div class="lhs">ClinVar</div><div class="rhs"><p>202012</p></div></div><div class="row bg1"><div class="lhs">COSMIC</div><div class="rhs"><p>92</p></div></div><div class="bg2 row"><div class="lhs">Database</div><div class="rhs"><p>homo_sapiens_core_109_37</p></div></div><div class="row bg1"><div class="lhs">dbSNP</div><div class="rhs"><p>154</p></div></div><div class="bg2 row"><div class="lhs">GENCODE</div><div class="rhs"><p>GENCODE 19</p></div></div><div class="row bg1"><div class="lhs">Genebuild</div><div class="rhs"><p>2011-04</p></div></div><div class="bg2 row"><div class="lhs">gnomADe</div><div class="rhs"><p>r2.1</p></div></div><div class="row bg1"><div class="lhs">HGMD-PUBLIC</div><div class="rhs"><p>20204</p></div></div><div class="bg2 row"><div class="lhs">Polyphen</div><div class="rhs"><p>2.2.2</p></div></div><div class="bg1 row"><div class="lhs">Regbuild</div><div class="rhs"><p>1.0</p></div></div><div class="row bg2"><div class="lhs">SIFT</div><div class="rhs"><p>sift5.2.2</p></div></div><div class="row bg1"><div class="lhs">Time</div><div class="rhs"><p>2023-02-15 16:21:07</p></div></div><div class="bg2 row"><div class="lhs">VEP</div><div class="rhs"><p>v109</p></div></div></div></div></div><div class="row"><div class="lhs"><span class="ht _ht">Command line equivalent</span></div><div class="rhs"><pre class="code" style="height: 30px">./vep --af --appris --biotype --buffer_size 500 --check_existing --distance 5000 --mane --polyphen b --pubmed --regulatory --sift b --species homo_sapiens --symbol --transcript_version --tsl --cache --input_file [input_data] --output_file [output_file] --port 3337</pre></div></div></div></div></div><div class="ajax initial_panel"><div class="js_panel" id="ResultsSummary"><h3><a rel="job_stats" class="toggle set_cookie open" href="#">Summary statistics</a></h3><input type="hidden" class="panel_type" value="VEPResultsSummary"><div class="job_stats"><div class="toggleable" style="display: block;"><div class="population_genetics_pie"><div style="float:left;" class="right-margin">
    <table class="ss autocenter" style="width: 100%" cellspacing="0" cellpadding="0">
      <thead><tr class="ss_header"><th title="Category">Category</th><th title="Count">Count</th></tr></thead>
      <tbody><tr class="bg1"><td>Variants processed</td><td>164638</td></tr><tr class="bg2"><td>Variants filtered out</td><td>0</td></tr><tr class="bg1"><td>Novel / existing variants</td><td>76489 (46.5) / 88149 (53.5)</td></tr><tr class="bg2"><td>Overlapped genes</td><td>27951</td></tr><tr class="bg1"><td>Overlapped transcripts</td><td>106698</td></tr><tr class="bg2"><td>Overlapped regulatory features</td><td>33643</td></tr></tbody>
    </table>
    
  </div>
      <div class="pie_chart_holder right-margin">
        <div class="pie_chart" title="Consequences (all)">
          <h4>Consequences (all)</h4>
          <div id="graphHolder0" style="width:350px;height:183px"><svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="350" height="183"><desc>Created with Raphaël</desc><defs></defs><path fill="#02599c" stroke="#ffffff" d="M65,85L68.61902130040937,144.89075625526183A60,60,0,1,1,64.99999999999999,25Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#32cd32" stroke="#ffffff" d="M65,85L64.99999999999999,25A60,60,0,0,1,106.85928730861059,42.01395498519092Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#a2b5cd" stroke="#ffffff" d="M65,85L106.85928730861059,42.01395498519092A60,60,0,0,1,122.15065822308094,66.72974371639546Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#636363" stroke="#ffffff" d="M65,85L122.15065822308094,66.72974371639546A60,60,0,0,1,124.1794677877444,94.88891259741551Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#a2b5cd" stroke="#ffffff" d="M65,85L124.1794677877444,94.88891259741551A60,60,0,0,1,113.98585500484316,119.64658726980886Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#a52a2a" stroke="#ffffff" d="M65,85L113.98585500484316,119.64658726980886A60,60,0,0,1,100.12940607216811,133.64077331844877Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#ff4500" stroke="#ffffff" d="M65,85L100.12940607216811,133.64077331844877A60,60,0,0,1,86.53390397191114,141.00259797302718Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#a52a2a" stroke="#ffffff" d="M65,85L86.53390397191114,141.00259797302718A60,60,0,0,1,82.91620387149004,142.2626373723321Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#32cd32" stroke="#ffffff" d="M65,85L82.91620387149004,142.2626373723321A60,60,0,0,1,79.27243823423724,143.27776168359492Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#ffd700" stroke="#ffffff" d="M65,85L79.27243823423724,143.27776168359492A60,60,0,0,1,68.61902130040936,144.89075625526183Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#000000" stroke="none" d="M65,85L68.61902130040937,144.89075625526183A60,60,0,1,1,64.99999999999999,25Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L64.99999999999999,25A60,60,0,0,1,106.85928730861059,42.01395498519092Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L106.85928730861059,42.01395498519092A60,60,0,0,1,122.15065822308094,66.72974371639546Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L122.15065822308094,66.72974371639546A60,60,0,0,1,124.1794677877444,94.88891259741551Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L124.1794677877444,94.88891259741551A60,60,0,0,1,113.98585500484316,119.64658726980886Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L113.98585500484316,119.64658726980886A60,60,0,0,1,100.12940607216811,133.64077331844877Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L100.12940607216811,133.64077331844877A60,60,0,0,1,86.53390397191114,141.00259797302718Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L86.53390397191114,141.00259797302718A60,60,0,0,1,82.91620387149004,142.2626373723321Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L82.91620387149004,142.2626373723321A60,60,0,0,1,79.27243823423724,143.27776168359492Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L79.27243823423724,143.27776168359492A60,60,0,0,1,68.61902130040936,144.89075625526183Z" style="fill-opacity: 0;" fill-opacity="0"></path><circle cx="142" cy="19.480000972747803" r="5" fill="#02599c" stroke="none"></circle><text x="157" y="21.880000591278076" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>intron_variant: 51%</tspan></text><circle cx="142" cy="34.84000120162963" r="5" fill="#32cd32" stroke="none"></circle><text x="157" y="37.24000139236449" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>non_coding_transcript_variant: 12%</tspan></text><circle cx="142" cy="50.20000143051146" r="5" fill="#a2b5cd" stroke="none"></circle><text x="157" y="52.60000123977659" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>downstream_gene_variant: 8%</tspan></text><circle cx="142" cy="65.5600016593933" r="5" fill="#636363" stroke="none"></circle><text x="157" y="67.95999727249142" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>intergenic_variant: 8%</tspan></text><circle cx="142" cy="80.92000188827512" r="5" fill="#a2b5cd" stroke="none"></circle><text x="157" y="83.32000856399532" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>upstream_gene_variant: 7%</tspan></text><circle cx="142" cy="96.28000211715695" r="5" fill="#a52a2a" stroke="none"></circle><text x="157" y="98.68000841140741" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>regulatory_region_variant: 5%</tspan></text><circle cx="142" cy="111.64000234603878" r="5" fill="#ff4500" stroke="none"></circle><text x="157" y="114.04000825881951" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>NMD_transcript_variant: 4%</tspan></text><circle cx="142" cy="127.00000257492061" r="5" fill="#a52a2a" stroke="none"></circle><text x="157" y="129.4000081062316" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>TF_binding_site_variant: 1%</tspan></text><circle cx="142" cy="142.36000280380244" r="5" fill="#32cd32" stroke="none"></circle><text x="157" y="144.7600079536437" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>non_coding_transcript_exon_variant: 1%</tspan></text><circle cx="142" cy="157.72000303268428" r="5" fill="#ffd700" stroke="none"></circle><text x="157" y="160.1200078010558" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>Others</tspan></text></svg></div>
        </div>
      </div>
      <div class="pie_chart_holder right-margin">
        <div class="pie_chart" title="Coding consequences">
          <h4>Coding consequences</h4>
          <div id="graphHolder1" style="width:350px;height:183px"><svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="350" height="183"><desc>Created with Raphaël</desc><defs></defs><path fill="#ffd700" stroke="#ffffff" d="M65,85L105.8646160324844,128.93271169092137A60,60,0,1,1,64.99999999999999,25Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#76ee00" stroke="#ffffff" d="M65,85L64.99999999999999,25A60,60,0,0,1,108.85157053445299,125.95167593227285Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#ff0000" stroke="#ffffff" d="M65,85L108.85157053445299,125.95167593227285A60,60,0,0,1,107.66416798331998,127.18730579559514Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#ff0000" stroke="#ffffff" d="M65,85L107.66416798331998,127.18730579559514A60,60,0,0,1,106.83042440425692,128.01413249339973Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#ffd700" stroke="#ffffff" d="M65,85L106.83042440425692,128.01413249339973A60,60,0,0,1,106.09641756465524,128.71595204672428Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#76ee00" stroke="#ffffff" d="M65,85L106.09641756465524,128.71595204672428A60,60,0,0,1,105.91106800799209,128.88945789646354Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#458b00" stroke="#ffffff" d="M65,85L105.91106800799209,128.88945789646354A60,60,0,0,1,105.88784773981926,128.9110909361787Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#ff00ff" stroke="#ffffff" d="M65,85L105.88784773981926,128.9110909361787A60,60,0,0,1,105.86461603248442,128.93271169092134Z" style="stroke-width: 1px; stroke-linejoin: round;" stroke-width="1" stroke-linejoin="round"></path><path fill="#000000" stroke="none" d="M65,85L105.8646160324844,128.93271169092137A60,60,0,1,1,64.99999999999999,25Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L64.99999999999999,25A60,60,0,0,1,108.85157053445299,125.95167593227285Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L108.85157053445299,125.95167593227285A60,60,0,0,1,107.66416798331998,127.18730579559514Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L107.66416798331998,127.18730579559514A60,60,0,0,1,106.83042440425692,128.01413249339973Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L106.83042440425692,128.01413249339973A60,60,0,0,1,106.09641756465524,128.71595204672428Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L106.09641756465524,128.71595204672428A60,60,0,0,1,105.91106800799209,128.88945789646354Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L105.91106800799209,128.88945789646354A60,60,0,0,1,105.88784773981926,128.9110909361787Z" style="fill-opacity: 0;" fill-opacity="0"></path><path fill="#000000" stroke="none" d="M65,85L105.88784773981926,128.9110909361787A60,60,0,0,1,105.86461603248442,128.93271169092134Z" style="fill-opacity: 0;" fill-opacity="0"></path><circle cx="142" cy="34.840001583099365" r="5" fill="#ffd700" stroke="none"></circle><text x="157" y="37.240002155303955" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>missense_variant: 62%</tspan></text><circle cx="142" cy="50.200001811981195" r="5" fill="#76ee00" stroke="none"></circle><text x="157" y="52.60000200271605" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>synonymous_variant: 37%</tspan></text><circle cx="142" cy="65.56000204086303" r="5" fill="#ff0000" stroke="none"></circle><text x="157" y="67.95999803543089" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>stop_lost: 0%</tspan></text><circle cx="142" cy="80.92000226974486" r="5" fill="#ff0000" stroke="none"></circle><text x="157" y="83.32000169754025" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>stop_gained: 0%</tspan></text><circle cx="142" cy="96.28000249862669" r="5" fill="#ffd700" stroke="none"></circle><text x="157" y="98.68000917434688" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>start_lost: 0%</tspan></text><circle cx="142" cy="111.64000272750852" r="5" fill="#76ee00" stroke="none"></circle><text x="157" y="114.04000902175898" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>stop_retained_variant: 0%</tspan></text><circle cx="142" cy="127.00000295639035" r="5" fill="#458b00" stroke="none"></circle><text x="157" y="129.40000886917107" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>coding_sequence_variant: 0%</tspan></text><circle cx="142" cy="142.36000318527218" r="5" fill="#ff00ff" stroke="none"></circle><text x="157" y="144.76000871658317" text-anchor="start" style="text-anchor: start; font: 12px &quot;Luxi Sans&quot;, &quot;Helvetica&quot;, sans-serif;" font="12px 'Luxi Sans','Helvetica', sans-serif" stroke="none" fill="#000000"><tspan>incomplete_terminal_codon_variant: 0%</tspan></text></svg></div>
        </div>
      </div><div><input class="graph_config" type="hidden" name="legendpos" value="'east'"><input class="graph_config" type="hidden" name="legendmark" value="'circle'"><input class="graph_config" type="hidden" name="maxSlices" value="100"><input class="graph_config" type="hidden" name="minPercent" value="0"><input class="graph_dimensions" type="hidden" value="[65,85,60]"><input class="js_param" type="hidden" name="cons_colours" value="{&quot;mature_miRNA_variant&quot;:&quot;#458b00&quot;,&quot;5_prime_UTR_variant&quot;:&quot;#7ac5cd&quot;,&quot;splice_acceptor_variant&quot;:&quot;#FF581A&quot;,&quot;TFBS_ablation&quot;:&quot;#a52a2a&quot;,&quot;splice_donor_variant&quot;:&quot;#FF581A&quot;,&quot;feature_elongation&quot;:&quot;#ff0000&quot;,&quot;frameshift_variant&quot;:&quot;#9400D3&quot;,&quot;stop_lost&quot;:&quot;#ff0000&quot;,&quot;stop_gained&quot;:&quot;#ff0000&quot;,&quot;non_coding_transcript_exon_variant&quot;:&quot;#32cd32&quot;,&quot;transcript_amplification&quot;:&quot;#ff69b4&quot;,&quot;coding_sequence_variant&quot;:&quot;#458b00&quot;,&quot;stop_retained_variant&quot;:&quot;#76ee00&quot;,&quot;splice_region_variant&quot;:&quot;#ff7f50&quot;,&quot;regulatory_region_ablation&quot;:&quot;#a52a2a&quot;,&quot;intron_variant&quot;:&quot;#02599c&quot;,&quot;incomplete_terminal_codon_variant&quot;:&quot;#ff00ff&quot;,&quot;upstream_gene_variant&quot;:&quot;#a2b5cd&quot;,&quot;start_lost&quot;:&quot;#ffd700&quot;,&quot;TFBS_amplification&quot;:&quot;#a52a2a&quot;,&quot;start_retained_variant&quot;:&quot;#76ee00&quot;,&quot;3_prime_UTR_variant&quot;:&quot;#7ac5cd&quot;,&quot;non_coding_transcript_variant&quot;:&quot;#32cd32&quot;,&quot;intergenic_variant&quot;:&quot;#636363&quot;,&quot;downstream_gene_variant&quot;:&quot;#a2b5cd&quot;,&quot;regulatory_region_variant&quot;:&quot;#a52a2a&quot;,&quot;splice_donor_5th_base_variant&quot;:&quot;#ff0000&quot;,&quot;synonymous_variant&quot;:&quot;#76ee00&quot;,&quot;feature_truncation&quot;:&quot;#ff0000&quot;,&quot;protein_altering_variant&quot;:&quot;#FF0080&quot;,&quot;regulatory_region_amplification&quot;:&quot;#a52a2a&quot;,&quot;NMD_transcript_variant&quot;:&quot;#ff4500&quot;,&quot;sequence_variant&quot;:&quot;#ff0000&quot;,&quot;TF_binding_site_variant&quot;:&quot;#a52a2a&quot;,&quot;inframe_insertion&quot;:&quot;#ff69b4&quot;,&quot;splice_polypyrimidine_tract_variant&quot;:&quot;#ff0000&quot;,&quot;splice_donor_region_variant&quot;:&quot;#ff0000&quot;,&quot;transcript_ablation&quot;:&quot;#ff0000&quot;,&quot;missense_variant&quot;:&quot;#ffd700&quot;,&quot;inframe_deletion&quot;:&quot;#ff69b4&quot;}"><input type="hidden" class="graph_data" value="[[381425,'intron_variant'],[91977,'non_coding_transcript_variant'],[58282,'downstream_gene_variant'],[56582,'intergenic_variant'],[53608,'upstream_gene_variant'],[39277,'regulatory_region_variant'],[30781,'NMD_transcript_variant'],[7607,'TF_binding_site_variant'],[7511,'non_coding_transcript_exon_variant'],[7356,'missense_variant'],[5574,'3_prime_UTR_variant'],[4390,'synonymous_variant'],[1317,'5_prime_UTR_variant'],[1212,'splice_region_variant'],[1119,'splice_polypyrimidine_tract_variant'],[122,'splice_donor_region_variant'],[86,'splice_donor_variant'],[59,'splice_acceptor_variant'],[54,'stop_lost'],[52,'splice_donor_5th_base_variant'],[37,'stop_gained'],[32,'start_lost'],[8,'stop_retained_variant'],[1,'mature_miRNA_variant'],[1,'coding_sequence_variant'],[1,'incomplete_terminal_codon_variant']]"><input type="hidden" class="graph_data" value="[[7356,'missense_variant'],[4390,'synonymous_variant'],[54,'stop_lost'],[37,'stop_gained'],[32,'start_lost'],[8,'stop_retained_variant'],[1,'coding_sequence_variant'],[1,'incomplete_terminal_codon_variant']]"></div></div></div></div></div></div><div class="ajax initial_panel"><div class="js_panel" id="Results"><div><h3>Results preview</h3><input type="hidden" class="panel_type" value="VEPResults"><input class="js_param" type="hidden" name="auto_values" value="{&quot;Consequence&quot;:[&quot;start_retained_variant&quot;,&quot;mature_miRNA_variant&quot;,&quot;splice_donor_variant&quot;,&quot;inframe_deletion&quot;,&quot;transcript_amplification&quot;,&quot;splice_acceptor_variant&quot;,&quot;transcript_ablation&quot;,&quot;stop_retained_variant&quot;,&quot;missense_variant&quot;,&quot;downstream_gene_variant&quot;,&quot;intron_variant&quot;,&quot;sequence_variant&quot;,&quot;upstream_gene_variant&quot;,&quot;start_lost&quot;,&quot;splice_donor_region_variant&quot;,&quot;feature_truncation&quot;,&quot;coding_sequence_variant&quot;,&quot;splice_region_variant&quot;,&quot;inframe_insertion&quot;,&quot;non_coding_transcript_variant&quot;,&quot;stop_gained&quot;,&quot;frameshift_variant&quot;,&quot;regulatory_region_variant&quot;,&quot;stop_lost&quot;,&quot;TF_binding_site_variant&quot;,&quot;TFBS_amplification&quot;,&quot;synonymous_variant&quot;,&quot;protein_altering_variant&quot;,&quot;feature_elongation&quot;,&quot;intergenic_variant&quot;,&quot;splice_donor_5th_base_variant&quot;,&quot;regulatory_region_ablation&quot;,&quot;incomplete_terminal_codon_variant&quot;,&quot;TFBS_ablation&quot;,&quot;regulatory_region_amplification&quot;,&quot;3_prime_UTR_variant&quot;,&quot;5_prime_UTR_variant&quot;,&quot;splice_polypyrimidine_tract_variant&quot;,&quot;NMD_transcript_variant&quot;,&quot;non_coding_transcript_exon_variant&quot;],&quot;Allele&quot;:[&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;],&quot;BIOTYPE&quot;:[&quot;protein_coding&quot;,&quot;nonsense_mediated_decay&quot;,&quot;processed_transcript&quot;,&quot;miRNA&quot;,&quot;processed_pseudogene&quot;,&quot;misc_RNA&quot;,&quot;lincRNA&quot;,&quot;snRNA&quot;,&quot;snoRNA&quot;,&quot;antisense&quot;,&quot;rRNA&quot;,&quot;transcribed_unprocessed_pseudogene&quot;,&quot;transcribed_processed_pseudogene&quot;,&quot;unprocessed_pseudogene&quot;,&quot;sense_intronic&quot;,&quot;pseudogene&quot;,&quot;retained_intron&quot;,&quot;unitary_pseudogene&quot;,&quot;sense_overlapping&quot;,&quot;IG_V_gene&quot;,&quot;IG_D_gene&quot;,&quot;3prime_overlapping_ncRNA&quot;,&quot;IG_V_pseudogene&quot;,&quot;non_stop_decay&quot;,&quot;polymorphic_pseudogene&quot;,&quot;Mt_tRNA&quot;,&quot;Mt_rRNA&quot;,&quot;TR_V_pseudogene&quot;,&quot;IG_C_pseudogene&quot;,&quot;TR_C_gene&quot;,&quot;TR_J_gene&quot;,&quot;TR_V_gene&quot;,&quot;TR_J_pseudogene&quot;,&quot;translated_processed_pseudogene&quot;,&quot;IG_J_pseudogene&quot;,&quot;IG_J_gene&quot;,&quot;IG_C_gene&quot;,&quot;TR_D_gene&quot;,&quot;LRG_gene&quot;],&quot;SIFT&quot;:[&quot;deleterious&quot;,&quot;deleterious_low_confidence&quot;,&quot;tolerated&quot;,&quot;tolerated_low_confidence&quot;],&quot;Feature_type&quot;:[&quot;Transcript&quot;,&quot;MotifFeature&quot;,&quot;RegulatoryFeature&quot;],&quot;PolyPhen&quot;:[&quot;benign&quot;,&quot;possibly_damaging&quot;,&quot;probably_damaging&quot;,&quot;unknown&quot;],&quot;IMPACT&quot;:[&quot;MODERATE&quot;,&quot;MODIFIER&quot;,&quot;HIGH&quot;,&quot;LOW&quot;]}"><div><div class="toolbox right-margin"><div class="toolbox-head"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="width: 16px;background-position: -924px -3px;vertical-align: top;height: 16px;background-size: 1000px 781px"> <span class="ht _ht"><span class="_ht_tip hidden">Navigate
 through the results of your VEP job. By default the results for 5 
variants are displayed; note that variants may have more than one result
 if they overlap multiple transcripts</span>Navigation</span> <small>(per variant)</small></div><div style="padding:5px;"><b>Page: </b><img src="VEP_results_MikeRaiko_files/nav-l2.gif" style="vertical-align:top; height:16px; width:16px; opacity: 0.5;"><img src="VEP_results_MikeRaiko_files/nav-l1.gif" style="vertical-align:top; height:16px; width:16px; opacity: 0.5;"> 1 of 32928 <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=6;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=10" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=6;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=10;update_panel=1"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 800px 624.8px;width: 16px;background-position: -596.8px -92px;vertical-align: top;height: 16px" title="Next page"></a><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=164636;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=164638" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=164636;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=164638;update_panel=1"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 800px 624.8px;height: 16px;width: 16px;background-position: -617.6px -92px;vertical-align: top" title="Last page"></a><span style="padding: 0px 10px 0px 10px; color: grey">|</span><b>Show: </b>  <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=1;tl=G9fJSvE7sdcvGHwt-8953987;to=1" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=1;tl=G9fJSvE7sdcvGHwt-8953987;to=1;update_panel=1">1</a> <span class="count-highlight">&nbsp;5&nbsp;</span> <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=10;tl=G9fJSvE7sdcvGHwt-8953987;to=10" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=10;tl=G9fJSvE7sdcvGHwt-8953987;to=10;update_panel=1">10</a> <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=50;tl=G9fJSvE7sdcvGHwt-8953987;to=50" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=50;tl=G9fJSvE7sdcvGHwt-8953987;to=50;update_panel=1">50</a> <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=164638;tl=G9fJSvE7sdcvGHwt-8953987;to=164638" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=164638;tl=G9fJSvE7sdcvGHwt-8953987;to=164638;update_panel=1">All<img class="_ht" src="VEP_results_MikeRaiko_files/alert.png" style="vertical-align: top;"></a> variants</div></div><div class="toolbox right-margin"><div class="toolbox-head"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="height: 16px;vertical-align: top;background-position: -292px -47px;width: 16px;background-size: 1000px 781px"> <span class="ht _ht"><span class="_ht_tip hidden">Filter
 your results to find interesting or significant data. You can apply 
several filters on any category of data in your results using a range of
 operators, add multiple filters, and edit active filters</span>Filters</span></div><div style="padding:0px 5px 0px 5px;"><div></div><div style="clear: left;"><form action="#" class="_apply_filter" style="margin: 0;"><input type="hidden" name="ajax_url" value="/Homo_sapiens/Component/Tools/VEP/Results/results?update_panel=1"><select class="autocomplete right-margin" name="field1"><option value="ID" selected="selected">Uploaded variant</option><option value="Location">Location</option><option value="Allele">Allele</option><option value="Consequence">Consequence</option><option value="IMPACT">Impact</option><option value="SYMBOL">Symbol</option><option value="Gene">Gene</option><option value="Feature_type">Feature type</option><option value="Feature">Feature</option><option value="BIOTYPE">Biotype</option><option value="EXON">Exon</option><option value="INTRON">Intron</option><option value="cDNA_position">cDNA position</option><option value="CDS_position">CDS position</option><option value="Protein_position">Protein position</option><option value="Amino_acids">Amino acids</option><option value="Codons">Codons</option><option value="Existing_variation">Existing variant</option><option value="DISTANCE">Distance to transcript</option><option value="STRAND">Feature strand</option><option value="FLAGS">FLAGS</option><option value="SYMBOL_SOURCE">Symbol source</option><option value="HGNC_ID">HGNC ID</option><option value="MANE_SELECT">MANE SELECT</option><option value="MANE_PLUS_CLINICAL">MANE PLUS CLINICAL</option><option value="TSL">Transcript support level</option><option value="APPRIS">APPRIS</option><option value="SIFT">SIFT</option><option value="PolyPhen">PolyPhen</option><option value="AF">AF</option><option value="CLIN_SIG">Clinical significance</option><option value="SOMATIC">Somatic status</option><option value="PHENO">Phenotype or disease</option><option value="PUBMED">Pubmed</option><option value="MOTIF_NAME">Motif name</option><option value="MOTIF_POS">Motif position</option><option value="HIGH_INF_POS">High info position</option><option value="MOTIF_SCORE_CHANGE">Motif score change</option><option value="TRANSCRIPTION_FACTORS">TRANSCRIPTION FACTORS</option></select><select class="_operator_dd right-margin" name="operator1"><option value="is" selected="selected">is</option><option value="ne">is not</option><option value="re">matches</option><option value="lt">&lt;</option><option value="gt">&gt;</option><option value="lte">&lt;=</option><option value="gte">&gt;=</option><option value="in">in file</option></select><input class="autocomplete _value_switcher right-margin" type="text" placeholder="defined" name="value1"><span class="_value_switcher hidden"><span class="small"> <a href="http://grch37.ensembl.org/Homo_sapiens/UserData/SelectFile?db=core;tl=G9fJSvE7sdcvGHwt-8953987;tool=VEP" class="modal_link data" rel="modal_user_data">Upload file</a> </span></span><input value="Add" class="fbutton" type="submit"><input type="hidden" name="tl" value="G9fJSvE7sdcvGHwt-8953987"><input type="hidden" name="time" value="1676479823696.696"></form></div></div></div><div class="toolbox"><div class="toolbox-head"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 1000px 781px;height: 16px;background-position: -880px -3px;width: 16px;vertical-align: top"> Download</div><div style="padding:5px;"><div><b>All:</b><span style="float:right; margin-left:10px;"> <a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Download/Tools/VEP?format=vcf;tl=G9fJSvE7sdcvGHwt-8953987">VCF</a> <a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Download/Tools/VEP?format=vep;tl=G9fJSvE7sdcvGHwt-8953987">VEP</a> <a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Download/Tools/VEP?format=txt;tl=G9fJSvE7sdcvGHwt-8953987">TXT</a></span></div><div style="margin-top: 5px"><b>BioMart:</b><span style="float:right; margin-left:10px;"><a class="_ht" rel="external" href="http://grch37.ensembl.org/biomart/martview?VIRTUALSCHEMANAME=default&amp;ATTRIBUTES=hsapiens_snp.default.snp.refsnp_id|hsapiens_snp.default.snp.refsnp_source|hsapiens_snp.default.snp.chr_name|hsapiens_snp.default.snp.chrom_start&amp;FILTERS=hsapiens_snp.default.filters.snp_filter.%22rs3131972,rs13303106,rs6696281%22&amp;VISIBLEPANEL=filterpanel">Variants</a> <a class="_ht" rel="external" href="http://grch37.ensembl.org/biomart/martview?VIRTUALSCHEMANAME=default&amp;ATTRIBUTES=hsapiens_gene_ensembl.default.feature_page.ensembl_gene_id|hsapiens_gene_ensembl.default.feature_page.chromosome_name|hsapiens_gene_ensembl.default.feature_page.start_position|hsapiens_gene_ensembl.default.feature_page.end_position&amp;FILTERS=hsapiens_gene_ensembl.default.filters.ensembl_gene_id.%22ENSG00000177757,ENSG00000240453,ENSG00000230368,ENSG00000228794,ENSG00000188976,ENSG00000187961,ENSG00000187583%22&amp;VISIBLEPANEL=filterpanel">Genes</a></span></div></div></div><span class="left-margin"><a class="button " href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP?db=core;expand_form=true;tl=G9fJSvE7sdcvGHwt-8953987">New job</a></span></div><div>
    <div class="dataTables_wrapper"><div class="dataTables_top"><div class="col_toggle"><div class="toggle">Show/hide columns (22 hidden)<ul class="floating_popup"><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Uploaded variant</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Location</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Allele</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Consequence</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Impact</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Symbol</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Gene</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Feature type</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Feature</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Biotype</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Exon</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Intron</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">cDNA position</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">CDS position</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Protein position</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Amino acids</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Codons</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Existing variant</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Distance to transcript</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Feature strand</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">FLAGS</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Symbol source</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">HGNC ID</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">MANE SELECT</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">MANE PLUS CLINICAL</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Transcript support level</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">APPRIS</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">SIFT</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">PolyPhen</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">AF</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Clinical significance</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Somatic status</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Phenotype or disease</span></span></li><li><input data-role="none" type="checkbox"><span><span class="ht _ht">Pubmed</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Motif name</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Motif position</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">High info position</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">Motif score change</span></span></li><li><input data-role="none" type="checkbox" checked="checked"><span><span class="ht _ht">TRANSCRIPTION FACTORS</span></span></li></ul></div></div><div class="invisible"></div></div><div style="overflow:auto"><div style="height: 1px; width: 2243px;" aria-describedby="ui-tooltip-0"></div></div><div style="overflow:auto"><table class="ss data_table autocenter _scrollytable_id_32357782608827057" style="width: 100%" cellspacing="0" cellpadding="0">
      
      
    <thead style=""><tr class="ss_header" style="white-space: nowrap; width: 100%;"><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Identifier of uploaded variant</span>Uploaded variant</span></th><th class="sort_position_html sorting_asc" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Location of variant in standard coordinate format (chr:start or chr:start-end)</span>Location</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">The variant allele used to calculate the consequence</span>Allele</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Consequence type</span>Consequence</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Gene symbol (e.g. HGNC)</span>Symbol</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Stable ID of affected gene</span>Gene</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Type of feature - Transcript, RegulatoryFeature or MotifFeature</span>Feature type</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Stable ID of feature</span>Feature</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Biotype of transcript or regulatory feature</span>Biotype</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Identifier(s) of co-located known variants</span>Existing variant</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Strand of the feature (1/-1)</span>Feature strand</span></th><th class="sort_numeric sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">Frequency of existing variant in 1000 Genomes combined population</span>AF</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">The stable identifier of a transcription factor binding profile (TFBP) aligned at this position</span>Motif name</span></th><th class="sort_numeric sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">The relative position of the variation in the aligned TFBP</span>Motif position</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">A flag indicating if the variant falls in a high information position of the TFBP</span>High info position</span></th><th class="sort_numeric sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">The difference in motif score of the reference and variant sequences for the TFBP</span>Motif score change</span></th><th class="sort_string sorting" rowspan="1" colspan="1" style="white-space: normal;"><span class="ht _ht"><span class="_ht_tip hidden">List of transcription factors which bind to the transcription factor binding profile</span>TRANSCRIPTION FACTORS</span></th></tr></thead><tbody><tr class="bg1"><td>rs3131972</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:752671-752771;tl=G9fJSvE7sdcvGHwt-8953987">1:752721-752721</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>FAM87B</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000177757;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000177757</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000177757;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000326734.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000326734.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000326734.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>lincRNA</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972">rs3131972</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972"></a></span></td><td>1</td><td>0.6534</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs3131972</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:752671-752771;tl=G9fJSvE7sdcvGHwt-8953987">1:752721-752721</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr>, <nobr><span class="colour" style="background-color:#32cd32">&nbsp;</span> <span class="_ht ht">non_coding_transcript_variant</span></nobr></td><td>RP11-206L10.10</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000240453;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000240453</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000240453;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000435300.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000435300.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000435300.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>processed_transcript</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972">rs3131972</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972"></a></span></td><td>-1</td><td>0.6534</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs3131972</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:752671-752771;tl=G9fJSvE7sdcvGHwt-8953987">1:752721-752721</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a52a2a">&nbsp;</span> <span class="_ht ht">regulatory_region_variant</span></nobr></td><td>-</td><td>-</td><td>RegulatoryFeature</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Regulation/Summary?db=core;rf=ENSR00000918393;tl=G9fJSvE7sdcvGHwt-8953987">ENSR00000918393</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Regulation?db=core;rf=ENSR00000918393;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>promoter</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972">rs3131972</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972"></a></span></td><td>-</td><td>0.6534</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs3131972</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:752671-752771;tl=G9fJSvE7sdcvGHwt-8953987">1:752721-752721</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a52a2a">&nbsp;</span> <span class="_ht ht">TF_binding_site_variant</span></nobr></td><td>-</td><td>-</td><td>MotifFeature</td><td>ENSM00800836535</td><td>-</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972">rs3131972</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs3131972"></a></span></td><td>1</td><td>0.6534</td><td>ENSPFM0577</td><td>8</td><td>N</td><td>0.006</td><td>TEAD4::SOX6</td></tr><tr class="bg1"><td>rs11240777</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:798909-799009;tl=G9fJSvE7sdcvGHwt-8953987">1:798959-798959</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>FAM41C</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000230368;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000230368</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000230368;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000427857.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000427857.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000427857.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>lincRNA</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs11240777</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:798909-799009;tl=G9fJSvE7sdcvGHwt-8953987">1:798959-798959</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>FAM41C</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000230368;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000230368</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000230368;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000432963.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000432963.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000432963.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>lincRNA</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs11240777</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:798909-799009;tl=G9fJSvE7sdcvGHwt-8953987">1:798959-798959</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>LINC01128</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000228794;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000228794</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000228794;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000445118.2;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000445118.2</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000445118.2;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>lincRNA</td><td>-</td><td>1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs11240777</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:798909-799009;tl=G9fJSvE7sdcvGHwt-8953987">1:798959-798959</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>FAM41C</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000230368;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000230368</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000230368;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000446136.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000446136.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000446136.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>lincRNA</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303106</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:891895-891995;tl=G9fJSvE7sdcvGHwt-8953987">1:891945-891945</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000327044.6;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000327044.6</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000327044.6;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106">rs13303106</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106"></a></span></td><td>-1</td><td>0.4541</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303106</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:891895-891995;tl=G9fJSvE7sdcvGHwt-8953987">1:891945-891945</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000338591.3;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000338591.3</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000338591.3;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106">rs13303106</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106"></a></span></td><td>1</td><td>0.4541</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303106</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:891895-891995;tl=G9fJSvE7sdcvGHwt-8953987">1:891945-891945</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000463212.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000463212.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000463212.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106">rs13303106</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106"></a></span></td><td>1</td><td>0.4541</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303106</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:891895-891995;tl=G9fJSvE7sdcvGHwt-8953987">1:891945-891945</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000469563.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000469563.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000469563.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106">rs13303106</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106"></a></span></td><td>-1</td><td>0.4541</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303106</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:891895-891995;tl=G9fJSvE7sdcvGHwt-8953987">1:891945-891945</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr>, <nobr><span class="colour" style="background-color:#32cd32">&nbsp;</span> <span class="_ht ht">non_coding_transcript_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000477976.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000477976.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000477976.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106">rs13303106</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106"></a></span></td><td>-1</td><td>0.4541</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303106</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:891895-891995;tl=G9fJSvE7sdcvGHwt-8953987">1:891945-891945</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr>, <nobr><span class="colour" style="background-color:#32cd32">&nbsp;</span> <span class="_ht ht">non_coding_transcript_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000487214.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000487214.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000487214.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>processed_transcript</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106">rs13303106</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs13303106"></a></span></td><td>-1</td><td>0.4541</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000327044.6;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000327044.6</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000327044.6;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000338591.3;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000338591.3</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000338591.3;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td>-</td><td>1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000463212.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000463212.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000463212.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td>-</td><td>1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000466300.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000466300.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000466300.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>nonsense_mediated_decay</td><td>-</td><td>1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr>, <nobr><span class="colour" style="background-color:#32cd32">&nbsp;</span> <span class="_ht ht">non_coding_transcript_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000469563.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000469563.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000469563.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000477976.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000477976.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000477976.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000481067.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000481067.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000481067.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td>-</td><td>1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr>, <nobr><span class="colour" style="background-color:#32cd32">&nbsp;</span> <span class="_ht ht">non_coding_transcript_variant</span></nobr></td><td>NOC2L</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000188976</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000188976;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000487214.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000487214.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000487214.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>processed_transcript</td><td>-</td><td>-1</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs13303010</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:894523-894623;tl=G9fJSvE7sdcvGHwt-8953987">1:894573-894573</a></td><td>G</td><td><nobr><span class="colour" style="background-color:#a52a2a">&nbsp;</span> <span class="_ht ht">regulatory_region_variant</span></nobr></td><td>-</td><td>-</td><td>RegulatoryFeature</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Regulation/Summary?db=core;rf=ENSR00000000111;tl=G9fJSvE7sdcvGHwt-8953987">ENSR00000000111</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Regulation?db=core;rf=ENSR00000000111;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>promoter</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000338591.3;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000338591.3</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000338591.3;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr></td><td>PLEKHN1</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187583</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000379407.3;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000379407.3</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000379407.3;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr></td><td>PLEKHN1</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187583</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000379409.2;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000379409.2</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000379409.2;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#02599c">&nbsp;</span> <span class="_ht ht">intron_variant</span></nobr></td><td>PLEKHN1</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187583</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000379410.3;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000379410.3</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000379410.3;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>protein_coding</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000466300.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000466300.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000466300.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>nonsense_mediated_decay</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg1"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">upstream_gene_variant</span></nobr></td><td>PLEKHN1</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187583</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187583;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000480267.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000480267.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000480267.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr><tr class="bg2"><td>rs6696281</td><td class=" sorting_1"><a class="_ht" href="http://grch37.ensembl.org/Homo_sapiens/Location/View?contigviewbottom=variation_feature_variation%3Dnormal;db=core;r=1:903054-903154;tl=G9fJSvE7sdcvGHwt-8953987">1:903104-903104</a></td><td>T</td><td><nobr><span class="colour" style="background-color:#a2b5cd">&nbsp;</span> <span class="_ht ht">downstream_gene_variant</span></nobr></td><td>KLHL17</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987">ENSG00000187961</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Gene?db=core;g=ENSG00000187961;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>Transcript</td><td><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=ENST00000481067.1;tl=G9fJSvE7sdcvGHwt-8953987">ENST00000481067.1</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Transcript?db=core;t=ENST00000481067.1;tl=G9fJSvE7sdcvGHwt-8953987"></a></td><td>retained_intron</td><td><span><a class="_zmenu" href="http://grch37.ensembl.org/Homo_sapiens/Variation/Explore?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281">rs6696281</a><a class="hidden _zmenu_link" href="http://grch37.ensembl.org/Homo_sapiens/ZMenu/Variation?db=core;tl=G9fJSvE7sdcvGHwt-8953987;v=rs6696281"></a></span></td><td>1</td><td>0.2400</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr></tbody></table></div><div class="dataTables_bottom" style="display: none;"><div class="dataTables_info" style="display: none;">Showing 1 to 30 of 30 entries</div><div class="dataTables_paginate paging_full_numbers" style="display: none;"><span class="first paginate_button paginate_button_disabled">&lt;&lt;</span><span class="previous paginate_button paginate_button_disabled">&lt;</span><span></span><span class="next paginate_button paginate_button_disabled">&gt;</span><span class="last paginate_button paginate_button_disabled">&gt;&gt;</span></div><div class="invisible"></div></div></div>
    <div class="data_table_config"><input type="hidden" name="code" value="Results::"><input type="hidden" name="aaSorting" value="[[1,&quot;asc&quot;]]"><input type="hidden" name="hiddenColumns" value="[&quot;27&quot;,&quot;16&quot;,&quot;23&quot;,&quot;22&quot;,&quot;15&quot;,&quot;28&quot;,&quot;21&quot;,&quot;14&quot;,&quot;30&quot;,&quot;10&quot;,&quot;4&quot;,&quot;18&quot;,&quot;11&quot;,&quot;31&quot;,&quot;24&quot;,&quot;20&quot;,&quot;26&quot;,&quot;12&quot;,&quot;13&quot;,&quot;32&quot;,&quot;25&quot;,&quot;33&quot;]"><input type="hidden" name="bLengthChange" value="false"><input type="hidden" name="bFilter" value="false"><input type="hidden" name="expopts" value="[]"></div>
  </div><div><b>Page: </b><img src="VEP_results_MikeRaiko_files/nav-l2.gif" style="vertical-align:top; height:16px; width:16px; opacity: 0.5;"><img src="VEP_results_MikeRaiko_files/nav-l1.gif" style="vertical-align:top; height:16px; width:16px; opacity: 0.5;"> 1 of 32928 <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=6;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=10" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=6;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=10;update_panel=1"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-position: -596.8px -92px;width: 16px;vertical-align: top;height: 16px;background-size: 800px 624.8px" title="Next page"></a><a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=164636;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=164638" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=164636;size=5;tl=G9fJSvE7sdcvGHwt-8953987;to=164638;update_panel=1"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png " style="background-size: 800px 624.8px;height: 16px;vertical-align: top;background-position: -617.6px -92px;width: 16px" title="Last page"></a><span style="padding: 0px 10px 0px 10px; color: grey">|</span><b>Show: </b>  <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=1;tl=G9fJSvE7sdcvGHwt-8953987;to=1" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=1;tl=G9fJSvE7sdcvGHwt-8953987;to=1;update_panel=1">1</a> <span class="count-highlight">&nbsp;5&nbsp;</span> <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=10;tl=G9fJSvE7sdcvGHwt-8953987;to=10" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=10;tl=G9fJSvE7sdcvGHwt-8953987;to=10;update_panel=1">10</a> <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=50;tl=G9fJSvE7sdcvGHwt-8953987;to=50" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=50;tl=G9fJSvE7sdcvGHwt-8953987;to=50;update_panel=1">50</a> <a href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results?from=1;size=164638;tl=G9fJSvE7sdcvGHwt-8953987;to=164638" class="_reload"><input type="hidden" value="/Homo_sapiens/Component/Tools/VEP/Results/results?from=1;size=164638;tl=G9fJSvE7sdcvGHwt-8953987;to=164638;update_panel=1">All<img class="_ht" src="VEP_results_MikeRaiko_files/alert.png" style="vertical-align: top;"></a> variants</div></div></div></div><div class="ajax initial_panel"><div class="js_panel" id="NewJobButton"><input type="hidden" class="panel_type" value="Content"><a class="button top-margin" href="http://grch37.ensembl.org/Homo_sapiens/Tools/VEP?db=core;expand_form=true;tl=G9fJSvE7sdcvGHwt-8953987">New job</a></div></div><div class="session_messages js_panel" id="ensembl_panel_3"><input type="hidden" class="panel_type" value="Message"></div></div><p class="invisible">.</p>
      
    </div>
  
      </div>
  
        <div id="footer">
          <div class="column-wrapper">
    <div class="column-two left">
      <p>Ensembl GRCh37 release 109 - Feb 2023 ©
        <span class="print_hide"><a href="http://www.ebi.ac.uk/" class="nowrap constant">EMBL-EBI</a></span>
        <span class="screen_hide_inline">EMBL-EBI<br>http://grch37.ensembl.org</span>
      </p>
    </div>
    <div class="column-two right print_hide"><p></p></div>
            <p class="invisible">.</p>
          </div>
          <div class="column-wrapper">
            <p class="invisible">.</p>
          </div>
        </div>
  
      </div>
    </div>
  </div>
  <form id="core_params" action="#" style="display:none">
    <fieldset><input type="hidden" name="tl" value="G9fJSvE7sdcvGHwt-8953987"><input type="hidden" name="db" value="core"></fieldset>
  </form>
  <input type="hidden" id="species_path" name="species_path" value="/Homo_sapiens">
  <input type="hidden" id="gdpr_version" name="gdpr_version" value="2.0.0">
  <input type="hidden" id="gdpr_policy_url" name="gdpr_policy_url" value="https://www.ebi.ac.uk/data-protection/ensembl/privacy-notice">
  <input type="hidden" id="gdpr_terms_url" name="gdpr_terms_url" value="https://www.ebi.ac.uk/about/terms-of-use">
  <input type="hidden" id="gdpr_cookie_name" name="gdpr_cookie_name" value="ENSEMBL_PRIVACY_POLICY">
  <input type="hidden" id="species_common_name" name="species_common_name" value="Human">
  <input type="hidden" id="ensembl_image_root" name="ensembl_image_root" value="/i/">
  <input type="hidden" id="max_region_length" name="max_region_length" value="1000100">
  <input type="hidden" id="ensembl_species_image" name="ensembl_species_image" value="/i/species/Homo_sapiens.png">
    
  <div id="modal_bg" style="display: none;"></div>
  <div id="modal_panel" class="js_panel ui-draggable ui-resizable" style="height: 602px; width: 1319px; top: 75px; left: 73.5px; display: none;">
    <input type="hidden" class="panel_type" value="ModalContainer">
    <div class="modal_title">
      <ul class="tabs" style="display: none;">
        <li><a class="modal_user_data" href="http://grch37.ensembl.org/Homo_sapiens/UserData/ManageData?time=1676479822">Personal Data</a></li>
      </ul>
      <div class="modal_caption"></div>
      <div class="modal_close" title="Close" alt="Close"></div>
    </div>
    <div id="modal_user_data" class="modal_content js_panel fixed_width active" style="display:none"></div>
    <div id="modal_default" class="modal_content js_panel fixed_width" style=""><div class="modal_nav nav">
    <input type="hidden" class="panel_type" value="LocalContext">
    <div class="header">Personal Data</div>
    <ul class="local_context"><li class="active top_level"><a title="Login" class="Login" href="http://grch37.ensembl.org/Multi/Account/Login">Login</a></li><li class="top_level"><a href="http://grch37.ensembl.org/Multi/Account/Register" class="Register" title="Register">Register</a></li><li class="top_level"><a href="http://grch37.ensembl.org/Multi/Account/Password/Lost" class="Lost_Password" title="Lost Password">Lost Password</a></li><li class="last top_level"><a href="http://grch37.ensembl.org/Multi/Account/Help" class="Help" title="Help">Help</a></li></ul></div><div class="modal_wrapper"><div class="panel"><div class="content"><div class="js_panel" id="Urgent"><div class="session_messages"></div></div><div class="js_panel" id="Login"><div><div><form id="login" method="post" class="_check check bgcolour std" action="/Multi/Account/User/Authenticate"><fieldset><input value="/Multi/Account/Bookmark/Add?description=Variant Effect Predictor results - Homo_sapiens - GRCh37 Archive browser 109;url=//grch37.ensembl.org/Homo_sapiens/Tools/VEP/Results%3Ftl%3DG9fJSvE7sdcvGHwt-8953987" type="hidden" data-role="none" name="then"><input name="modal_tab" data-role="none" type="hidden" value="modal_user_data"><div class="form-field"><label for="cZpnc8Om_1" class="ff-label">Email:</label><div class="ff-right"><input name="email" data-role="none" type="text" class="required _email ftext" id="cZpnc8Om_1"><span class="snote"> <strong title="Required field">*</strong> </span></div></div><div class="form-field"><label class="ff-label" for="cZpnc8Om_2">Password:</label><div class="ff-right"><input data-role="none" type="password" name="password" id="cZpnc8Om_2" class="ftext _password required"><span class="snote"> <strong title="Required field">*</strong> </span></div></div><div class="form-field ff-multi"><div class="ff-right"><input class="fbutton" id="cZpnc8Om_3" value="Log in" name="submit" data-role="none" type="submit"></div><div class="ff-notes"><p><a href="http://grch37.ensembl.org/Multi/Account/Password/Lost" class="modal_link">Lost password</a> | <a href="http://grch37.ensembl.org/Multi/Account/Register" class="modal_link">Register</a></p></div></div><div class="fnotes">Fields marked <b>*</b> are required</div></fieldset></form></div></div></div><div class="js_panel" id="Other"><div class="session_messages"></div></div></div></div></div></div>
    <div class="modal_overlay" style="display: none;"><img src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" class="autosprite-src-default-png overlay_close" style="background-size: 1000px 781px;height: 20px;width: 20px;background-position: -642px -115px" alt="close" title="Cancel"><div class="overlay_content"></div></div>
    <div class="modal_overlay_bg" style="display: none;"></div>
  <div class="ui-resizable-handle ui-resizable-e" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-s" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-se ui-icon ui-icon-gripsmall-diagonal-se" style="z-index: 90;"></div></div>
  
    
    
  <input type="hidden" id="hidden_species_list" name="hidden_species_list" value="Homo_sapiens">


<ul class="ui-autocomplete ui-front ui-menu ui-widget ui-widget-content ui-corner-all" id="ui-id-1" tabindex="0" style="display: none;"></ul><div id="ui-tooltip-0" role="tooltip" class="ui-tooltip ui-widget ui-corner-all ui-widget-content helptip-bottom" style="top: 1037px; left: 1211px; display: block; cursor: default;"><div class="ui-tooltip-content">Scroll to see more columns »</div></div><img src="VEP_results_MikeRaiko_files/dc70e5b4bed7893237228c404e2992b3.jpg" style="display: none"><img src="VEP_results_MikeRaiko_files/02a60942bb4c3f6fa98d4c07eeb89af7.png" style="display: none"><img src="VEP_results_MikeRaiko_files/92a5cd07513546550b5d607d385f401f.png" style="display: none"><img src="VEP_results_MikeRaiko_files/b07ae2d13f623431d4c5599a9ed36286.png" style="display: none"><img src="VEP_results_MikeRaiko_files/80781a8a7c2a789249be5426971bb266.png" style="display: none"><img src="VEP_results_MikeRaiko_files/c426c975e6ef844f7ebdfb248abce5af.png" style="display: none"><img src="VEP_results_MikeRaiko_files/2233bda0867971b025fbe642f50aef47.png" style="display: none"></body><grammarly-desktop-integration data-grammarly-shadow-root="true"></grammarly-desktop-integration></html>