{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9e204b71",
   "metadata": {},
   "source": [
    "# Project 3. E.coli outbreak investigation. De novo assembly and annotation of bacterial genomes.\n",
    "\n",
    "Lab notebook by Nikita Vyatkin"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42310685",
   "metadata": {},
   "source": [
    "This week, you will follow in the footsteps of the bioinformaticians investigating the outbreak by assembling the genome of the deadly E. coli X strain. Specifically, we will provide you with Illumina reads from the TY2482 sample, which were generated at Beijing Genome Institute and deposited into the Short Read Archive (http://www.ncbi.nlm.nih.gov/sra) for public access. After we have assembled the E. coli X genome, we will determine which other strain(s) it is most similar to in an effort to determine how it could have arisen. In particular, once we know where it came from, we can ask what new genes it possesses for pathogenicity, and how it evolved these new functions.\n",
    "\n",
    "Our workflow will be to answer the following questions:\n",
    "1. What is the genome sequence of E. coli X?\n",
    "2. What strain of E. coli is E. coli X most similar to? (Where did it come from?)\n",
    "3. What are the genes that E. coli X contains?\n",
    "4. Which of these genes make E. coli X distinct?\n",
    "5. How did E. coli X evolve to obtain these genes?\n",
    "6. How did E. coli X become pathogenic?\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b2d804f",
   "metadata": {},
   "source": [
    "## Assembling the Genome\n",
    "\n",
    "DNA sequencing technology cannot read whole genomes in one go, so we will assemble genome from the short reads generated by the Illumina instrument. We will use the [SPAdes](https://cab.spbu.ru/software/spades/) assembler (Bankevich et al., 2012) to assemble the E. coli X genome from reads into contigs.\n",
    "\n",
    "A contig is a “contiguous” segment of the genome that has been reconstructed by an assembly algorithm. In addition to contigs, SPAdes uses information about the distances between reads within read-pairs (called insert size) to combine contigs into ordered collections of adjacent contigs called scaffolds.Genome assembly programs often fill a gap in a scaffold of length m by a sequence of m occurrences of “N” (a placeholder for unknown nucleotides).\n",
    "\n",
    "In modern DNA sequencing projects, DNA fragments are sequenced from both the 5' and 3' ends, giving rise to paired reads separated by some insert size. The forward and reverse read in a paired read are generated from the forward and reverse strand, respectively. A collection of paired reads is called a sequencing library; whereas most paired libraries generate reads with insert sizes below 1 kb, libraries with longer (2 kb-10 kb) insert sizes are called mate-pairs.\n",
    "\n",
    "Many sequencing projects generate several libraries of paired reads with different insert sizes; for example, the sequencing project at the center of this task employs three libraries of insert lengths 270, 2000, and 6000 nucleotides.\n",
    "\n",
    "The advantage of using multiple libraries is that libraries with small insert sizes are better suited for resolving short repeats, whereas libraries with larger insert sizes are better suited for resolving long repeats. For example, all bacterial genomes contain ribosomal operons, which are often repeated more than six times and are approximately 5000 nucleotides long. Thus, to resolve ribosomal operons, we need mate-pair libraries with insert sizes exceeding 5000 nt.\n",
    "\n",
    "Once we have sequenced a genome, there are many metrics for assessing the resulting assembly. We will use QUAST (Gurevich et al., 2013), which takes many metrics into account to evaluate the overall quality of an assembly.\n",
    "\n",
    "One of the common metrics is the N50 statistic. N50 is defined as the maximal contig length for which all contigs greater than or equal to that length comprise at least half of the sum of the lengths of all the contigs. For example, consider the five toy contigs with the following lengths: [10, 20, 30, 60, 70]. Here, the total length of contigs is 190, and contigs of length 60 and 70 account for at least 50% of the total length of contigs (60 + 70 = 130), but the contig of length 70 does not account for 50% of the total length of contigs. Thus, N50 is equal to 60."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60fd9182",
   "metadata": {},
   "source": [
    "## Dataset\n",
    "\n",
    "For this project, we provide three libraries from the TY2482 sample with the following\n",
    "insert sizes and orientation:\n",
    "* SRR292678 - paired end, insert size 470 bp ([forward reads](https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R1_001.fastq.gz), [reverse reads](https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R2_001.fastq.gz), 400 Mb each)\n",
    "* SRR292862 - mate pair, insert size 2 kb, ([forward reads](https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R1_001.fastq.gz), [reverse reads](https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz), 200 Mb each)\n",
    "* SRR292770 - mate pair, insert size 6 kb, ([forward reads](https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R1_001.fastq.gz), [reverse reads](https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R2_001.fastq.gz), 200 Mb each)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0aea8b9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir raw_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7b90652b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-11-21 18:03:20--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R1_001.fastq.gz\n",
      "Resolving d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)... 13.33.244.215, 13.33.244.96, 13.33.244.69, ...\n",
      "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|13.33.244.215|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 425982530 (406M) [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292678sub_S1_L001_R1_001.fastq.gz’\n",
      "\n",
      "SRR292678sub_S1_L00 100%[===================>] 406,25M  3,62MB/s    in 2m 1s   \n",
      "\n",
      "2022-11-21 18:05:21 (3,37 MB/s) - ‘raw_data/SRR292678sub_S1_L001_R1_001.fastq.gz’ saved [425982530/425982530]\n",
      "\n",
      "--2022-11-21 18:05:21--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R2_001.fastq.gz\n",
      "Reusing existing connection to d28rh4a8wq0iu5.cloudfront.net:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 426246191 (406M) [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292678sub_S1_L001_R2_001.fastq.gz’\n",
      "\n",
      "SRR292678sub_S1_L00 100%[===================>] 406,50M  1,22MB/s    in 2m 21s  \n",
      "\n",
      "2022-11-21 18:07:42 (2,89 MB/s) - ‘raw_data/SRR292678sub_S1_L001_R2_001.fastq.gz’ saved [426246191/426246191]\n",
      "\n",
      "--2022-11-21 18:07:42--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to d28rh4a8wq0iu5.cloudfront.net:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 226372891 (216M) [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292862_S2_L001_R1_001.fastq.gz’\n",
      "\n",
      "SRR292862_S2_L001_R 100%[===================>] 215,89M  3,57MB/s    in 60s     \n",
      "\n",
      "2022-11-21 18:08:43 (3,57 MB/s) - ‘raw_data/SRR292862_S2_L001_R1_001.fastq.gz’ saved [226372891/226372891]\n",
      "\n",
      "--2022-11-21 18:08:43--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz\n",
      "Reusing existing connection to d28rh4a8wq0iu5.cloudfront.net:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 231496590 (221M) [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292862_S2_L001_R2_001.fastq.gz’\n",
      "\n",
      "SRR292862_S2_L001_R  72%[=============>      ] 160,64M  3,61MB/s    in 44s     \n",
      "\n",
      "2022-11-21 18:24:29 (3,65 MB/s) - Read error at byte 168446833/231496590 (Success). Retrying.\n",
      "\n",
      "--2022-11-21 18:24:30--  (try: 2)  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz\n",
      "Connecting to d28rh4a8wq0iu5.cloudfront.net (d28rh4a8wq0iu5.cloudfront.net)|13.33.244.215|:443... connected.\n",
      "HTTP request sent, awaiting response... 206 Partial Content\n",
      "Length: 231496590 (221M), 63049757 (60M) remaining [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292862_S2_L001_R2_001.fastq.gz’\n",
      "\n",
      "SRR292862_S2_L001_R 100%[++++++++++++++=====>] 220,77M  3,62MB/s    in 17s     \n",
      "\n",
      "2022-11-21 18:24:47 (3,60 MB/s) - ‘raw_data/SRR292862_S2_L001_R2_001.fastq.gz’ saved [231496590/231496590]\n",
      "\n",
      "--2022-11-21 18:24:47--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to d28rh4a8wq0iu5.cloudfront.net:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 226840159 (216M) [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292770_S1_L001_R1_001.fastq.gz’\n",
      "\n",
      "SRR292770_S1_L001_R 100%[===================>] 216,33M  2,72MB/s    in 70s     \n",
      "\n",
      "2022-11-21 18:26:03 (3,11 MB/s) - ‘raw_data/SRR292770_S1_L001_R1_001.fastq.gz’ saved [226840159/226840159]\n",
      "\n",
      "--2022-11-21 18:26:03--  https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R2_001.fastq.gz\n",
      "Reusing existing connection to d28rh4a8wq0iu5.cloudfront.net:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 232544438 (222M) [application/x-gzip]\n",
      "Saving to: ‘raw_data/SRR292770_S1_L001_R2_001.fastq.gz’\n",
      "\n",
      "SRR292770_S1_L001_R 100%[===================>] 221,77M  1,53MB/s    in 78s     \n",
      "\n",
      "2022-11-21 18:27:22 (2,83 MB/s) - ‘raw_data/SRR292770_S1_L001_R2_001.fastq.gz’ saved [232544438/232544438]\n",
      "\n",
      "FINISHED --2022-11-21 18:27:22--\n",
      "Total wall clock time: 24m 2s\n",
      "Downloaded: 6 files, 1,5G in 8m 50s (2,88 MB/s)\n"
     ]
    }
   ],
   "source": [
    "! wget -P raw_data https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R1_001.fastq.gz https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292678sub_S1_L001_R2_001.fastq.gz https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R1_001.fastq.gz https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292862_S2_L001_R2_001.fastq.gz https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R1_001.fastq.gz https://d28rh4a8wq0iu5.cloudfront.net/bioinfo/SRR292770_S1_L001_R2_001.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "98d1dd09",
   "metadata": {},
   "outputs": [],
   "source": [
    "! gunzip raw_data/*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76450c25",
   "metadata": {},
   "source": [
    "### Simple manual data analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d2eeb109",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of reads in raw_data/SRR292678sub_S1_L001_R1_001.fastq: 5,499,346\n",
      "Number of reads in raw_data/SRR292678sub_S1_L001_R2_001.fastq: 5,499,346\n",
      "Number of reads in raw_data/SRR292770_S1_L001_R1_001.fastq: 5,102,041\n",
      "Number of reads in raw_data/SRR292770_S1_L001_R2_001.fastq: 5,102,041\n",
      "Number of reads in raw_data/SRR292862_S2_L001_R1_001.fastq: 5,102,041\n",
      "Number of reads in raw_data/SRR292862_S2_L001_R2_001.fastq: 5,102,041\n",
      "Number of reads in total: 31,406,856\n"
     ]
    }
   ],
   "source": [
    "out = ! wc -l raw_data/*.fastq\n",
    "\n",
    "out = [s.split() for s in out]\n",
    "\n",
    "for item in out:\n",
    "    print(f\"Number of reads in {item[1]}: {int(item[0]) // 4:,}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1ef5e2b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir fastqc_reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "999b7ab0",
   "metadata": {},
   "outputs": [],
   "source": [
    "! fastqc -q -t 6 -o fastqc_reports/ raw_data/*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ecc03e3",
   "metadata": {},
   "source": [
    "## K-mer profile and genome size estimation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4d0375ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: jellyfish count [options] file:path+\r\n",
      "\r\n",
      "Count k-mers in fasta or fastq files\r\n",
      "\r\n",
      "Options (default value in (), *required):\r\n",
      " -m, --mer-len=uint32                    *Length of mer\r\n",
      " -s, --size=uint64                       *Initial hash size\r\n",
      " -t, --threads=uint32                     Number of threads (1)\r\n",
      "     --sam=PATH                           SAM/BAM/CRAM formatted input file\r\n",
      " -F, --Files=uint32                       Number files open simultaneously (1)\r\n",
      " -g, --generator=path                     File of commands generating fast[aq]\r\n",
      " -G, --Generators=uint32                  Number of generators run simultaneously (1)\r\n",
      " -S, --shell=string                       Shell used to run generator commands ($SHELL or /bin/sh)\r\n",
      " -o, --output=string                      Output file (mer_counts.jf)\r\n",
      " -c, --counter-len=Length in bits         Length bits of counting field (7)\r\n",
      "     --out-counter-len=Length in bytes    Length in bytes of counter field in output (4)\r\n",
      " -C, --canonical                          Count both strand, canonical representation (false)\r\n",
      "     --bc=peath                           Bloom counter to filter out singleton mers\r\n",
      "     --bf-size=uint64                     Use bloom filter to count high-frequency mers\r\n",
      "     --bf-fp=double                       False positive rate of bloom filter (0.01)\r\n",
      "     --if=path                            Count only k-mers in this files\r\n",
      " -Q, --min-qual-char=string               Any base with quality below this character is changed to N\r\n",
      "     --quality-start=int32                ASCII for quality values (64)\r\n",
      "     --min-quality=int32                  Minimum quality. A base with lesser quality becomes an N\r\n",
      " -p, --reprobes=uint32                    Maximum number of reprobes (126)\r\n",
      "     --text                               Dump in text format (false)\r\n",
      "     --disk                               Disk operation. Do not do size doubling (false)\r\n",
      " -L, --lower-count=uint64                 Don't output k-mer with count < lower-count\r\n",
      " -U, --upper-count=uint64                 Don't output k-mer with count > upper-count\r\n",
      "     --timing=Timing file                 Print timing information\r\n",
      "     --usage                              Usage\r\n",
      " -h, --help                               This message\r\n",
      "     --full-help                          Detailed help\r\n",
      " -V, --version                            Version\r\n"
     ]
    }
   ],
   "source": [
    "! jellyfish count --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "49f5f27d",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir jellyfish_analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f91083e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "! jellyfish count -C -t 8 -s 20000000 -m 31 -o jellyfish_analysis/kmer_counts_31.jf raw_data/SRR292678sub_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc6f7dd4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c7adca6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f42b891f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4422b67",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfaf1b9e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3923be7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a80bc4fd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0eeffa46",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b0ef854",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
